---
title: "Analysis for APOBEC HNSCC manuscript"
output:
  html_document:
    df_print: paged
---


```{r import data}

#import the HPV status data from 
# library(feather)
# apobec_hnscc_df <- read_feather("input_data/nfkb.feather")
# unique(apobec_hnscc_df$patient_id) # All TCGA samples

virusscan.data <- read.table(file = "input_data/virusscan/vscan_counts.tsv",sep = "\t",header = T,stringsAsFactors = F)

# load("input_data/trinuc_data_HNSC_HPVpos.RData")
# load("input_data/trinuc_data_HNSC_final_call_HPVneg_unfiltered.RData")

```

# processing MAF file and splitting into HPV pos and neg

```{r loading and splitting MAF file}

HNSC.MAF <- read.csv("input_data/NCI/gdc_download_20180201_160847/1aa33f25-3893-4f37-a6a4-361c9785d07e/TCGA.HNSC.mutect.1aa33f25-3893-4f37-a6a4-361c9785d07e.DR-10.0.somatic.maf",skip=5,header = T,sep = "\t",stringsAsFactors = F)
source("R/flip_function.R")
source("R/unique_tumor_addition.R")
source("R/hg39_to_hg19_converter.R")
source("R/DNP_remover.R")
source("R/tumor_allele_adder.R")


HNSC.MAF <- hg38.to.hg19.converter(chain = "input_data/hg38Tohg19.chain",hg38_maf = HNSC.MAF)
HNSC.MAF <- DNP.remover(MAF = HNSC.MAF)
HNSC.MAF <- tumor.allele.adder(MAF = HNSC.MAF)


HNSC.MAF$HPV_call <- NA
HNSC.MAF$Virusscan_counts <- NA

for(i in 1:length(unique(HNSC.MAF$Unique_patient_identifier))){
  if(length(which(virusscan.data$patient_id==unique(HNSC.MAF$Unique_patient_identifier)[i]))>0){
    HNSC.MAF$HPV_call[HNSC.MAF$Unique_patient_identifier==unique(HNSC.MAF$Unique_patient_identifier)[i]] <- 
      #apobec_hnscc_df$categ[which(apobec_hnscc_df$patient_id==unique(HNSC.MAF$Unique_patient_identifier)[i])]
      ifelse(virusscan.data$VScan_counts[which(virusscan.data$patient_id==unique(HNSC.MAF$Unique_patient_identifier)[i])]>100,
             "HPV+",
             "HPV-")
    HNSC.MAF$Virusscan_counts[HNSC.MAF$Unique_patient_identifier==unique(HNSC.MAF$Unique_patient_identifier)[i]] <- 
      virusscan.data$VScan_counts[which(virusscan.data$patient_id==unique(HNSC.MAF$Unique_patient_identifier)[i])]
  }
}

# HNSC.MAF$HPV_call[which(HNSC.MAF$Unique_patient_identifier=="TCGA-CN-A6V1")] <- "ambiguous"

# unique(HNSC.MAF$Unique_patient_identifier)
# unique(HNSC.MAF$HPV_call)
# write.table(x = HNSC.MAF[,c("Unique_patient_identifier","HPV_call","Virusscan_counts")],file = "output_data/testing_MAF_and_counts.txt",quote = F,sep = "\t",row.names = F)

HNSC.MAF.hpvpos <- HNSC.MAF[which(HNSC.MAF$HPV_call=="HPV+"),]
MAF_for_analysis <- HNSC.MAF.hpvpos
length(unique(MAF_for_analysis$Unique_patient_identifier))
# MAF_for_analysis[which(is.na(MAF_for_analysis$Unique_patient_identifier)),]
save(MAF_for_analysis, file="output_data/HNSC_HPVpos_MAF.RData")

HNSC.MAF.hpvneg <- HNSC.MAF[which(HNSC.MAF$HPV_call=="HPV-"),]
MAF_for_analysis <- HNSC.MAF.hpvneg
length(unique(HNSC.MAF.hpvneg$Unique_patient_identifier))
save(MAF_for_analysis, file="output_data/HNSC_HPVneg_MAF.RData")


MAF_for_analysis <- HNSC.MAF
save(MAF_for_analysis, file="output_data/HNSC_MAF.RData")

message("HPV positive tumor with TP53 mutation:") 
HNSC.MAF.hpvpos$Unique_patient_identifier[which(HNSC.MAF.hpvpos$Hugo_Symbol=="TP53")]

```

# trinucleotide heatmaps from cluster

```{r Figure trinuc context HPVpos, fig.height=2.5, fig.width=10}
library(ggplot2)
load("output_data/selection_from_cluster/HNSC_HPVpos/trinuc_output/trinuc_data_HNSC_HPVpos.RData")
HPV.pos.trinuc.mutation_data <- trinuc.mutation_data
HPV.pos.trinuc.heatmap <- ggplot(data=HPV.pos.trinuc.mutation_data, aes(Downstream, Upstream)) +
  geom_tile(aes(fill = proportion*100), colour = "white") + scale_fill_gradient(low = "white", high = "steelblue", name="Percent")
HPV.pos.trinuc.heatmap <- HPV.pos.trinuc.heatmap + facet_grid(.~section_labels, labeller = label_parsed) 
HPV.pos.trinuc.heatmap <- HPV.pos.trinuc.heatmap +  geom_text(aes(label = round(proportion, 4)*100),size=2)
# p <- p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
# panel.background = element_blank(), axis.line = element_line(colour = "black"))
HPV.pos.trinuc.heatmap <- HPV.pos.trinuc.heatmap + theme_bw() + theme(panel.grid.major = element_blank(),
                                                                      panel.grid.minor = element_blank(),
                                                                      axis.ticks = element_blank(),
                                                                      strip.text=element_text(size=15),
                                                                      axis.title.x = element_text(size=15),
                                                                      axis.title.y = element_text(size=15),
                                                                      axis.text.x = element_text(size=12),
                                                                      axis.text.y=element_text(size=12),plot.title = element_text(hjust = 0.5),legend.position = "left") 
# ggtitle(paste("Trinucleotide profile for ",tumor.name,sep=""))
# p


HPV.pos.trinuc.heatmap

ggsave(paste("Figures/","HPVpos","_trinuc_heatmap.png",sep=""),height = 1.5,width = 7,plot = HPV.pos.trinuc.heatmap,dpi=300)

```

```{r Figure trinuc context HPVneg, fig.height=2.5, fig.width=10}
load("output_data/selection_from_cluster/HNSC_HPVneg/trinuc_output/trinuc_data_HNSC_HPVneg.RData")
HPV.neg.trinuc.mutation_data <- trinuc.mutation_data
HPV.neg.trinuc.heatmap <- ggplot(data=HPV.neg.trinuc.mutation_data, aes(Downstream, Upstream)) +
  geom_tile(aes(fill = proportion*100), colour = "white") + scale_fill_gradient(low = "white", high = "steelblue", name="Percent")
HPV.neg.trinuc.heatmap <- HPV.neg.trinuc.heatmap + facet_grid(.~section_labels, labeller = label_parsed) 
HPV.neg.trinuc.heatmap <- HPV.neg.trinuc.heatmap +  geom_text(aes(label = round(proportion, 4)*100),size=2)
# p <- p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
# panel.background = element_blank(), axis.line = element_line(colour = "black"))
HPV.neg.trinuc.heatmap <- HPV.neg.trinuc.heatmap + theme_bw() + theme(panel.grid.major = element_blank(),
                                                                      panel.grid.minor = element_blank(),
                                                                      axis.ticks = element_blank(),
                                                                      strip.text=element_text(size=15),
                                                                      axis.title.x = element_text(size=15),
                                                                      axis.title.y = element_text(size=15),
                                                                      axis.text.x = element_text(size=12),
                                                                      axis.text.y=element_text(size=12),plot.title = element_text(hjust = 0.5),legend.position = "left") 
# ggtitle(paste("Trinucleotide profile for ",tumor.name,sep=""))
# p

HPV.neg.trinuc.heatmap

ggsave(filename = paste("Figures/","HPVneg","_trinuc_heatmap.png",sep=""),height = 1.5,width = 7,plot = HPV.neg.trinuc.heatmap,dpi = 300)

```





```{r mutation rate of HPV positive vs negative}

HPV.pos.mut.rates <- read.csv(file = "output_data/selection_from_cluster/HNSC_HPVpos/mutsig_output/MAF_HNSC_HPVpos.txt.gene_rates.txt",header = T,sep = "\t",stringsAsFactors = F)
# head(HPV.pos.mut.rates)

HPV.neg.mut.rates <- read.csv(file = "output_data/selection_from_cluster/HNSC_HPVneg/mutsig_output/MAF_HNSC_HPVneg.txt.gene_rates.txt",header = T,sep = "\t",stringsAsFactors = F)
# head(HPV.neg.mut.rates)

# HPV.neg.mut.rates


all.equal(HPV.pos.mut.rates$gene,HPV.neg.mut.rates$gene)

# HPV.neg.mut.rates$r_x_X[which(HPV.neg.mut.rates$gene=="PIK3CA")]
# HPV.pos.mut.rates$r_x_X[which(HPV.pos.mut.rates$gene=="PIK3CA")]

mutation_rates <- data.frame(gene=HPV.pos.mut.rates$gene,positive_mut_rates=HPV.pos.mut.rates$r_x_X,negative_mut_rates=HPV.neg.mut.rates$r_x_X)
library(ggrepel)

# mutation_rates[which(mutation_rates$gene=="PIK3CA"),]

source("R/fancy_scientific_code.R")




mutation_rates_full_scatter <- ggplot(data = mutation_rates, aes(x=positive_mut_rates,y=negative_mut_rates)) +
  geom_point(alpha=0.2,col="black",size=0.5) + 
  geom_smooth(method='lm',color="red") + 
  # geom_text_repel(data=subset(mutation_rates, positive_mut_rates > 5e-5 | negative_mut_rates > 3e-5 ), aes(label=gene),size=3) + 
  geom_point(data = subset(mutation_rates, gene %in% c("FBXW7","PIK3CA","MAPK1")),aes(x=positive_mut_rates,y=negative_mut_rates),size=3,col="blue") + 
  geom_text_repel(data = subset(mutation_rates, gene %in% c("FBXW7","PIK3CA","MAPK1")),aes(x=positive_mut_rates,y=negative_mut_rates,label=gene),size=5,col="blue",fontface="bold") +
  labs(x="Mutation rate in HPV+ tumors",y="Mutation rate in HPV- tumors") + 
  coord_equal(ratio=1) + 
  theme_bw() +
  geom_abline(slope=1, intercept=0) + 
  scale_x_continuous(labels=fancy_scientific) + 
  scale_y_continuous(labels=fancy_scientific) + theme(plot.margin = margin(r=.2,unit = "in"))


mutation_rates_reduced <- ggplot(data = mutation_rates, aes(x=positive_mut_rates,y=negative_mut_rates)) + geom_point(alpha=0.2,col="black",size=0.5) + geom_point(data = subset(mutation_rates, gene %in% c("FBXW7","PIK3CA","MAPK1")),aes(x=positive_mut_rates,y=negative_mut_rates),size=3,col="blue") + geom_text_repel(data = subset(mutation_rates, gene %in% c("FBXW7","PIK3CA","MAPK1")),aes(x=positive_mut_rates,y=negative_mut_rates,label=gene),size=6,col="blue",fontface="bold") +
  labs(x="Mutation rate in HPV+ tumors",y="Mutation rate in HPV- tumors") + coord_equal(ratio=1,xlim=c(0,0.5e-5),ylim=c(0,0.5e-5)) + theme_bw() + geom_smooth(method='lm',formula=y~x,color="red") + geom_abline(slope=1, intercept=0) + scale_x_continuous(labels=fancy_scientific) + scale_y_continuous(labels=fancy_scientific) 

# mutation_rates_full_scatter <- ggplot(data = mutation_rates, aes(y=positive_mut_rates,x=negative_mut_rates)) +
#   geom_point(alpha=0.2,col="black",size=0.5) + 
#     geom_smooth(method='lm',color="red") + 
#   # geom_text_repel(data=subset(mutation_rates, positive_mut_rates > 5e-5 | negative_mut_rates > 3e-5 ), aes(label=gene),size=3) + 
#   geom_point(data = subset(mutation_rates, gene %in% c("FBXW7","PIK3CA","MAPK1")),aes(y=positive_mut_rates,x=negative_mut_rates),size=3,col="blue") + 
#   geom_text_repel(data = subset(mutation_rates, gene %in% c("FBXW7","PIK3CA","MAPK1")),aes(y=positive_mut_rates,x=negative_mut_rates,label=gene),size=5,col="blue",fontface="bold",box.padding = 5) +
#   labs(y=bquote("Mutation rate in "~HPV^{"-"}~ "\n tumors"), x=bquote("Mutation rate in "~HPV^{"+"}~ "tumors"))+ 
#   coord_equal(ratio=1) + 
#   theme_bw() + 
#   geom_abline(slope=1, intercept=0) + 
#   scale_x_continuous(labels=fancy_scientific) + 
#   scale_y_continuous(labels=fancy_scientific) + theme(plot.margin = margin(0, 1, 0, 0,unit = "in"))

mutation_rates_full_scatter

summary(lm(data = mutation_rates,formula = positive_mut_rates~negative_mut_rates))

ggsave(plot = mutation_rates_full_scatter,filename = "Figures/mutation_rates_full_scatter.png",height = 3,width = 8,dpi=300)

ggsave(plot = mutation_rates_reduced,filename = "Figures/mutation_rates_reduced.png",height = 2.5,width = 2.5)


# ggplot(data = mutation_rates, aes(x = positive_mut_rates, y=negative_mut_rates)) + geom_point() + geom_smooth() + coord_equal()
# ggplot(data = mutation_rates, aes(y = positive_mut_rates, x=negative_mut_rates)) + geom_point() + geom_smooth() + coord_equal()

```





# prevalence-prevalence and gamma-gamma plots
```{r trinuc with weights}
library(reshape2)
load("output_data/HNSC_MAF.RData")
HNSC.MAF <- MAF_for_analysis
source("R/trinuc_signatures_w_weights.R")
trinuc.contexts <- trinuc.profile.function_withweights(input.MAF = HNSC.MAF,
                                                       signature.choice = "signatures.cosmic", 
                                                       minimum.mutations.per.tumor=50,save.figs=F)
length(trinuc.contexts[[2]])

save(trinuc.contexts,file="output_data/trinuc_contexts_HNSC_MAF.RData")




```




```{r load selection output and merge with APOBEC and HPV status}
library(tidyverse)
load("output_data/selection_from_cluster/HNSC/selection_output/HNSC_all_selection_output.RData")
HNSC.selection.output <- as.tibble(selection.output$complete_mutation_data) %>%
  select(Gene, starts_with("Nucleo"), 
         Chromosome, starts_with("Reference"),
         starts_with("Alternative"), Tumor_origin, 
         Unique_patient_identifier, starts_with("Amino"), Codon_position, synonymous.mu, trinucs, Gamma_epistasis)  
# HNSC.selection.output

# Assigning HPV status
load("output_data/HNSC_MAF.RData")
HNSC.MAF <- MAF_for_analysis
HNSC.selection.output$HPV_call <- NA
for(i in 1:length(unique(HNSC.selection.output$Unique_patient_identifier))){
  HNSC.selection.output$HPV_call[which(HNSC.selection.output$Unique_patient_identifier == unique(HNSC.selection.output$Unique_patient_identifier)[i])] <- HNSC.MAF$HPV_call[which(HNSC.MAF$Unique_patient_identifier == unique(HNSC.selection.output$Unique_patient_identifier)[i])[1]]
}


load("output_data/trinuc_contexts_HNSC_MAF.RData")

weights.df <- trinuc.contexts$signature.weights[[1]]$weights
for(i in 2:length(trinuc.contexts$signature.weights)){
  weights.df <- rbind(weights.df,trinuc.contexts$signature.weights[[i]]$weights)
}

HNSC.selection.output$Sig_2 <- NA
HNSC.selection.output$Sig_13 <- NA


for(i in 1:length(unique(HNSC.selection.output$Unique_patient_identifier))){
  if(length(which(rownames(weights.df) == unique(HNSC.selection.output$Unique_patient_identifier)[i]))>0){
    HNSC.selection.output$Sig_2[which(HNSC.selection.output$Unique_patient_identifier == unique(HNSC.selection.output$Unique_patient_identifier)[i])] <- weights.df$Signature.2[which(rownames(weights.df) == unique(HNSC.selection.output$Unique_patient_identifier)[i])]
    HNSC.selection.output$Sig_13[which(HNSC.selection.output$Unique_patient_identifier == unique(HNSC.selection.output$Unique_patient_identifier)[i])] <- weights.df$Signature.13[which(rownames(weights.df) == unique(HNSC.selection.output$Unique_patient_identifier)[i])]
  }
}


HNSC.selection.output <- mutate(.data = HNSC.selection.output, APOBEC_weight = Sig_2 + Sig_13)

# head(HNSC.selection.output)






HNSC.selection.output$TCW_TKW <- NA
HNSC.selection.output$TCN_TKN <- NA

for(j in 1:nrow(HNSC.selection.output)){
  if(is.na(HNSC.selection.output$Nucleotide_trinuc_context[j])){
    #need to make a call of the same nucleotide position and amino acid alternative, find which is NOT NA, and make this the new "j" 
    matches <- which(HNSC.selection.output$Nucleotide_chromosome_position == HNSC.selection.output$Nucleotide_chromosome_position[j] &
                       HNSC.selection.output$Alternative_Nucleotide == HNSC.selection.output$Alternative_Nucleotide[j])
    
    new.j <- matches[which(!is.na(HNSC.selection.output$Nucleotide_trinuc_context[matches]))]
    if(((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="T") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="A")) | 
       ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="T") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="A")) |
       ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="G") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="C")) |
       ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="G") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="C"))){
      HNSC.selection.output$TCW_TKW[j] <- 1
    }else{
      HNSC.selection.output$TCW_TKW[j] <- 0
    }
    
  }else{
    if(((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[j]=="T") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="A")) | 
       ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[j]=="T") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="A")) |
       ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[j]=="G") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="C")) |
       ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[j]=="G") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="C"))){
      HNSC.selection.output$TCW_TKW[j] <- 1
    }else{
      HNSC.selection.output$TCW_TKW[j] <- 0
    }
  }
  
  # Now, mutations that could be TCN --> TKN
  if(is.na(HNSC.selection.output$Nucleotide_trinuc_context[j])){
    #need to make a call of the same nucleotide position and amino acid alternative, find which is NOT NA, and make this the new "j" 
    matches <- which(HNSC.selection.output$Nucleotide_chromosome_position == HNSC.selection.output$Nucleotide_chromosome_position[j] &
                       HNSC.selection.output$Alternative_Nucleotide == HNSC.selection.output$Alternative_Nucleotide[j])
    
    new.j <- matches[which(!is.na(HNSC.selection.output$Nucleotide_trinuc_context[matches]))]
    if(((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="T") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="A")) | 
       ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="T") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="A")) |
       ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="G") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="C")) |
       ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="G") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="C")) |
       
       ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCC" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="T") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="GGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="A")) | 
       ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCG" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="T") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="CGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="A")) |
       ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCC" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="G") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="GGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="C")) |
       ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCG" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="G") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="CGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="C"))
    ){
      HNSC.selection.output$TCN_TKN[j] <- 1
    }else{
      HNSC.selection.output$TCN_TKN[j] <- 0
    }
    
  }else{
    if(((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[j]=="T") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="A")) | 
       ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[j]=="T") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="A")) |
       ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[j]=="G") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="C")) |
       ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[j]=="G") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="C"))|
       
       ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCC" & HNSC.selection.output$Alternative_Nucleotide[j]=="T") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[j]=="GGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="A")) | 
       ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCG" & HNSC.selection.output$Alternative_Nucleotide[j]=="T") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[j]=="CGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="A")) |
       ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCC" & HNSC.selection.output$Alternative_Nucleotide[j]=="G") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[j]=="GGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="C")) |
       ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCG" & HNSC.selection.output$Alternative_Nucleotide[j]=="G") | 
        (HNSC.selection.output$Nucleotide_trinuc_context[j]=="CGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="C"))){
      HNSC.selection.output$TCN_TKN[j] <- 1
    }else{
      HNSC.selection.output$TCN_TKN[j] <- 0
    }
  }
  
  
}


HNSC.selection.output$Name <- NA
for(i in 1:nrow(HNSC.selection.output)){
  HNSC.selection.output$Name[i] <- paste(HNSC.selection.output$Gene[i]," ",ifelse(!is.na(HNSC.selection.output$Amino_acid_reference[i]),paste(HNSC.selection.output$Amino_acid_reference[i],HNSC.selection.output$Amino_acid_position[i],HNSC.selection.output$Amino_acid_alternative[i]," ", HNSC.selection.output$Chromosome[i],"_",HNSC.selection.output$Nucleotide_chromosome_position[i],HNSC.selection.output$Alternative_Nucleotide[i],sep=""),paste(HNSC.selection.output$Reference_Nucleotide[i],HNSC.selection.output$Nucleotide_chromosome_position[i],HNSC.selection.output$Alternative_Nucleotide[i],"NCSNV")),sep="")
}

HNSC.selection.output$Name_short <- NA
for(i in 1:nrow(HNSC.selection.output)){
  HNSC.selection.output$Name_short[i] <- paste(HNSC.selection.output$Gene[i]," ",ifelse(!is.na(HNSC.selection.output$Amino_acid_reference[i]),paste(HNSC.selection.output$Amino_acid_reference[i],HNSC.selection.output$Amino_acid_position[i],HNSC.selection.output$Amino_acid_alternative[i],sep=""),paste(HNSC.selection.output$Reference_Nucleotide[i],HNSC.selection.output$Nucleotide_chromosome_position[i],HNSC.selection.output$Alternative_Nucleotide[i],"NCSNV")),sep="")
}



# HNSC.selection.output
save(HNSC.selection.output,file = "output_data/HNSC_selection_with_APOBEC.RData")

HNSC.selection.output.recur <- subset(HNSC.selection.output, Nucleotide_change_tally>1)
save(HNSC.selection.output.recur, file = "output_data/HNSC_selection_with_APOBEC_recur.RData")
```


```{r}
trinuc.w.HPV <- weights.df

trinuc.w.HPV$HPV <- NA

for(i in 1:nrow(trinuc.w.HPV)){
  trinuc.w.HPV$HPV[i] <- HNSC.selection.output$HPV_call[which(HNSC.selection.output$Unique_patient_identifier==rownames(weights.df)[i])[1]]
}

length(which(trinuc.w.HPV$HPV=="HPV+"))
# length(unique(HNSC.selection.output$Unique_patient_identifier[which(HNSC.selection.output$HPV_call=="HPV+" & !is.na(HNSC.selection.output$APOBEC_weight))]))

length(unique(HNSC.selection.output$Unique_patient_identifier[which(HNSC.selection.output$HPV_call=="HPV+" & HNSC.selection.output$APOBEC_weight > 0)]))
# extra.names <- unique(HNSC.selection.output$Unique_patient_identifier[which(HNSC.selection.output$HPV_call=="HPV+" & HNSC.selection.output$APOBEC_weight > 0)])

length(which((trinuc.w.HPV$`2` > 0 | trinuc.w.HPV$`13`>0) & trinuc.w.HPV$HPV=="HPV+"))

# trinuc.w.HPV[extra.names,]
# TCGA-BA-4077 different in both 
# which(rownames(trinuc.w.HPV)=="TCGA-BA-4077")

# virusscan.data[which(virusscan.data$patient_id=="TCGA-BA-4077"),] # should be postitive, why is it negative in trinuc.w.HPV?

# now they match
# 7/20

length(which((trinuc.w.HPV$`2` > 0 | trinuc.w.HPV$`13`>0) & trinuc.w.HPV$HPV=="HPV-"))/length(which(trinuc.w.HPV$HPV=="HPV-"))

length(which((trinuc.w.HPV$`2` > 0 | trinuc.w.HPV$`13`>0) & trinuc.w.HPV$HPV=="HPV+"))/length(which(trinuc.w.HPV$HPV=="HPV+"))

# mean weights
mean(trinuc.w.HPV$`2`[which(trinuc.w.HPV$HPV=="HPV+")])
mean(trinuc.w.HPV$`13`[which(trinuc.w.HPV$HPV=="HPV+")])

mean(trinuc.w.HPV$`2`[which(trinuc.w.HPV$HPV=="HPV-")])
mean(trinuc.w.HPV$`13`[which(trinuc.w.HPV$HPV=="HPV-")])

```


```{r prevalence prevalence plot, eval=T, include=FALSE}
# load("output_data/HNSC_selection_with_APOBEC_TCW_recur.RData")
# head(HNSC.selection.output.recur)

# load("~/Documents/Selection_analysis/HNSC/selection_output/HNSC_selection_output.RData")
#
# selection.all.muts <- as.tibble(selection.output$all_mutations) %>%
#   subset(freq>1)
#
# selection.all.muts$Name <- NA
# for(i in 1:nrow(selection.all.muts)){
#   selection.all.muts$Name[i] <- paste(selection.all.muts$Gene[i]," ",ifelse(!is.na(selection.all.muts$AA_Ref[i]),paste(selection.all.muts$AA_Ref[i],selection.all.muts$AA_Pos[i],selection.all.muts$AA_Change[i],sep=""),paste(selection.all.muts$Nuc_Ref[i],selection.all.muts$Nucleotide_position[i],selection.all.muts$Nuc_Change[i],"NCSNV")),sep="")
# }
# # colnames(selection.all.muts)
#
# HNSC.selection.output.recur$Name_short <- NA
# for(i in 1:nrow(HNSC.selection.output.recur)){
#   HNSC.selection.output.recur$Name_short[i] <- paste(HNSC.selection.output.recur$Gene[i]," ",ifelse(!is.na(HNSC.selection.output.recur$Amino_acid_reference[i]),paste(HNSC.selection.output.recur$Amino_acid_reference[i],HNSC.selection.output.recur$Amino_acid_position[i],HNSC.selection.output.recur$Amino_acid_alternative[i],sep=""),paste(HNSC.selection.output.recur$Reference_Nucleotide[i],HNSC.selection.output.recur$Nucleotide_chromosome_position[i],HNSC.selection.output.recur$Alternative_Nucleotide[i],"NCSNV")),sep="")
# }
# head(HNSC.selection.output.recur)

#TODO: should we then delete the mutations that are not recurrent within these smaller datasets?

recur.hpv.pos <- subset(HNSC.selection.output.recur, HPV_call=="HPV+")
recur.hpv.pos <- subset(recur.hpv.pos, Name_short %in% names(table(recur.hpv.pos$Name_short))[which(table(recur.hpv.pos$Name_short)>1)]) #just the recurrent mutations 
hpv.pos.names <- unique(recur.hpv.pos$Name_short)

recur.hpv.neg <- subset(HNSC.selection.output.recur, HPV_call=="HPV-")
recur.hpv.neg <- subset(recur.hpv.neg, Name_short %in% names(table(recur.hpv.neg$Name_short))[which(table(recur.hpv.neg$Name_short)>1)]) #just the recurrent mutations 
hpv.neg.names <- unique(recur.hpv.neg$Name_short)

intersect(hpv.pos.names,hpv.neg.names)
both.names <- union(hpv.neg.names,hpv.pos.names)

#make a dataframe with nrows == union of names, and columnes the number in each type of tumor and then the prevalence in each type

prevalence.df <- as.data.frame(matrix(data = NA, nrow = length(both.names), ncol=5))
colnames(prevalence.df) <- c("Name","tally_HPVpos","tally_HPVneg","prev_HPVpos","prev_HPVneg")

prevalence.df$Name <- both.names
prevalence.df$tally_HPVpos <- 0
prevalence.df$tally_HPVneg <- 0
for(i in 1:nrow(prevalence.df)){
  if(length(which(hpv.pos.names == prevalence.df$Name[i]))>0){
    prevalence.df$tally_HPVpos[i] <- length(which(recur.hpv.pos$Name_short == prevalence.df$Name[i]))
  }
  if(length(which(hpv.neg.names == prevalence.df$Name[i]))>0){
    prevalence.df$tally_HPVneg[i] <- length(which(recur.hpv.neg$Name_short == prevalence.df$Name[i]))
  }
}

prevalence.df$prev_HPVpos <- prevalence.df$tally_HPVpos/length(unique(HNSC.selection.output$Tumor_origin[which(HNSC.selection.output$HPV_call=="HPV+")]))
prevalence.df$prev_HPVneg <- prevalence.df$tally_HPVneg/length(unique(HNSC.selection.output$Tumor_origin[which(HNSC.selection.output$HPV_call=="HPV-")])) #TODO why only 411? did one tumor have no SNP?

library(ggrepel)
# prev_plot <- ggplot(data = prevalence.df) + geom_point(aes(y = prev_HPVneg, x = prev_HPVpos),alpha=0.5,size=5) + geom_text_repel(data = subset(prevalence.df, (prev_HPVneg>0.025 | prev_HPVpos>0.04) | (prev_HPVneg>0.0 & prev_HPVpos>0)),aes(y = prev_HPVneg, x = prev_HPVpos,label=Name),box.padding = 1.5,size=4,color="darkred",fontface="bold") + theme_bw() + labs(y=bquote("Prevalence in "~HPV^{"-"}~ "\n tumors"), x=bquote("Prevalence in "~HPV^{"+"}~ "tumors")) + theme(axis.text=element_text(size=12), axis.title=element_text(size=10,face="bold"),plot.margin = margin(r=.2,unit = "in")) + coord_equal() #+ geom_label(aes(x = prev_HPVneg, y = prev_HPVpos, label=Name))

# ggsave(filename = "Figures/prevalence_plot.png",plot = prev_plot,height = 3,width = 8,dpi = 300)

# recur.hpv.pos$prevalence <- recur.hpv.pos$Nucleotide_change_tally,
# prev_plot <- ggplot(data = prevalence.df) + geom_point(aes(x = prev_HPVneg, y = prev_HPVpos),alpha=0.5,size=5) +
#   geom_text_repel(data = subset(prevalence.df, (prev_HPVneg>0.025 | prev_HPVpos>0.04) | (prev_HPVneg>0.0 & prev_HPVpos>0)),aes(x = prev_HPVneg, y = prev_HPVpos,label=Name),box.padding = 2,size=4,color="darkred",fontface="bold") + theme_bw() + labs(x=bquote("Prevalence in "~HPV^{"-"}~ "\n tumors"), y=bquote("Prevalence in "~HPV^{"+"}~ "tumors")) + theme(axis.text=element_text(size=12), axis.title=element_text(size=10,face="bold"),plot.margin = margin(r=.2,unit = "in")) + coord_equal()
# 
# prev_plot
# 
# ggsave(filename = "Figures/prevalence_plot.png",plot = prev_plot,height = 6,width = 4,dpi = 300)
# ggplot(data = HNSC.selection.output.recur) + geom_point(
# HNSC.selection.output[which(HNSC.selection.output$HPV_call=="positive" & HNSC.selection.output$Gene=="TP53"),]

```

```{r prevalence and mutation rate plots}
# getting them to match up! 
prev_plot <- ggplot(data = prevalence.df) + 
  geom_point(aes(y = prev_HPVneg, x = prev_HPVpos),alpha=0.5,size=1.5) + 
  geom_text_repel(data = subset(prevalence.df, (prev_HPVneg>0.025 | prev_HPVpos>0.04) | (prev_HPVneg>0.0 & prev_HPVpos>0)),aes(y = prev_HPVneg, x = prev_HPVpos,label=Name),box.padding =0.6,size=1.8,color="darkred",fontface="bold") + 
  theme_bw() + 
  labs(y=bquote("Prevalence in "~HPV^{"-"}~ "\n tumors"), x=bquote("Prevalence in "~HPV^{"+"}~ "tumors")) + theme(axis.text=element_text(size=6), axis.title=element_text(size=5,face="bold"),plot.margin = margin(r=.1,t=.1,unit = "in")) + 
  coord_equal()   #+ geom_label(aes(x = prev_HPVneg, y = prev_HPVpos, label=Name))
prev_plot
ggsave(filename = "Figures/prevalence_plot.png",plot = prev_plot,height = 1.2,width = 3.25,dpi = 600)


mutation_rates_full_scatter <- ggplot(data = mutation_rates, aes(x=positive_mut_rates,y=negative_mut_rates)) +
  geom_point(alpha=0.2,col="black",size=0.5) + 
  geom_smooth(method='lm',color="red",size=.5) + 
  # geom_text_repel(data=subset(mutation_rates, positive_mut_rates > 5e-5 | negative_mut_rates > 3e-5 ), aes(label=gene),size=3) + 
  geom_point(data = subset(mutation_rates, gene %in% c("FBXW7","PIK3CA","MAPK1")),aes(x=positive_mut_rates,y=negative_mut_rates),size=1.5,col="blue") + 
  geom_text_repel(data = subset(mutation_rates, gene %in% c("FBXW7","PIK3CA","MAPK1")),aes(x=positive_mut_rates,y=negative_mut_rates,label=gene),size=1.8,col="blue",fontface="bold") +
  labs(y=bquote("Mutation rate in "~HPV^{"-"}~ "\n tumors"), x=bquote("Mutation rate in "~HPV^{"+"}~ "tumors")) + 
  coord_equal() + 
  theme_bw() +
  geom_abline(slope=1, intercept=0) + 
  scale_x_continuous(labels=fancy_scientific) + 
  scale_y_continuous(labels=fancy_scientific) + theme(axis.text=element_text(size=6),
                                                      axis.title=element_text(size=5,face="bold"),
                                                      plot.margin = margin(r=.1,t=.1,unit = "in"))  

mutation_rates_full_scatter
ggsave(plot = mutation_rates_full_scatter,filename = "Figures/mutation_rates_full_scatter.png",height = 1.2,width = 3.25,dpi=600)


# library(cowplot)

# plot_grid(prev_plot,mutation_rates_full_scatter,nrow = 2)



```



```{r gamma gamma plot, eval=T}


load("output_data/selection_from_cluster/HNSC_HPVpos/selection_output/HNSC_HPVpos_selection_output.RData")
HPV.pos.selectionoutput <- selection.output
HPV.pos.selection_minrecur <- subset(HPV.pos.selectionoutput$all_mutations, freq>1)


load("output_data/selection_from_cluster/HNSC_HPVneg/selection_output/HNSC_HPVneg_selection_output.RData")
HPV.neg.selectionoutput <- selection.output
HPV.neg.selection_minrecur <- subset(HPV.neg.selectionoutput$all_mutations, freq>1)


HPV.pos.selection_minrecur$Name_short <- NA
for(i in 1:nrow(HPV.pos.selection_minrecur)){
  HPV.pos.selection_minrecur$Name_short[i] <- paste(HPV.pos.selection_minrecur$Gene[i]," ",ifelse(!is.na(HPV.pos.selection_minrecur$AA_Ref[i]),paste(HPV.pos.selection_minrecur$AA_Ref[i],HPV.pos.selection_minrecur$AA_Pos[i],HPV.pos.selection_minrecur$AA_Change[i],sep=""),paste(HPV.pos.selection_minrecur$Nuc_Ref[i],HPV.pos.selection_minrecur$Nucleotide_position[i],HPV.pos.selection_minrecur$Nuc_Change[i],"NCSNV")),sep="")
}


HPV.neg.selection_minrecur$Name_short <- NA
for(i in 1:nrow(HPV.neg.selection_minrecur)){
  HPV.neg.selection_minrecur$Name_short[i] <- paste(HPV.neg.selection_minrecur$Gene[i]," ",ifelse(!is.na(HPV.neg.selection_minrecur$AA_Ref[i]),paste(HPV.neg.selection_minrecur$AA_Ref[i],HPV.neg.selection_minrecur$AA_Pos[i],HPV.neg.selection_minrecur$AA_Change[i],sep=""),paste(HPV.neg.selection_minrecur$Nuc_Ref[i],HPV.neg.selection_minrecur$Nucleotide_position[i],HPV.neg.selection_minrecur$Nuc_Change[i],"NCSNV")),sep="")
}


gamma.df <- cbind(prevalence.df,NA,NA)
colnames(gamma.df)[6:7] <- c("gamma_HPVpos","gamma_HPVneg")

for(i in 1:nrow(gamma.df)){
  if(length(which(HPV.pos.selection_minrecur$Name_short == gamma.df$Name[i]))>0){
    gamma.df$gamma_HPVpos[i] <- HPV.pos.selection_minrecur$gamma_epistasis[which(HPV.pos.selection_minrecur$Name_short == gamma.df$Name[i])][1]
  }
  if(length(which(HPV.neg.selection_minrecur$Name_short == gamma.df$Name[i]))>0){
    gamma.df$gamma_HPVneg[i] <- HPV.neg.selection_minrecur$gamma_epistasis[which(HPV.neg.selection_minrecur$Name_short == gamma.df$Name[i])][1]
  }
}

fancy_scientific <- function(l) {
  # turn in to character string in scientific notation
  l <- format(l, scientific = TRUE)
  l <- gsub("0e\\+00","0",l)
  # quote the part before the exponent to keep all the digits
  l <- gsub("^(.*)e", "'\\1'e", l)
  # turn the 'e+' into plotmath format
  l <- gsub("e", "%*%10^", l)
  # return this as an expression
  parse(text=l)
}



library(ggrepel)
# gamma_plot <- ggplot(data = gamma.df) + geom_point(aes(x = log10(gamma_HPVneg), y = log10(gamma_HPVpos)),alpha=0.5,size=5) + geom_text_repel(aes(x = log10(gamma_HPVneg), y = log10(gamma_HPVpos),label=Name),box.padding = 0.8,size=5,color="darkred",fontface="bold") + theme_bw() + labs(x="Log of selection intensity in HPV- tumors", y="Log of selection intensity in HPV+ tumors") + theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold")) #+ geom_label(aes(x = prev_HPVneg, y = prev_HPVpos, label=Name))


gamma_plot <- ggplot(data = gamma.df) + geom_point(aes(x = gamma_HPVneg, y = gamma_HPVpos),alpha=0.5,size=5) + geom_text_repel(aes(x = gamma_HPVneg, y = gamma_HPVpos,label=Name),box.padding = 0.8,size=5,color="darkred",fontface="bold") + theme_bw() + labs(x=bquote("Selection intensity in "~HPV^{"-"}~ "tumors"), y=bquote("Selection intensity in "~HPV^{"+"}~ "tumors")) + theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold"))  + scale_y_log10(labels = fancy_scientific) + scale_x_log10(labels = fancy_scientific) + coord_equal() #+ coord_equal(xlim = c(0,),ylim = c(0,))

gamma_plot

ggsave(filename = "Figures/gamma_gamma_plot.png",plot = gamma_plot,height = 4,width = 5)

```



# tornado plots 

```{r processing HPV neg for tornado plots}

### Figures for manuscript

# Selection plots with the same colors 
# setwd("~/Documents/Selection_analysis/HNSC/NCI/")
# source("~/GitHub/site_specific_selection_intensity/functions_script.R")

# load in the HPV neg, load in the HPV pos, find all unique genes and generate color palette 
# This will be sorted by selection, not frequency. 
# Neg 
load("output_data/selection_from_cluster/HNSC_HPVneg/selection_output/HNSC_HPVneg_selection_output.RData")

selection.subset.neg <- selection.output$all_mutations[which(selection.output$all_mutations$freq>1),]
selection.subset.neg <- selection.subset.neg[which(!is.na(selection.subset.neg$Gene)),]
selection.subset.neg <- selection.subset.neg[order(-selection.subset.neg$gamma_epistasis),]

if(nrow(selection.subset.neg)>25){
  selection.subset.neg.ordered <- selection.subset.neg[1:25,] #
}else{
  selection.subset.neg.ordered <- selection.subset.neg 
}
selection.subset.neg.ordered <- selection.subset.neg.ordered[order(selection.subset.neg.ordered$gamma_epistasis),]
selection.subset.neg.ordered$Name <- NA

for(i in 1:nrow(selection.subset.neg.ordered)){
  selection.subset.neg.ordered$Name[i] <- paste(selection.subset.neg.ordered$Gene[i]," ",ifelse(!is.na(selection.subset.neg.ordered$AA_Ref[i]),paste(selection.subset.neg.ordered$AA_Ref[i],selection.subset.neg.ordered$AA_Pos[i],selection.subset.neg.ordered$AA_Change[i],sep=""),"NCSNV"))
}



# selection.subset.neg.ordered$Name <- paste(selection.subset.neg.ordered$Gene," ",selection.subset.neg.ordered$AA_Ref,selection.subset.neg.ordered$AA_Pos,selection.subset.neg.ordered$AA_Change,sep="")

length(unique(selection.subset.neg.ordered$Name))

if(length(which(selection.subset.neg.ordered$Name=="TP53   NCSNV"))>0){
  selection.subset.neg.ordered$Name[which(selection.subset.neg.ordered$Name=="TP53   NCSNV")] <- paste("TP53    NCSNV",letters[length((which(selection.subset.neg.ordered$Name=="TP53   NCSNV"))):1],sep="")
}
selection.subset.neg.ordered$Name <- factor(selection.subset.neg.ordered$Name, levels=selection.subset.neg.ordered$Name)
# selection.subset.neg.ordered

```

```{r processing HPV pos for tornado plots}

# Pos

load("output_data/selection_from_cluster/HNSC_HPVpos/selection_output/HNSC_HPVpos_selection_output.RData")
selection.subset.pos <- selection.output$all_mutations[which(selection.output$all_mutations$freq>1),]
selection.subset.pos <- selection.subset.pos[which(!is.na(selection.subset.pos$Gene)),]
selection.subset.pos <- selection.subset.pos[order(-selection.subset.pos$gamma_epistasis),]

if(nrow(selection.subset.pos)>25){
  selection.subset.pos.ordered <- selection.subset.pos[1:25,] #
}else{
  selection.subset.pos.ordered <- selection.subset.pos 
}
selection.subset.pos.ordered <- selection.subset.pos.ordered[order(selection.subset.pos.ordered$gamma_epistasis),]
selection.subset.pos.ordered$Name <- NA

for(i in 1:nrow(selection.subset.pos.ordered)){
  selection.subset.pos.ordered$Name[i] <- paste(selection.subset.pos.ordered$Gene[i]," ",ifelse(!is.na(selection.subset.pos.ordered$AA_Ref[i]),paste(selection.subset.pos.ordered$AA_Ref[i],selection.subset.pos.ordered$AA_Pos[i],selection.subset.pos.ordered$AA_Change[i],sep=""),"NCSNV"))
}



# selection.subset.pos.ordered$Name <- paste(selection.subset.pos.ordered$Gene," ",selection.subset.pos.ordered$AA_Ref,selection.subset.pos.ordered$AA_Pos,selection.subset.pos.ordered$AA_Change,sep="")
selection.subset.pos.ordered$Name <- factor(selection.subset.pos.ordered$Name, levels=selection.subset.pos.ordered$Name)
# selection.subset.pos.ordered


```

```{r finding colors used for unique genes in combined build}

##Find colors used for unique genes in combined build. 
source("R/fancy_scientific_code.R")
selection.subset.combined.ordered <- rbind(selection.subset.neg.ordered,selection.subset.pos.ordered)

unique.genes <- unique(selection.subset.combined.ordered$Gene)
#Make a dataframe of just the unique genes
selection.subset.combined.ordered.unique <- as.data.frame(matrix(nrow=0,ncol=ncol(selection.subset.combined.ordered)))
colnames(selection.subset.combined.ordered.unique) <- colnames(selection.subset.combined.ordered)
for(i in 1:length(unique.genes)){
  selection.subset.combined.ordered.unique <- rbind(selection.subset.combined.ordered.unique,selection.subset.combined.ordered[which(selection.subset.combined.ordered$Gene==unique.genes[i])[1],])
}

#from http://stackoverflow.com/questions/18265941/two-horizontal-bar-charts-with-shared-axis-in-ggplot2-similar-to-population-pyr
library('grid')
library('gridExtra')


g.mid <- ggplot(selection.subset.combined.ordered.unique,aes(x=1,y=Name)) +
  geom_text(aes(label=Name),size=7) +
  # geom_segment(aes(x=0.94,xend=0.96,yend=Name)) +
  # geom_segment(aes(x=1.04,xend=1.065,yend=Name)) +
  ggtitle("") +
  ylab(NULL) + 
  scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065)) + ggtitle("HNSCC HPV+") +
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))
# g.mid

g1 <- ggplot(data=selection.subset.combined.ordered.unique,aes(x=Name,y=mu, fill=Gene)) +
  geom_bar(stat="identity") + ggtitle("Mutation rate") +
  theme(axis.title.x = element_blank(), 
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(), 
        plot.margin = unit(c(1,-1,1,10), "mm")) +
  theme(panel.background = element_blank()) +
  theme(panel.grid.major =element_line(color="lightgrey"),panel.grid.minor =element_line(color="lightgrey")) +
  theme(panel.grid.major.y = element_blank(),panel.grid.minor.y = element_blank()) +
  geom_text(aes(label=round(mu*1e5,2)), vjust=-0.5, hjust=0.5, position=position_dodge(width=0.9),size=4,angle=90) +
  geom_text(aes(label=freq,y=-2e-7), position=position_dodge(width=0.9),size=6,angle=0,color="black") +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(legend.position = 'none',axis.text.x = element_text(size=12)) +
  scale_y_reverse(labels=fancy_scientific) + coord_flip() 
# g1

g2 <- ggplot(data=selection.subset.combined.ordered.unique, aes(x=Name,y=gamma_epistasis,fill=Gene)) +
  geom_bar(stat="identity") + ggtitle("Selection intensity") +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(), 
        axis.text.y = element_blank(), axis.ticks.y = element_blank(),
        plot.margin = unit(c(1,2,1,-1), "mm")) +
  theme(panel.background = element_blank()) +
  theme(panel.grid.major =element_line(color="lightgrey"),panel.grid.minor =element_line(color="lightgrey")) +
  theme(panel.grid.major.y = element_blank(),panel.grid.minor.y = element_blank()) +
  theme(plot.title = element_text(hjust = 0.5),axis.text.x = element_text(size=12)) +
  coord_flip() + theme(legend.position = c(0.85, .5),legend.text = element_text(size=10)) + theme(legend.position="none")
# g2

gg1 <- ggplot_gtable(ggplot_build(g1))
gg2 <- ggplot_gtable(ggplot_build(g2))
gg.mid <- ggplot_gtable(ggplot_build(g.mid))

# grid.arrange(gg1,gg.mid,gg2,ncol=3,widths=c(4.25/10,2/10,3.5/10))


gg.combined <- arrangeGrob(gg1,gg.mid,gg2,ncol=3,widths=c(5/10,3.5/10,5/10))

ggplot_build(g1)$data
ggplot_build(g.mid)$data

color.data <- ggplot_build(g1)$data 

gene.colors <- selection.subset.combined.ordered.unique$Gene
names(gene.colors) <- color.data[[1]]$fill


color.df <- as.data.frame(matrix(data = NA, nrow=length(unique(selection.subset.combined.ordered.unique$Gene)),ncol=2))
colnames(color.df) <- c("gene","color")
color.df$gene <- unique(selection.subset.combined.ordered.unique$Gene)
for(i in 1:nrow(selection.subset.combined.ordered.unique)){
  if(is.na(color.df$color[which(color.df$gene==selection.subset.combined.ordered.unique$Gene[i])])){
    color.df$color[which(color.df$gene==selection.subset.combined.ordered.unique$Gene[i])] <- 
      color.data[[1]]$fill[i]
  }
}

# color.df

color.vec <- color.df$color
names(color.vec) <- color.df$gene
```

```{r hpv neg tornado plot}
### Now, for the separate plots
# HPV negative 

neg.colors <- color.vec[names(color.vec) %in% selection.subset.neg.ordered$Gene]



g.mid <- ggplot(selection.subset.neg.ordered,aes(x=1,y=Name)) +
  geom_text(aes(label=Name),size=7) +
  # geom_segment(aes(x=0.94,xend=0.96,yend=Name)) +
  # geom_segment(aes(x=1.04,xend=1.065,yend=Name)) +
  ggtitle("") +
  ylab(NULL) + 
  # scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065)) + ggtitle("HNSCC HPV-") +
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))
# g.mid

g1 <- ggplot(data=selection.subset.neg.ordered,aes(x=Name,y=mu, fill=Gene)) +
  geom_bar(stat="identity") + ggtitle("Mutation rate") +
  theme(axis.title.x = element_blank(), 
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(), 
        plot.margin = unit(c(1,-1,1,10), "mm")) +
  theme(panel.background = element_blank()) + scale_fill_manual("Legend", values = neg.colors) +
  theme(panel.grid.major =element_line(color="lightgrey"),panel.grid.minor =element_line(color="lightgrey")) +
  theme(panel.grid.major.y = element_blank(),panel.grid.minor.y = element_blank()) +
  # geom_text(aes(label=round(mu*1e5,2)), vjust=-0.5, hjust=0.5, position=position_dodge(width=0.9),size=4,angle=90) +
  geom_text(aes(label=freq,y=-1e-7), position=position_dodge(width=0.9),size=6,angle=0,color="black") +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(legend.position = 'none',axis.text.x = element_text(size=12)) +
  scale_y_reverse(labels=fancy_scientific) + coord_flip() + 
  theme(axis.text.x = element_text(size = 13))
# g1

g2 <- ggplot(data=selection.subset.neg.ordered, aes(x=Name,y=gamma_epistasis,fill=Gene)) +
  geom_bar(stat="identity") + ggtitle("Selection intensity") +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(), 
        axis.text.y = element_blank(), axis.ticks.y = element_blank(),
        plot.margin = unit(c(1,2,1,-1), "mm")) +
  theme(panel.background = element_blank()) + scale_fill_manual("Legend", values = neg.colors) +
  theme(panel.grid.major =element_line(color="lightgrey"),panel.grid.minor =element_line(color="lightgrey")) +
  theme(panel.grid.major.y = element_blank(),panel.grid.minor.y = element_blank()) +
  theme(plot.title = element_text(hjust = 0.5),axis.text.x = element_text(size=12)) +
  coord_flip() + theme(legend.position = c(0.85, .5),legend.text = element_text(size=10)) + theme(legend.position="none") + scale_y_continuous(labels=fancy_scientific) + theme(axis.text.x = element_text(size = 13))
# g2

gg1 <- ggplot_gtable(ggplot_build(g1))
gg2 <- ggplot_gtable(ggplot_build(g2))
gg.mid <- ggplot_gtable(ggplot_build(g.mid))

grid.arrange(gg1,gg.mid,gg2,ncol=3,widths=c(3/10,3.5/10,3/10))


gg.combined <- arrangeGrob(gg1,gg.mid,gg2,ncol=3,widths=c(3/10,2/10,3/10))
ggsave(gg.combined, filename = paste("Figures/same_color_gamma_epistasis_plot_","HNSC_HPV-.png",sep=""),units = "in",height=8,width = 12,dpi = 400)
```



```{r hpv pos tornado plot}

# HPV positive

pos.colors <- color.vec[names(color.vec) %in% selection.subset.pos.ordered$Gene]



g.mid <- ggplot(selection.subset.pos.ordered,aes(x=1,y=Name)) +
  geom_text(aes(label=Name),size=7) +
  # geom_segment(aes(x=0.94,xend=0.96,yend=Name)) +
  # geom_segment(aes(x=1.04,xend=1.065,yend=Name)) +
  ggtitle("") +
  ylab(NULL) + 
  # scale_x_continuous(expand=c(0,0),limits=c(0.94,1.065)) + ggtitle("HNSCC HPV-") +
  theme(axis.title=element_blank(),
        panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        panel.background=element_blank(),
        axis.text.x=element_text(color=NA),
        axis.ticks.x=element_line(color=NA),
        plot.margin = unit(c(1,-1,1,-1), "mm"))
# g.mid

g1 <- ggplot(data=selection.subset.pos.ordered,aes(x=Name,y=mu, fill=Gene)) +
  geom_bar(stat="identity") + ggtitle("Mutation rate") +
  theme(axis.title.x = element_blank(), 
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank(), 
        plot.margin = unit(c(1,-1,1,10), "mm")) +
  theme(panel.background = element_blank()) + scale_fill_manual("Legend", values = pos.colors) +
  theme(panel.grid.major =element_line(color="lightgrey"),panel.grid.minor =element_line(color="lightgrey")) +
  theme(panel.grid.major.y = element_blank(),panel.grid.minor.y = element_blank()) +
  # geom_text(aes(label=round(mu*1e5,2)), vjust=-0.5, hjust=0.5, position=position_dodge(width=0.9),size=4,angle=90) +
  geom_text(aes(label=freq,y=-2e-7), position=position_dodge(width=0.9),size=6,angle=0,color="black") +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(legend.position = 'none',axis.text.x = element_text(size=12)) +
  scale_y_reverse(labels=fancy_scientific) + coord_flip() + theme(axis.text.x = element_text(size = 12))
# g1

g2 <- ggplot(data=selection.subset.pos.ordered, aes(x=Name,y=gamma_epistasis,fill=Gene)) +
  geom_bar(stat="identity") + ggtitle("Selection intensity") +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(), 
        axis.text.y = element_blank(), axis.ticks.y = element_blank(),
        plot.margin = unit(c(1,10,1,-1), "mm")) +
  theme(panel.background = element_blank()) + scale_fill_manual("Legend", values = pos.colors) +
  theme(panel.grid.major =element_line(color="lightgrey"),panel.grid.minor =element_line(color="lightgrey")) +
  theme(panel.grid.major.y = element_blank(),panel.grid.minor.y = element_blank()) +
  theme(plot.title = element_text(hjust = 0.5),axis.text.x = element_text(size=12)) +
  coord_flip() + theme(legend.position = c(0.85, .5),legend.text = element_text(size=10)) + theme(legend.position="none") + scale_y_continuous(labels=fancy_scientific)
# g2

gg1 <- ggplot_gtable(ggplot_build(g1))
gg2 <- ggplot_gtable(ggplot_build(g2))
gg.mid <- ggplot_gtable(ggplot_build(g.mid))

grid.arrange(gg1,gg.mid,gg2,ncol=3,widths=c(3/10,3.5/10,3/10))


gg.combined <- arrangeGrob(gg1,gg.mid,gg2,ncol=3,widths=c(3/10,2/10,3/10))
ggsave(gg.combined, filename = paste("Figures/same_color_gamma_epistasis_plot_","HNSC_HPV+.png",sep=""),units = "in",height=8,width = 13,dpi = 400)


```


# heatmap and dendrogram of the signatures

```{r ggplot and dendrogram and heatmap data analysis}
# library(plyr)
library(reshape2)
# library(dplyr)
library(tidyverse)
library(ggplot2)
library(ggdendro)
library(gridExtra)
library(dendextend)

# useful: https://stackoverflow.com/questions/42047896/joining-a-dendrogram-and-a-heatmap
# and https://plot.ly/ggplot2/ggdendro-dendrograms/


load("output_data/trinuc_contexts_HNSC_MAF.RData")
weights.df <- trinuc.contexts$signature.weights[[1]]$weights
for(i in 2:length(trinuc.contexts$signature.weights)){
  weights.df <- rbind(weights.df,trinuc.contexts$signature.weights[[i]]$weights)
}


# prevalence of weights
weight.prevalence <- rep(NA,30)

for(i in 1:length(weight.prevalence)){
  weight.prevalence[i] <- length(which(weights.df[,i]>0))
}

message("Order of signatures in decreasing prevalence")
order(weight.prevalence,decreasing = T)

# Dendrogram data
# not scaling because all data should be on the same scale

colnames(weights.df) <- 1:30

dend <- as.dendrogram(hclust(dist(weights.df)))
dend_data <- dendro_data(dend)

dend_t <- as.dendrogram(hclust(dist(t(weights.df))))
dend_data_t <- dendro_data(dend_t)

# Setup segment data

segment_data_tumors <- with(
  segment(dend_data), 
  data.frame(x = y, y = x, xend = yend, yend = xend))

segment_data_signatures <- with(
  segment(dend_data_t), 
  data.frame(x = y, y = x, xend = yend, yend = xend))

# position data
tumor_pos_table <- with(
  dend_data$labels,
  data.frame(ycenter = x, tumor=as.character(label), height=1))

signature_pos_table <- with(
  dend_data_t$labels,
  data.frame(xcenter = x, signature=as.character(label), width=1))

tumor_names <- rownames(weights.df)
signature_names <- colnames(weights.df)


# library(feather)
# apobec_hnscc_df <- read_feather("input_data/nfkb.feather")

# head(apobec_hnscc_df)


# tables to position x and y 
# detach("package:plyr", unload=TRUE) 
# tumor_sample_pos_table <- data.frame(sample = tumor_names) %>%
#     mutate(x_center = (1:n()), 
#            width = 1)
# 
# signature_sample_pos_table <- data.frame(sample = signature_names) %>%
#     mutate(x_center = (1:n()), 
#            width = 1)


# heatmap_data_test <- weights.df %>%
#   add_column(id = rownames(weights.df)) %>%
#   reshape2::melt(value.name = "Signature weight")

heatmap_data <- weights.df %>%
  add_column(id = rownames(weights.df)) %>%
  reshape2::melt(value.name = "Signature weight") %>%
  rename(tumor = id, signature = variable) %>%
  left_join(signature_pos_table) %>%
  left_join(tumor_pos_table)

# Limits for the vertical axes
tumor_axis_limits <- with(
  tumor_pos_table, 
  c(min(ycenter - 0.5 * height), max(ycenter + 0.5 * height))
) + 
  0.1 * c(-1, 1) # extra spacing: 0.1
signature_axis_limits <- with(
  signature_pos_table, 
  c(min(xcenter - 0.5 * width), max(xcenter + 0.5 * width))
) + 
  0.1 * c(-1, 1) # extra spacing: 0.1

# ADD HPV status
segment_data_tumors$color <- "black"
# length(which(segment_data_tumors$xend==0));nrow(tumor_pos_table)

# segment_data_tumors$y[which(segment_data_tumors$xend==0)]
segment_data_tumors$tumor <- NA
segment_data_tumors$tumor[which(segment_data_tumors$xend==0)] <- as.character(tumor_pos_table$tumor)
segment_data_tumors$HPV_status <- NA



for(i in 1:nrow(segment_data_tumors[which(segment_data_tumors$xend==0),])){
  segment_data_tumors$HPV_status[which(segment_data_tumors$xend==0)][i] <- HNSC.selection.output$HPV_call[which(HNSC.selection.output$Unique_patient_identifier==segment_data_tumors$tumor[which(segment_data_tumors$xend==0)][i])[1]]
}

# which(is.na(segment_data_tumors$color) & !is.na(segment_data_tumors$tumor)) #all accounted for

head(segment_data_tumors,10)

for(i in 1:nrow(segment_data_tumors)){
  if(!is.na(segment_data_tumors$HPV_status[i])){
    
    if(segment_data_tumors$HPV_status[i]=="ambiguous"){
      segment_data_tumors$color[i] <- "gray" 
    }
    if(segment_data_tumors$HPV_status[i]=="HPV+"){
      segment_data_tumors$color[i] <- "blue" 
    }
    if(segment_data_tumors$HPV_status[i]=="HPV-"){
      segment_data_tumors$color[i] <- "red" 
    }
  }
  
}


```

```{r generate heatmap}
# Heatmap plot
plt_hmap <- ggplot(heatmap_data, 
                   aes(x = xcenter, y = ycenter, fill = `Signature weight`, 
                       height = height, width = width)) + 
  geom_tile() +
  scale_fill_gradient2("Signature weight", high = "darkblue", low = "white") +
  scale_x_continuous(breaks = signature_pos_table$xcenter, 
                     labels = signature_pos_table$signature, 
                     expand = c(0, 0),position = "top") + 
  # For the y axis, alternatively set the labels as: gene_position_table$gene
  scale_y_continuous(breaks = tumor_pos_table[, "ycenter"], 
                     labels = rep("", nrow(tumor_pos_table)),
                     limits = tumor_axis_limits, 
                     expand = c(0, 0)) + 
  labs(x = "Signature", y = "") +
  theme_bw() +
  theme(axis.text.x.top = element_text(size = rel(1.4), vjust=0.5,hjust=0.5 ,  angle = 90),axis.title.y = element_blank(),
        # margin: top, right, bottom, and left
        plot.margin = unit(c(-.051, -.02, -.051, -.051), "cm"),
        # plot.margin = unit(c(-1, 0.2, 0.2, 2), "cm"),
        panel.grid.minor = element_blank(),legend.position = "none",axis.title = element_blank()) 

# Dendrogram plot
plt_dendr <- ggplot(segment_data_tumors) + 
  geom_segment(aes(x = x, y = y, xend = xend, yend = yend,color=color),size=.2) + 
  scale_x_reverse(expand = c(0, 0.01)) + 
  scale_y_continuous(breaks = tumor_pos_table$y_center, 
                     labels = tumor_pos_table$tumor, 
                     limits = tumor_axis_limits, 
                     expand = c(0, 0)) + scale_colour_identity() +
  labs(x = "", y = "Tumors", colour = "", size = "") + 
  theme_bw() +
  theme(panel.grid.minor = element_blank(),axis.title.x  = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.title = element_blank(),panel.border = element_blank(),panel.grid.major = element_blank(),plot.margin = unit(c(-.1, -.1, -.1, -.1), "cm"))

plt_dendr_signature <- ggplot(segment_data_signatures) + 
  geom_segment(aes(x = xend, y = yend, xend = x, yend = y)) + 
  # scale_x_reverse(expand = c(0, 0.5)) + 
  scale_y_continuous(breaks = signature_pos_table$y_center, 
                     labels = signature_pos_table$signature, 
                     limits = signature_axis_limits, 
                     expand = c(0, 0),position="top") +
  labs(x = "Distance", y = "Signatures", colour = "", size = "") +
  theme_bw() + 
  theme(panel.grid.minor = element_blank(),axis.text = element_blank(),axis.ticks.y = element_blank(),axis.title = element_blank(),panel.border = element_blank(),panel.grid.major = element_blank(),plot.margin = unit(c(-.1, -.1, -.1, -.1), "cm")) + coord_flip()

library(cowplot)
heatmap_and_dendro <- plot_grid(NULL,plt_dendr_signature,plt_dendr, plt_hmap, align = 'hv', rel_widths = c(0.7, 1), rel_heights = c(.4,2))

heatmap_and_dendro

ggsave(filename = "Figures/heatmap_and_dendro.png",plot = heatmap_and_dendro,height = 5,width = 7)
ggsave(filename = "Figures/heatmap_and_dendro.pdf",plot = heatmap_and_dendro,height = 9,width = 6)
# 
# 
# plt_dendr_axis <- axis_canvas(plt_hmap,axis = 'y') + ggplot(segment_data_tumors) + 
#     geom_segment(aes(x = x, y = y, xend = xend, yend = yend)) + 
#     scale_x_reverse(expand = c(0, 0.01)) + 
#     scale_y_continuous(breaks = tumor_pos_table$y_center, 
#                        labels = tumor_pos_table$gene, 
#                        limits = tumor_axis_limits, 
#                        expand = c(0, 0)) + 
#     labs(x = "Distance", y = "", colour = "", size = "") +
#     theme_bw() + 
#     theme(panel.grid.minor = element_blank(),axis.title.x  = element_blank(),plot.margin = unit(c(0, 0.0, 0.0, -0.7), "cm"))

```




# HPV vs SNP count  

```{r HPV status vs SNV count}
library(ggplot2)
load("output_data/HNSC_selection_with_APOBEC.RData")
# which(HNSC.selection.output$HPV_call == "ambiguous")
HNSC.selection.HPVknown <- HNSC.selection.output[which(HNSC.selection.output$HPV_call != "ambiguous"),] #also removes NA
# which(is.na(HNSC.selection.HPVknown$HPV_call))
# nrow(HNSC.selection.HPVknown) == nrow(HNSC.selection.output)
# HNSC.selection.HPVknown$`APOBEC signal`

# unique(HNSC.selection.HPVknown$Unique_patient_identifier)

SNV.APOBEC.df <- as.data.frame(matrix(data = NA, nrow=length(unique(HNSC.selection.HPVknown$Unique_patient_identifier)),ncol=5))
colnames(SNV.APOBEC.df) <- c("Tumor","SNV_count","APOBEC","APOBEC_weight","HPV_status")

SNV.APOBEC.df$Tumor <- unique(HNSC.selection.HPVknown$Unique_patient_identifier)

for(i in 1:nrow(SNV.APOBEC.df)){
  SNV.APOBEC.df$SNV_count[i] <- length(which(HNSC.selection.HPVknown$Unique_patient_identifier==SNV.APOBEC.df$Tumor[i]))
  if(!is.na(HNSC.selection.HPVknown$APOBEC_weight[which(HNSC.selection.HPVknown$Unique_patient_identifier==SNV.APOBEC.df$Tumor[i])[1]])){
    SNV.APOBEC.df$APOBEC[i] <- if(HNSC.selection.HPVknown$APOBEC_weight[which(HNSC.selection.HPVknown$Unique_patient_identifier==SNV.APOBEC.df$Tumor[i])[1]]>0){"Yes"
    }else{if(HNSC.selection.HPVknown$APOBEC_weight[which(HNSC.selection.HPVknown$Unique_patient_identifier==SNV.APOBEC.df$Tumor[i])[1]]==0){"No"}}}
  SNV.APOBEC.df$HPV_status[i] <- HNSC.selection.HPVknown$HPV_call[which(HNSC.selection.HPVknown$Unique_patient_identifier==SNV.APOBEC.df$Tumor[i])[1]]
  SNV.APOBEC.df$APOBEC_weight[i] <- HNSC.selection.HPVknown$APOBEC_weight[which(HNSC.selection.HPVknown$Unique_patient_identifier==SNV.APOBEC.df$Tumor[i])[1]]
}

# View(SNV.APOBEC.df)

write.table(x = SNV.APOBEC.df,file = "output_data/SNV_APOBEC_HPV_status.txt",sep="\t",quote = F,row.names = F)

source("R/fancy_scientific_code.R")

HPV.vs.APOBEC <- ggplot(data = SNV.APOBEC.df) + geom_boxplot(aes(y=SNV_count,x=APOBEC),color="dark red") + geom_jitter(aes(y=SNV_count,x=APOBEC),width= 0.2,alpha=0.5) + facet_grid(.~HPV_status) + theme_bw() + scale_y_log10(labels=fancy_scientific) + labs(x="APOBEC signature detected",y="SNV count")


ggsave(filename = "Figures/HPV_status_and_APOBEC_vs_SNV.png",plot = HPV.vs.APOBEC)
#TODO: make sure the NAs are being removed 
wilcox.test(data=subset(SNV.APOBEC.df,HPV_status=="HPV-"),SNV_count~APOBEC,conf.int=T) 

wilcox.test(data=subset(SNV.APOBEC.df,HPV_status=="HPV+"),SNV_count~APOBEC, conf.int = T,alternative="two.sided") 

hpv.pos.snv.apobec <- subset(SNV.APOBEC.df,HPV_status=="HPV+")
hpv.neg.snv.apobec <- subset(SNV.APOBEC.df,HPV_status=="HPV-")

mean(hpv.pos.snv.apobec$SNV_count[which(hpv.pos.snv.apobec$APOBEC=="Yes")])-mean(hpv.pos.snv.apobec$SNV_count[which(hpv.pos.snv.apobec$APOBEC=="No")])

median(hpv.pos.snv.apobec$SNV_count[which(hpv.pos.snv.apobec$APOBEC=="Yes")])-median(hpv.pos.snv.apobec$SNV_count[which(hpv.pos.snv.apobec$APOBEC=="No")])

# SNV.APOBEC.df[which.min(SNV.APOBEC.df$SNV_count),]
# ggplot(data = SNV.APOBEC.df) + geom_point(aes(x=APOBEC_weight, y=APOBEC)) 

mean(hpv.neg.snv.apobec$SNV_count[which(hpv.neg.snv.apobec$APOBEC=="Yes")])-mean(hpv.neg.snv.apobec$SNV_count[which(hpv.neg.snv.apobec$APOBEC=="No")])

median(hpv.neg.snv.apobec$SNV_count[which(hpv.neg.snv.apobec$APOBEC=="Yes")])-median(hpv.neg.snv.apobec$SNV_count[which(hpv.neg.snv.apobec$APOBEC=="No")])


summary(hpv.pos.snv.apobec$SNV_count[which(hpv.pos.snv.apobec$APOBEC=="Yes")])
summary(hpv.pos.snv.apobec$SNV_count[which(hpv.pos.snv.apobec$APOBEC=="No")])

summary(hpv.neg.snv.apobec$SNV_count[which(hpv.neg.snv.apobec$APOBEC=="Yes")])
summary(hpv.neg.snv.apobec$SNV_count[which(hpv.neg.snv.apobec$APOBEC=="No")])
#plot for the manuscript


HPV.vs.APOBEC_ms <- ggplot(data = subset(SNV.APOBEC.df, !is.na(SNV.APOBEC.df$APOBEC))) + geom_boxplot(aes(y=SNV_count,x=APOBEC),color="dark red") + geom_jitter(aes(y=SNV_count,x=APOBEC),width= 0.2,alpha=0.5) + facet_grid(.~HPV_status) + theme_bw() + scale_y_log10(labels=fancy_scientific) + labs(x="APOBEC signature detected",y="SNV count")  + theme(strip.text.x = element_text(size=18)) +theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold"))

HPV.vs.APOBEC_ms

ggsave(filename = "Figures/SNV_count_vs_HPV_and_APOBEC.png",plot = HPV.vs.APOBEC_ms)






```


# Logistic regression of APOBEC and HPV

```{r logistic regression HPV APOBEC}


load("output_data/HNSC_selection_with_APOBEC.RData")

SNV.APOBEC.df <- read.table("output_data/SNV_APOBEC_HPV_status.txt",header = T)

SNV.APOBEC.df.knownAPOBEC <- SNV.APOBEC.df[which(!is.na(SNV.APOBEC.df$APOBEC)),] 
# length(which(SNV.APOBEC.df.knownAPOBEC$HPV_status=="HPV+" & SNV.APOBEC.df.knownAPOBEC$APOBEC_weight==0))
SNV.APOBEC.df.knownAPOBEC$HPV_status_value <- ifelse(SNV.APOBEC.df.knownAPOBEC$HPV_status=="HPV+",1,0)

HPV_status_vs_APOBEC_weight <- ggplot() + geom_point(data = SNV.APOBEC.df.knownAPOBEC,aes(y=HPV_status_value,x=APOBEC_weight),alpha=0.2,size=4) + stat_smooth(data = SNV.APOBEC.df.knownAPOBEC,aes(y=HPV_status_value,x=APOBEC_weight),method="glm", method.args=list(family="binomial"), se=T) + labs(y="Probability HPV positive",x="APOBEC signature weight") + theme_bw() 

# library(ggplot2)
HPV_status_vs_APOBEC_weight
ggsave(plot = HPV_status_vs_APOBEC_weight, filename ="Figures/HPV_status_APOBEC_weight.png",width = 3.25,height = 2,dpi = 300)

log_regression <- glm(HPV_status_value~APOBEC_weight,data = SNV.APOBEC.df.knownAPOBEC,family = binomial)

summary(log_regression)
# nrow(SNV.APOBEC.df.knownAPOBEC)
# which(is.na(SNV.APOBEC.df.knownAPOBEC$APOBEC_weight))
# which(is.na(SNV.APOBEC.df.knownAPOBEC$APOBEC))

# length(unique(SNV.APOBEC.df.knownAPOBEC$Tumor))==nrow(SNV.APOBEC.df.knownAPOBEC)
message("HPV+ tumors with APOBEC signal")
length(which(SNV.APOBEC.df.knownAPOBEC$APOBEC=="Yes" & SNV.APOBEC.df.knownAPOBEC$HPV_status=="HPV+"))/length(which(SNV.APOBEC.df.knownAPOBEC$HPV_status=="HPV+"))


message("HPV- tumors with APOBEC signal")
length(which(SNV.APOBEC.df.knownAPOBEC$APOBEC=="Yes" & SNV.APOBEC.df.knownAPOBEC$HPV_status=="HPV-"))/length(which(SNV.APOBEC.df.knownAPOBEC$HPV_status=="HPV-"))

```

```{r}
# Among samples with an APOBEC signal, total mutation load was... 

# SNV.APOBEC.yes <- subset(SNV.APOBEC.df,APOBEC=="Yes")

wilcox.test(data = subset(SNV.APOBEC.df,APOBEC=="Yes"),SNV_count~HPV_status,conf.int=T)
wilcox.test(data = subset(SNV.APOBEC.df,APOBEC=="No"),SNV_count~HPV_status,conf.int=T)

# SNV.APOBEC.df$APOBEC <- factor(SNV.APOBEC.df$APOBEC,levels = c("No APOBEC Signature","APOBEC Signature","NA"))
SNV.APOBEC.df.comparingAPOBEC <- SNV.APOBEC.df

SNV.APOBEC.df.comparingAPOBEC$APOBEC <- as.character(SNV.APOBEC.df.comparingAPOBEC$APOBEC)
SNV.APOBEC.df.comparingAPOBEC$APOBEC[which(SNV.APOBEC.df.comparingAPOBEC$APOBEC=="Yes")] <- "APOBEC signature detected"
SNV.APOBEC.df.comparingAPOBEC$APOBEC[which(SNV.APOBEC.df.comparingAPOBEC$APOBEC=="No")] <- "No APOBEC signature detected"

SNV_count_comparing_APOBEC_plot <- ggplot(data = subset(SNV.APOBEC.df.comparingAPOBEC, !is.na(SNV.APOBEC.df.comparingAPOBEC$APOBEC))) + geom_boxplot(aes(y=SNV_count,x=HPV_status),color="dark red") + geom_jitter(aes(y=SNV_count,x=HPV_status),width= 0.2,alpha=0.5) + facet_grid(.~APOBEC) + theme_bw() + scale_y_log10(labels=fancy_scientific) + labs(x="HPV status",y="SNV count")  + theme(strip.text.x = element_text(size=15)) +theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold"))

SNV_count_comparing_APOBEC_plot

ggsave(filename = "Figures/SNV_count_comparing_APOBEC.png",plot = SNV_count_comparing_APOBEC_plot)

```

```{r combining SNV HPV APOBEC plots}
library(cowplot)


HPV.vs.APOBEC_ms_forcombine <- ggplot(data = subset(SNV.APOBEC.df, !is.na(SNV.APOBEC.df$APOBEC))) + geom_boxplot(aes(y=SNV_count,x=APOBEC),color="dark red") + geom_jitter(aes(y=SNV_count,x=APOBEC),width= 0.2,alpha=0.5) + facet_grid(.~HPV_status) + theme_bw() + scale_y_log10(labels=fancy_scientific) + labs(x="APOBEC signature detected",y="SNV count")  + theme(strip.text.x = element_text(size=18)) +theme(axis.text=element_text(size=18), axis.title=element_text(size=20,face="bold"))
# plot_grid(
SNV_count_comparing_APOBEC_plot_forcombine <- ggplot(data = subset(SNV.APOBEC.df.comparingAPOBEC, !is.na(SNV.APOBEC.df.comparingAPOBEC$APOBEC))) + geom_boxplot(aes(y=SNV_count,x=HPV_status),color="dark red") + geom_jitter(aes(y=SNV_count,x=HPV_status),width= 0.2,alpha=0.5) + facet_grid(.~APOBEC) + theme_bw() + scale_y_log10(labels=fancy_scientific) + labs(x="HPV status",y="SNV count")  + theme(strip.text.x = element_text(size=15)) +theme(axis.text=element_text(size=18), axis.title=element_text(size=20,face="bold"),axis.title.y = element_blank())

combined_HPV_APOBEC_plot <- plot_grid(HPV.vs.APOBEC_ms_forcombine,SNV_count_comparing_APOBEC_plot_forcombine,align='h',ncol = 2,nrow = 1,labels = c("A","B"),label_size = 25)

combined_HPV_APOBEC_plot

ggsave(plot = combined_HPV_APOBEC_plot,filename = "Figures/combined_APOBEC_vs_HPV.png",height = 7,width = 14)

```




<!-- ```{r testing logistic regression} -->

<!-- test.df <- data.frame(x=1:100,y=c(rep(1,45),rep(0,5),rep(1,5),rep(0,45))) -->
<!-- ggplot(data = test.df,aes(x=x,y=y))+ geom_point(alpha=0.5) + stat_smooth(method="glm", method.args=list(family="binomial"), se=T) -->


<!-- ``` -->




```{r TCW to TKW vs APOBEC and HPV status }

SNV.APOBEC.df$`Proportion TCW to TKW` <- NA

for(i in 1:nrow(SNV.APOBEC.df)){
  SNV.APOBEC.df$`Proportion TCW to TKW`[i] <- length(which(HNSC.selection.HPVknown$TCW_TKW[which(HNSC.selection.HPVknown$Unique_patient_identifier==SNV.APOBEC.df$Tumor[i])]==1))/(length(which(HNSC.selection.HPVknown$TCW_TKW[which(HNSC.selection.HPVknown$Unique_patient_identifier==SNV.APOBEC.df$Tumor[i])]==1))+length(which(HNSC.selection.HPVknown$TCW_TKW[which(HNSC.selection.HPVknown$Unique_patient_identifier==SNV.APOBEC.df$Tumor[i])]==0)))
}

HPV.vs.APOBEC_proportion <- ggplot(data = SNV.APOBEC.df) + geom_boxplot(aes(y=`Proportion TCW to TKW`,x=APOBEC),color="dark red") + geom_jitter(aes(y=`Proportion TCW to TKW`,x=APOBEC),width= 0.2,alpha=0.5) + facet_grid(.~HPV_status) + theme_bw() + labs(x="APOBEC signature detected")

ggsave(filename = "Figures/HPV_status_and_APOBEC_vs_proportionTCW.png",plot = HPV.vs.APOBEC_proportion)


HPV.vs.APOBEC_proportion_ms <- ggplot(data = subset(SNV.APOBEC.df,!is.na(SNV.APOBEC.df$APOBEC))) + geom_boxplot(aes(y=`Proportion TCW to TKW`,x=APOBEC),color="dark red") + geom_jitter(aes(y=`Proportion TCW to TKW`,x=APOBEC),width= 0.2,alpha=0.5) + facet_grid(.~HPV_status) + theme_bw() + labs(x="APOBEC signature detected")

wilcox.test(data=SNV.APOBEC.df[which(SNV.APOBEC.df$APOBEC=="Yes"),],`Proportion TCW to TKW`~HPV_status,conf.int=T)
wilcox.test(data=SNV.APOBEC.df[which(SNV.APOBEC.df$APOBEC=="No"),],`Proportion TCW to TKW`~HPV_status,conf.int=T)


ggsave(filename = "Figures/HPV_status_and_APOBEC_vs_proportionTCW_forMS.png",plot = HPV.vs.APOBEC_proportion_ms)

```


```{r fisher exact test on APOBEC vs HPV}

# nrow(SNV.APOBEC.df)
# unique(SNV.APOBEC.df$APOBEC)
# length(which(SNV.APOBEC.df$APOBEC=="Yes" & SNV.APOBEC.df$HPV_status=="HPV+"))

# probably.wrong.apo.pos.hpv.pos <- as.character(SNV.APOBEC.df$Tumor[which(SNV.APOBEC.df$APOBEC=="Yes" & SNV.APOBEC.df$HPV_status=="HPV+")])



SNV.APOBEC.tib <- as_tibble(SNV.APOBEC.df) %>%
  group_by(APOBEC,HPV_status) %>%
  summarize(n())

# SNV.HPVplus.total <- SNV.APOBEC.tib$`n()`[which(SNV.APOBEC.tib$HPV_status=="HPV+")]

SNV.APOBEC.tib

# 7/(37+7)
# 37/(37+7)
# 
# 256/(256+124)
# 124/(256+124)

# fisher.test(matrix(data = c(2,2,250,250),nrow=2))
fisher.test(matrix(data = SNV.APOBEC.tib$`n()`[1:4],nrow=2))

```


##  Mutations that have the highest prevalence and selection intensity in HNSCC


```{r }
load("output_data/HNSC_selection_with_APOBEC_recur.RData")
just.ALL <- HNSC.selection.output.recur
ALL.names <- unique(just.ALL$Name)

ALL.prev.df <- as.data.frame(matrix(data = NA, nrow=length(ALL.names),ncol=5))
colnames(ALL.prev.df) <- c("Mutation","Frequency", "Trinuc context", "Alternative Nucleotide","Selection intensity")

ALL.prev.df$Mutation <- ALL.names

for(i in 1:nrow(ALL.prev.df)){
  ALL.prev.df$Frequency[i] <- length(which(just.ALL$Name == ALL.prev.df$Mutation[i]))
  ALL.prev.df$`Trinuc context`[i] <- just.ALL$Nucleotide_trinuc_context[which(just.ALL$Name == ALL.prev.df$Mutation[i])[1]]
  ALL.prev.df$`Alternative Nucleotide`[i] <- just.ALL$Alternative_Nucleotide[which(just.ALL$Name == ALL.prev.df$Mutation[i])[1]]
  ALL.prev.df$`Selection intensity`[i] <- round(just.ALL$Gamma_epistasis[which(just.ALL$Name == ALL.prev.df$Mutation[i])[1]],2)
}

ALL.prev.df <- ALL.prev.df[order(ALL.prev.df$Frequency,decreasing = T),]
```


```{r all mutations determining mutation weight, include = F}
library(deconstructSigs)

# load("HNSC_all_MAF.RData")
# load("HNSC_MAF.RData")


# load("trinuc_contexts_HPV_all.RData")
# length(unique(HNSC.HPVall.MAF$Unique_patient_identifier))
# length(trinuc.contexts[[2]])
load("output_data/trinuc_contexts_HNSC_MAF.RData")
tumor.trinuc.names <- NULL
for(i in 1:length(trinuc.contexts[[2]])){
  tumor.trinuc.names[i] <- rownames(trinuc.contexts[[2]][[i]]$weights)
}



# signatures.cosmic

# trinuc.contexts[[2]][[1]]$product
# as.matrix(trinuc.contexts[[2]][[1]]$weights) %*% as.matrix(signatures.cosmic) == as.matrix(trinuc.contexts[[2]][[1]]$product)


# head(just.TCW)
first.12 <- function(string_to_12){
  return(paste(unlist(strsplit(string_to_12,split = ""))[1:12],collapse = ""))
}

signature.breakdown.ALL <- as.data.frame(matrix(data = NA, nrow=nrow(just.ALL), ncol= 7))
colnames(signature.breakdown.ALL) <- c("Name","Tumor","Trinuc_context","Alternative_Nuc","deconstructSigs_context","APOBEC sum","NonAPOBEC sum")

signature.breakdown.ALL$Name <- just.ALL$Name
signature.breakdown.ALL$Tumor <- unlist(lapply(just.ALL$Tumor_origin,first.12))

for(i in 1:nrow(just.ALL)){
  if(is.na(just.ALL$Nucleotide_trinuc_context[i])){
    matches <- which(just.ALL$Nucleotide_chromosome_position == just.ALL$Nucleotide_chromosome_position[i] &
                       just.ALL$Alternative_Nucleotide == just.ALL$Alternative_Nucleotide[i])
    
    new.i <- matches[which(!is.na(just.ALL$Nucleotide_trinuc_context[matches]))]
    signature.breakdown.ALL$Trinuc_context[i] <- just.ALL$Nucleotide_trinuc_context[new.i]
  }else{
    signature.breakdown.ALL$Trinuc_context[i] <- just.ALL$Nucleotide_trinuc_context[i]
  }
}

signature.breakdown.ALL$Alternative_Nuc <- just.ALL$Alternative_Nucleotide

for(i in 1:nrow(signature.breakdown.ALL)){
  signature.breakdown.ALL$deconstructSigs_context[i] <- paste(
    strsplit(signature.breakdown.ALL$Trinuc_context[i],split = "")[[1]][1],
    "[",
    strsplit(signature.breakdown.ALL$Trinuc_context[i],split = "")[[1]][2],
    ">",
    signature.breakdown.ALL$Alternative_Nuc[i],
    "]",
    strsplit(signature.breakdown.ALL$Trinuc_context[i],split = "")[[1]][3],sep="")
  
}
# 
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="T[G>A]A")] <- "T[C>T]A"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="A[G>A]A")] <- "T[C>T]T"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="T[G>C]A")] <- "T[C>G]A"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="A[G>C]A")] <- "T[C>G]T"
# 
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="C[G>C]A")] <- "T[C>G]G"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="C[G>A]A")] <- "T[C>T]G"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="G[G>A]A")] <- "T[C>T]C"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="G[G>C]A")] <- "T[C>G]C"
# 
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="C[G>A]C")] <- "G[C>T]G"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="C[G>A]T")] <- "A[C>T]G"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="T[A>G]G")] <- "C[T>C]A"

flip.function <- function(nucleotide){
  if(nucleotide=="A"){
    return("T")
  }
  if(nucleotide=="T"){
    return("A")
  }
  if(nucleotide=="C"){
    return("G")
  }
  if(nucleotide=="G"){
    return("C")
  }
  if(nucleotide=="N"){
    return("N")
  }
}

unique.sigs <- unique(signature.breakdown.ALL$deconstructSigs_context)
unique.sigs.splt <- strsplit(x = unique.sigs,split = "")
for(i in 1:length(unique.sigs.splt)){
  if(unique.sigs.splt[[i]][3] %in% c("G","A")){
    this.new.str <- unique.sigs.splt[[i]]
    this.new.str[7] <- flip.function(unique.sigs.splt[[i]][1])
    this.new.str[3] <- flip.function(unique.sigs.splt[[i]][3])
    this.new.str[5] <- flip.function(unique.sigs.splt[[i]][5])
    this.new.str[1] <- flip.function(unique.sigs.splt[[i]][7])
    signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context==unique.sigs[i])] <- paste(this.new.str,collapse = "")
  }
}

# unique(signature.breakdown.ALL$deconstructSigs_context)

library(deconstructSigs)

for(i in 1:nrow(signature.breakdown.ALL)){
  if(length(which(tumor.trinuc.names==signature.breakdown.ALL$Tumor[i]))>0){
    this.context <- trinuc.contexts[[2]][[which(tumor.trinuc.names==signature.breakdown.ALL$Tumor[i])]]
    # t(this.context$weights)*signatures.cosmic[signature.breakdown.ALL$deconstructSigs_context[i]]
    # signatures.cosmic[signature.breakdown.ALL$deconstructSigs_context[i]]
    
    # this.context$tumor[,signature.breakdown.ALL$deconstructSigs_context[i]]
    # this.context$product[,signature.breakdown.ALL$deconstructSigs_context[i]]
    
    signature.breakdown.at.this.context <- t(this.context$weights)*signatures.cosmic[signature.breakdown.ALL$deconstructSigs_context[i]]
    
    signature.breakdown.ALL[i,c("APOBEC sum")] <- ifelse(all(is.na(signature.breakdown.at.this.context[c(2,13),])),NA,sum(signature.breakdown.at.this.context[c(2,13),]/this.context$product[,signature.breakdown.ALL$deconstructSigs_context[i]],na.rm = T))
    signature.breakdown.ALL[i,c("NonAPOBEC sum")] <- ifelse(all(is.na(signature.breakdown.at.this.context[c(1,3:12,14:30),])),NA,sum(signature.breakdown.at.this.context[c(1,3:12,14:30),]/this.context$product[,signature.breakdown.ALL$deconstructSigs_context[i]],na.rm = T))
  }
  # signature.breakdown.ALL$`APOBEC sum`[i] <- sum(signature.breakdown.ALL$Sig2[i],signature.breakdown.ALL$Sig13[i],na.rm = T)
}

#TODO: non apobec sum is just one minus the apobec sum... should we list all signatures for further analysis?

# head(signature.breakdown.ALL)

```








<!-- ```{r are APOBEC TCW type mutations enriched in higher gamma, eval=F} -->

<!-- ALL.prev.df -->
<!-- load("output_data/HNSC_MAF.RData") -->

<!-- HNSC.MAF <- MAF_for_analysis -->

<!-- ALL.prev.df$`APOBEC type TCW` <- "False" -->
<!-- ALL.prev.df$`APOBEC type TCW`[which(ALL.prev.df$Mutation %in% signature.breakdown.TCW$Name)] <- "True" -->

<!-- distribution.of.APOBEC.TCW <- ggplot(data = ALL.prev.df) + geom_violin(aes(y=log10(`Selection intensity`),x=`APOBEC type TCW`)) + geom_jitter(aes(y=log10(`Selection intensity`),x=`APOBEC type TCW`),alpha=0.5,width  = 0.1) + theme_bw() + labs(title="The distribution of selection intensities for APOBEC and nonAPOBEC mutations") -->

<!-- distribution.of.APOBEC.TCW <- ggplot(data = ALL.prev.df) + geom_boxplot(aes(y=`Selection intensity`,x=`APOBEC type TCW`),width=0.5,outlier.shape = NA) + geom_jitter(aes(y=`Selection intensity`,x=`APOBEC type TCW`),alpha=0.5,width  = 0.1) + theme_bw() + labs(y="Selection intensity") + scale_y_log10(labels=fancy_scientific)+ theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold")) + labs(x="TCW to TKW") -->

<!-- ggsave(plot = distribution.of.APOBEC.TCW,filename = "Figures/distribution_of_APOBEC_TCW.png",height = 5,width = 5) -->


<!-- wilcox.test(data=ALL.prev.df, `Selection intensity`~`APOBEC type TCW`) -->


<!-- ### trying to get the violin plot -->

<!-- # ALL.prev.df.logged <- ALL.prev.df -->
<!-- # ALL.prev.df.logged$`Selection intensity` <- log10(ALL.prev.df.logged$`Selection intensity`) -->
<!-- #  -->
<!-- # distribution.of.APOBEC.TCW.viol <- ggplot(data = ALL.prev.df.logged) + geom_violin(aes(y=`Selection intensity`,x=as.factor(`APOBEC type TCW`))) + geom_jitter(aes(y=`Selection intensity`,x=as.factor(`APOBEC type TCW`)),alpha=0.2,height = 0.1) + theme_bw() + labs(title="The distribution of selection intensities for \nAPOBEC and nonAPOBEC mutations") -->



<!-- ``` -->

```{r apobec weight sum plot ALL, echo=FALSE, fig.height=40, fig.width=5}

signature.breakdown.means <- as.data.frame(matrix(data=NA, nrow=length(unique(signature.breakdown.ALL$Name)),ncol=3))
colnames(signature.breakdown.means) <- c("Name","median APOBEC","median NonAPOBEC")
signature.breakdown.means$Name <- unique(signature.breakdown.ALL$Name)
for(i in 1:nrow(signature.breakdown.means)){
  signature.breakdown.means$`median APOBEC`[i] <- median(signature.breakdown.ALL$`APOBEC sum`[which(signature.breakdown.ALL$Name==signature.breakdown.means$Name[i])],na.rm = T)
  signature.breakdown.means$`median NonAPOBEC`[i] <- median(signature.breakdown.ALL$`NonAPOBEC sum`[which(signature.breakdown.ALL$Name==signature.breakdown.means$Name[i])],na.rm = T)
}

signature.breakdown.means <- signature.breakdown.means[order(signature.breakdown.means$`median APOBEC`,decreasing = T),]

signature.breakdown.ALL$Name <- factor(signature.breakdown.ALL$Name, levels = rev(signature.breakdown.means$Name))

signature.breakdown.ALL$short_name <- NA
for(i in 1:nrow(signature.breakdown.ALL)){
  if(endsWith(x = as.character(signature.breakdown.ALL$Name[i]),suffix = "NCSNV")){
   signature.breakdown.ALL$short_name[i] <- as.character(signature.breakdown.ALL$Name[i]) 
  }else{
  signature.breakdown.ALL$short_name[i] <- paste(strsplit(x = as.character(signature.breakdown.ALL$Name[i]), split=" ")[[1]][1:2],collapse = " ")
  }
}

ALL.APOBECplot <- ggplot(data = signature.breakdown.ALL) + geom_point(aes(x = `APOBEC sum`, y= Name),alpha=0.2)+# geom_point(aes(x = `NonAPOBEC sum`, y= Name),alpha=0.2,color="red") + 
  theme_bw() + labs(x="Proportion of mutation from APOBEC signatures per tumor", y= "Mutation", title="All mutations and the proportion \nof the trinucleotide context that was APOBEC", subtitle="Ranked in descending order of \nmedian proportion APOBEC contribution")

# ALL.APOBECplot

# ALL.to.TKN.APOBECplot
ggsave(plot = ALL.APOBECplot,filename = "Figures/ALL.to.TKN.APOBECplot.pdf",width = 7,height = 40)

save(signature.breakdown.ALL,file = "output_data/signature_breakdown_all.RData")
```



```{r display ALL data table, echo=F}

load("output_data/HNSC_MAF.RData")
HNSC.MAF <- MAF_for_analysis


ALL.prev.df <- cbind(ALL.prev.df,NA,NA,NA,NA)
colnames(ALL.prev.df)[6:7] <- c("Median APOBEC contribution", "Selection from APOBEC")
colnames(ALL.prev.df)[8:9] <- c("Median NonAPOBEC contribution", "Selection from NonAPOBEC")



for(i in 1:nrow(ALL.prev.df)){
  ALL.prev.df$`Median APOBEC contribution`[i] <- round(median(signature.breakdown.ALL$`APOBEC sum`[which(signature.breakdown.ALL$Name==ALL.prev.df$Mutation[i])],na.rm = T),2)
  ALL.prev.df$`Median NonAPOBEC contribution`[i] <- round(median(signature.breakdown.ALL$`NonAPOBEC sum`[which(signature.breakdown.ALL$Name==ALL.prev.df$Mutation[i])],na.rm = T),2)
  
}

# if there are substitutions that all came from tumors with less than enough mutations to determine signatures
# we need to remove these substitutions.
if(length(which(is.na(ALL.prev.df$`Median APOBEC contribution`)))>0){
  message(paste(length(which(is.na(ALL.prev.df$`Median APOBEC contribution`)))," substitutions occur in only tumors
that did not have enough mutations to measure trinucleotide context
and will be removed from the analysis",sep=""))
  ALL.prev.df <- ALL.prev.df[-which(is.na(ALL.prev.df$`Median APOBEC contribution`)),]
}
ALL.prev.df$`Selection from APOBEC` <- round((ALL.prev.df$Frequency * ALL.prev.df$`Selection intensity` * ALL.prev.df$`Median APOBEC contribution`)/length(unique(HNSC.MAF$Unique_patient_identifier)),2)

ALL.prev.df$`Selection from NonAPOBEC` <- round((ALL.prev.df$Frequency * ALL.prev.df$`Selection intensity` * ALL.prev.df$`Median NonAPOBEC contribution`)/length(unique(HNSC.MAF$Unique_patient_identifier)),2)

ALL.prev.df <- ALL.prev.df[order(ALL.prev.df$`Selection from APOBEC`,decreasing = T),]

library(DT)

datatable(data = ALL.prev.df)

save(ALL.prev.df, file = "output_data/variant_selection_contribution_wAPOBEC.RData")

# ALL.prev.df[which(ALL.prev.df$Mutation=="PIK3CA E545K 178936091A"),]

```

```{r Mutation type vs gamma from APOBEC or not, eval=T}

head(signature.breakdown.ALL)
head(ALL.prev.df)

ALL.prev.df.split <- as.data.frame(matrix(data = NA,nrow = 2*nrow(ALL.prev.df),ncol=5))
colnames(ALL.prev.df.split) <- c("Name","Selection from mutational context","APOBEC context", "TCW to TKW", "TCN to TKN")
ALL.prev.df.split$Name <- c(ALL.prev.df$Mutation,ALL.prev.df$Mutation)
ALL.prev.df.split$`APOBEC context` <-  c(rep("APOBEC processes",nrow(ALL.prev.df.split)/2),rep("non-APOBEC processes",nrow(ALL.prev.df.split)/2))


for(i in 1:length(unique(ALL.prev.df.split$Name))){
  these.pos <- which(ALL.prev.df.split$Name ==  unique(ALL.prev.df.split$Name)[i])
  # ALL.prev.df.split$`TCW to TKW`[these.pos] <- ALL.prev.df$`APOBEC type TCW`[i]
  # ALL.prev.df.split$`TCN to TKN`[these.pos] <- ALL.prev.df$`APOBEC type TCN`[i]
  ALL.prev.df.split$`Selection from mutational context`[these.pos] <- c(ALL.prev.df$`Selection from APOBEC`[i],ALL.prev.df$`Selection from NonAPOBEC`[i])
}

head(ALL.prev.df.split)

ALL.prev.df.split$short_name <- NA
for(i in 1:nrow(ALL.prev.df.split)){
  ALL.prev.df.split$short_name[i] <- paste(strsplit(x = ALL.prev.df.split$Name[i], split=" ")[[1]][1:2],collapse = " ")
}

# this quick fix messes up the non-coding mutations, fix the one we want labeled 

ALL.prev.df.split$short_name[which(ALL.prev.df.split$Name=="CCDC50 A 191098615 C NCSNV")] <- "CCDC50 Splice Site"

# ggplot(data = ALL.prev.df.split) +geom_violin(aes(x=`APOBEC context`, y = `Selection from mutational context`))  + geom_jitter(aes(x=`APOBEC context`, y = `Selection from mutational context`))

selection.from.contexts <- ggplot(data = ALL.prev.df.split) +geom_violin(aes(x=`APOBEC context`, y = log10(`Selection from mutational context`)))  + geom_jitter(aes(x=`APOBEC context`, y = log10(`Selection from mutational context`))) + labs(x="Mutational signature attributed to mutation")
source("R/fancy_scientific_code.R")

# selection.from.contexts <- ggplot(data = ALL.prev.df.split) +geom_violin(aes(x=`APOBEC context`, y = `Selection from mutational context`))  + geom_jitter(aes(x=`APOBEC context`, y = `Selection from mutational context`)) + scale_y_log10(labels=fancy_scientific) + labs(x="Mutational signatures attributed to mutation", y="Proportion of selection intensity attributed to signature and \nweighted by mutation prevalence among tumors") + theme_bw()

ALL.prev.df.split$`APOBEC context` <- factor(ALL.prev.df.split$`APOBEC context`, levels = c("non-APOBEC processes", "APOBEC processes"))

library(ggrepel)

selection.from.contexts.labels <- ggplot(data = ALL.prev.df.split) +#geom_violin(aes(x=`APOBEC context`, y = `Selection from mutational context`))  +
  geom_point(aes(x=`APOBEC context`, y = `Selection from mutational context`),alpha=0.2) + 
  geom_text_repel(data = subset(ALL.prev.df.split,`APOBEC context`=="APOBEC processes" & `Selection from mutational context` > 100),aes(x=`APOBEC context`, y = `Selection from mutational context`,label=short_name),color="darkred",fontface="bold",size=2.5) + #scale_y_log10(labels=fancy_scientific) + 
  geom_text_repel(data = subset(ALL.prev.df.split,`APOBEC context`=="non-APOBEC processes" & `Selection from mutational context` > 650),aes(x=`APOBEC context`, y = `Selection from mutational context`,label=short_name),box.padding = .31,color="darkred",fontface="bold",size=2.5) + 
  labs(y="Net realized selection intensity") + theme_bw() + theme(axis.text=element_text(size=8), axis.title=element_text(size=10,face="bold"))

# ggsave(filename = "Figures/selection_from_process.png",plot = selection.from.contexts)
# View(ALL.prev.df.split)
ggsave(filename = "Figures/selection_from_process_labels.png",plot = selection.from.contexts.labels,width = 3.25,height = 3.25,dpi = 600)

wilcox.test(`Selection from mutational context`~`APOBEC context`,data = ALL.prev.df.split)

# HNSC.MAF[which(HNSC.MAF$Start_Position==191098615),]

```



```{r PIK3CA data}
# library(feather)
# apobec_hnscc_df <- read_feather("input_data/nfkb.feather")

signature.breakdown.ALL$HPV_status <- NA
# signature.breakdown.ALL$HPV_status_tang2013 <- NA

signature.breakdown.ALL$tumor_has_apobec_sig <- NA

load("output_data/trinuc_contexts_HNSC_MAF.RData")
tumor.trinuc.names <- NULL
for(i in 1:length(trinuc.contexts[[2]])){
  tumor.trinuc.names[i] <- rownames(trinuc.contexts[[2]][[i]]$weights)
}



for(i in 1:nrow(signature.breakdown.ALL)){
  if(length(which(SNV.APOBEC.df$Tumor==signature.breakdown.ALL$Tumor[i]))==1){
    signature.breakdown.ALL$HPV_status[i] <- as.character(SNV.APOBEC.df$HPV_status[which(SNV.APOBEC.df$Tumor==signature.breakdown.ALL$Tumor[i])])
  }
  # if(length(which(tang.2013$sample.short==signature.breakdown.ALL$Tumor[i]))==1){
  # signature.breakdown.ALL$HPV_status_tang2013[i] <- tang.2013$Virus.description[which(tang.2013$sample.short==signature.breakdown.ALL$Tumor[i])]
  # }
  # signature.breakdown.ALL$tumor_has_apobec_sig[i] <- apobec_hnscc_df$has_apobec[which(apobec_hnscc_df$patient_id==signature.breakdown.ALL$Tumor[i])]
  if(length(which(tumor.trinuc.names==signature.breakdown.ALL$Tumor[i]))>0){
    this.context <- trinuc.contexts[[2]][[which(tumor.trinuc.names==signature.breakdown.ALL$Tumor[i])]]
    signature.breakdown.ALL$tumor_has_apobec_sig[i] <- ifelse(sum(this.context$weights[,c(2,13)])>0,"TRUE","FALSE")
  }
}
just.PIK3CA <- signature.breakdown.ALL[which(startsWith(x = as.character(signature.breakdown.ALL$Name),prefix = "PIK3CA")),]
# just.PIK3CA$HPV_status_tang2013[just.PIK3CA$HPV_status_tang2013=="N/A"] <- "HPV-"
# just.PIK3CA$HPV_status_tang2013[!is.na(just.PIK3CA$HPV_status_tang2013) & just.PIK3CA$HPV_status_tang2013!="HPV-"] <- "HPV+"
# for(i in 1:nrow(signature.breakdown.ALL)){
#   if(length(which(tumor.trinuc.names==signature.breakdown.ALL$Tumor[i]))>0){
#     this.context <- trinuc.contexts[[2]][[which(tumor.trinuc.names==signature.breakdown.ALL$Tumor[i])]]
#     # t(this.context$weights)*signatures.cosmic[signature.breakdown.ALL$deconstructSigs_context[i]]
#     # signatures.cosmic[signature.breakdown.ALL$deconstructSigs_context[i]]
# this.context$weights[,c(2,13)]
#     # this.context$tumor[,signature.breakdown.ALL$deconstructSigs_context[i]]
#     # this.context$product[,signature.breakdown.ALL$deconstructSigs_context[i]]
# 
#     signature.breakdown.at.this.context <- t(this.context$weights)*signatures.cosmic[signature.breakdown.ALL$deconstructSigs_context[i]]
# 
#     signature.breakdown.ALL[i,c("APOBEC sum")] <- ifelse(all(is.na(signature.breakdown.at.this.context[c(2,13),])),NA,sum(signature.breakdown.at.this.context[c(2,13),]/this.context$product[,signature.breakdown.ALL$deconstructSigs_context[i]],na.rm = T))
#     signature.breakdown.ALL[i,c("NonAPOBEC sum")] <- ifelse(all(is.na(signature.breakdown.at.this.context[c(1,3:12,14:30),])),NA,sum(signature.breakdown.at.this.context[c(1,3:12,14:30),]/this.context$product[,signature.breakdown.ALL$deconstructSigs_context[i]],na.rm = T))
#   }
#   # signature.breakdown.ALL$`APOBEC sum`[i] <- sum(signature.breakdown.ALL$Sig2[i],signature.breakdown.ALL$Sig13[i],na.rm = T)
# }
# 
# View(just.PIK3CA)
# just.PIK3CA$HPV_status

# unique(just.PIK3CA$Name)

just.PIK3CA.unique <- as.data.frame(matrix(data = NA,nrow = length(unique(just.PIK3CA$Name)),ncol=12))
colnames(just.PIK3CA.unique) <- c("Name","mutations in dataset","us_pos","us_neg","us_frac_pos","tang_pos","tang_neg","tang_frac","neg_HPV_and_neg_apobec","neg_HPV_and_pos_apobec","pos_HPV_and_pos_apobec","pos_HPV_and_neg_apobec")

for(i in 1:nrow(just.PIK3CA.unique)){
  just.PIK3CA.unique$Name[i] <- as.character(unique(just.PIK3CA$Name)[i])
  just.PIK3CA.unique$`mutations in dataset`[i] <- length(which(just.PIK3CA$Name==just.PIK3CA.unique$Name[i]))
  just.PIK3CA.unique$us_pos[i] <- length(which(just.PIK3CA$HPV_status[just.PIK3CA$Name==just.PIK3CA.unique$Name[i]]=="HPV+"))
  just.PIK3CA.unique$us_neg[i] <- length(which(just.PIK3CA$HPV_status[just.PIK3CA$Name==just.PIK3CA.unique$Name[i]]=="HPV-"))
  # just.PIK3CA.unique$tang_pos[i] <- length(which(just.PIK3CA$HPV_status_tang2013[just.PIK3CA$Name==just.PIK3CA.unique$Name[i]]=="HPV+"))
  # just.PIK3CA.unique$tang_neg[i] <- length(which(just.PIK3CA$HPV_status_tang2013[just.PIK3CA$Name==just.PIK3CA.unique$Name[i]]=="HPV-"))
  just.PIK3CA.unique$us_frac_pos[i] <- just.PIK3CA.unique$us_pos[i]/(just.PIK3CA.unique$us_neg[i]+just.PIK3CA.unique$us_pos[i])
  # just.PIK3CA.unique$tang_frac[i] <- just.PIK3CA.unique$tang_pos[i]/(just.PIK3CA.unique$tang_neg[i]+just.PIK3CA.unique$tang_pos[i])
  just.PIK3CA.unique$neg_HPV_and_neg_apobec[i] <- length(which(just.PIK3CA$tumor_has_apobec_sig[just.PIK3CA$Name==just.PIK3CA.unique$Name[i]]=="FALSE" &  just.PIK3CA$HPV_status[just.PIK3CA$Name==just.PIK3CA.unique$Name[i]]=="HPV-"))
  just.PIK3CA.unique$neg_HPV_and_pos_apobec[i] <- length(which(just.PIK3CA$tumor_has_apobec_sig[just.PIK3CA$Name==just.PIK3CA.unique$Name[i]]=="TRUE" &  just.PIK3CA$HPV_status[just.PIK3CA$Name==just.PIK3CA.unique$Name[i]]=="HPV-"))
  just.PIK3CA.unique$pos_HPV_and_pos_apobec[i] <- length(which(just.PIK3CA$tumor_has_apobec_sig[just.PIK3CA$Name==just.PIK3CA.unique$Name[i]]=="TRUE" &  just.PIK3CA$HPV_status[just.PIK3CA$Name==just.PIK3CA.unique$Name[i]]=="HPV+"))
  just.PIK3CA.unique$pos_HPV_and_neg_apobec[i] <- length(which(just.PIK3CA$tumor_has_apobec_sig[just.PIK3CA$Name==just.PIK3CA.unique$Name[i]]=="FALSE" &  just.PIK3CA$HPV_status[just.PIK3CA$Name==just.PIK3CA.unique$Name[i]]=="HPV+"))
}

datatable(just.PIK3CA.unique)
```

```{r save analysis}
save.image(file = "output_data/completed_analysis_APOBEC_HNSCC.RData")
```

<!-- ```{r heatmap and dendrogram} -->
<!-- load("output_data/trinuc_contexts_HNSC_MAF.RData") -->
<!-- # install.packages("ggdendro") -->
<!-- library(ggdendro) -->
<!-- library(reshape2) -->
<!-- #  -->
<!-- # head(trinuc.contexts[[2]][1]) -->

<!-- weights.df <- trinuc.contexts$signature.weights[[1]]$weights -->
<!-- for(i in 2:length(trinuc.contexts$signature.weights)){ -->
<!--   weights.df <- rbind(weights.df,trinuc.contexts$signature.weights[[i]]$weights) -->
<!-- } -->

<!-- #just with known HPV status -->
<!-- weights.df <- weights.df[which(rownames(weights.df) %in% HNSC.selection.HPVknown$Unique_patient_identifier),] -->

<!-- head(weights.df) -->
<!-- weights.df$id <- rownames(weights.df) -->
<!-- weights.df.melted <- melt(weights.df) -->

<!-- head(weights.df.melted) -->


<!-- ggplot(data = weights.df.melted) + geom_tile(aes(x = id, y=variable,fill=value)) + theme(axis.text.x = element_text(angle=90)) -->


<!-- # #setting up with dendrogram  -->
<!-- weights.df <- trinuc.contexts$signature.weights[[1]]$weights -->
<!-- for(i in 2:length(trinuc.contexts$signature.weights)){ -->
<!--   weights.df <- rbind(weights.df,trinuc.contexts$signature.weights[[i]]$weights) -->
<!-- } -->
<!-- #TODO:  keep in the ambiguous and have it as a difference color -->
<!-- # weights.df <- weights.df[which(rownames(weights.df) %in% HNSC.selection.HPVknown$Unique_patient_identifier),] -->


<!-- dd.col <- as.dendrogram(hclust(dist(weights.df))) -->
<!-- col.ordered <- order.dendrogram(dd.col) -->

<!-- dd.row <-  as.dendrogram(hclust(dist(t(weights.df)))) -->
<!-- row.ordered <- order.dendrogram(dd.row) -->

<!-- xx <- weights.df[col.ordered,row.ordered] -->
<!-- # xx_names <- attr(xx, "dimnames") -->
<!-- df <- as.data.frame(xx) -->
<!-- colnames(df) #<- xx_names[[2]] -->
<!-- df$tumor <- rownames(xx) -->
<!-- df$tumor <- with(df, factor(tumor, levels=tumor, ordered=TRUE)) -->

<!-- mdf <- melt(df, id.vars="tumor") -->
<!-- mdf$`HPV status` <- NA -->

<!-- for(i in 1:nrow(mdf)){ -->
<!--   mdf$`HPV status`[i] <- HNSC.selection.output$HPV_call[which(HNSC.selection.output$Unique_patient_identifier==as.character(mdf$tumor[i]))[1]] -->

<!-- } -->

<!-- ddata_x <- dendro_data(dd.row) -->
<!-- ddata_y <- dendro_data(dd.col) -->

<!-- ### Set up a blank theme -->
<!-- theme_none <- theme( -->
<!--   panel.grid.major = element_blank(), -->
<!--   panel.grid.minor = element_blank(), -->
<!--   panel.background = element_blank(), -->
<!--   axis.title.x = element_text(colour=NA), -->
<!--   axis.title.y = element_blank(), -->
<!--   axis.text.x = element_blank(), -->
<!--   axis.text.y = element_blank(), -->
<!--   axis.line = element_blank() -->
<!--   #axis.ticks.length = element_blank() -->
<!-- ) -->
<!-- length(unique(mdf$tumor)) -->
<!-- ### Create plot components ###     -->
<!-- # Heatmap -->
<!-- a <- rep(NA,length(mdf$`HPV status`)) -->
<!-- for(i in 1:length(a)){ -->
<!--   if(mdf$`HPV status`[i] == "HPV-"){a[i] <- "black"}  -->
<!--   if(mdf$`HPV status`[i] == "HPV+"){a[i] <- "red"} -->
<!--   if(mdf$`HPV status`[i] == "ambiguous"){a[i] <- "blue"} -->
<!-- } -->

<!-- # <- ifelse(mdf$`HPV status`=="HPV+","red","black") -->

<!-- p1 <- ggplot(mdf, aes(y=variable, x=tumor)) +  -->
<!--   geom_tile(aes(fill=value)) + scale_fill_gradient2() + theme(axis.text.y = element_text(angle=0)) + theme(axis.text.x=element_text(colour = a,size=2,angle=90,face="bold")) + theme(legend.position = "none") + theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) + theme(axis.title.y = element_blank()) -->

<!-- # Dendrogram 1 -->
<!-- p2 <- ggplot(segment(ddata_x)) +  -->
<!--   geom_segment(aes(x=x, y=y, xend=xend, yend=yend)) +  -->
<!--   theme_none + theme(axis.title.x=element_blank()) + coord_flip() +scale_y_reverse() + theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))# + theme(axis.text = element_blank()) -->

<!-- # Dendrogram 2 -->
<!-- p3 <- ggplot(segment(ddata_y)) +  -->
<!--   geom_segment(aes(x=x, y=y, xend=xend, yend=yend))  + theme_none + theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) -->

<!-- # install.packages("cowplot") -->
<!-- library(cowplot) -->

<!-- plot_grid(NULL,p3,p2,p1,ncol=2,rel_widths = c(2,5),align = "hv") -->

<!-- #  -->
<!-- # grid.newpage() -->
<!-- # print(p1, vp=viewport(0.8, 0.8, x=0.4, y=0.4)) -->
<!-- # print(p3, vp=viewport(width = 0.60, 0.2, x=0.40, y=0.9)) -->
<!-- # print(p2, vp=viewport(0.2, 0.8, x=0.9, y=0.45)) -->
<!-- #  -->
<!-- # hpvPsig2 <- subset(mdf, `HPV status`=="HPV+" & variable=="Signature.2") -->
<!-- # hpvNsig2 <- subset(mdf, `HPV status`=="HPV-" & variable=="Signature.2") -->
<!-- #  -->
<!-- # hpvPsig13 <- subset(mdf, `HPV status`=="HPV+" & variable=="Signature.13") -->
<!-- # hpvNsig13 <- subset(mdf, `HPV status`=="HPV-" & variable=="Signature.13") -->
<!-- #  -->
<!-- # mean(hpvPsig2$value) -->
<!-- #  -->
<!-- # mean(hpvNsig2$value) -->
<!-- # mean(hpvPsig13$value) -->
<!-- # mean(hpvNsig13$value) -->

<!-- ``` -->




<!-- ```{r} -->

<!-- dd <- dist(scale(weights.df), method = "euclidean") -->
<!-- hc <- hclust(dd, method = "ward.D2") -->
<!-- ggdendrogram(data = hc) -->


<!-- dd <- dist(weights.df, method = "euclidean") -->
<!-- hc <- hclust(dd, method = "ward.D2") -->
<!-- ggdendrogram(data = hc) -->
<!-- ``` -->

<!-- # ```{r} -->
<!-- # install.packages("gplots") -->
<!-- # library(gplots) -->
<!-- #  -->
<!-- # heatmap.2(t(weights.df)) -->
<!-- # ``` -->

