---
title: "HNSC APOBEC Targets"
author: "Vincent Cannataro"
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
  html_notebook:
    toc: yes
    toc_depth: 3
    toc_float:
      collapsed: no
---



```{r setup, include=FALSE, message=FALSE, warning=FALSE}
library(DT)
library(tidyverse)
```


```{r import and combine data, include=FALSE}

if(length(which(list.files(recursive = T)=="output_data/HNSC_selection_with_APOBEC_TCW_recur.RData"))==0){
  dir.create("output_data")
  # Many created in HNSC_APOBEC_analysis.Rmd
  
  load("~/Documents/Selection_analysis/HNSC/selection_output/HNSC_selection_output.RData")
  HNSC.selection.output <- as.tibble(selection.output$complete_mutation_data) %>%
    select(Gene, starts_with("Nucleo"), 
           Chromosome, starts_with("Reference"),
           starts_with("Alternative"), Tumor_origin, 
           Unique_patient_identifier, starts_with("Amino"), Codon_position, synonymous.mu, trinucs, Gamma_epistasis)  
  # HNSC.selection.output
  
  # Assigning HPV status
  load("~/Documents/manuscripts/APOBEC_HNSCC/HNSC_MAF.RData") 
  HNSC.selection.output$HPV_call <- NA
  for(i in 1:length(unique(HNSC.selection.output$Unique_patient_identifier))){
    HNSC.selection.output$HPV_call[which(HNSC.selection.output$Unique_patient_identifier == unique(HNSC.selection.output$Unique_patient_identifier)[i])] <- HNSC.MAF$HPV_call[which(HNSC.MAF$Unique_patient_identifier == unique(HNSC.selection.output$Unique_patient_identifier)[i])[1]]
  }
  
  
  load("~/Documents/manuscripts/APOBEC_HNSCC/trinuc_contexts_HPV_all.RData")
  
  weights.df <- trinuc.contexts$signature.weights[[1]]$weights
  for(i in 2:length(trinuc.contexts$signature.weights)){
    weights.df <- rbind(weights.df,trinuc.contexts$signature.weights[[i]]$weights)
  }
  
  HNSC.selection.output$Sig_2 <- NA
  HNSC.selection.output$Sig_13 <- NA
  
  
  for(i in 1:length(unique(HNSC.selection.output$Unique_patient_identifier))){
    if(length(which(rownames(weights.df) == unique(HNSC.selection.output$Unique_patient_identifier)[i]))>0){
      HNSC.selection.output$Sig_2[which(HNSC.selection.output$Unique_patient_identifier == unique(HNSC.selection.output$Unique_patient_identifier)[i])] <- weights.df$Signature.2[which(rownames(weights.df) == unique(HNSC.selection.output$Unique_patient_identifier)[i])]
      HNSC.selection.output$Sig_13[which(HNSC.selection.output$Unique_patient_identifier == unique(HNSC.selection.output$Unique_patient_identifier)[i])] <- weights.df$Signature.13[which(rownames(weights.df) == unique(HNSC.selection.output$Unique_patient_identifier)[i])]
    }
  }
  
  
  HNSC.selection.output <- mutate(.data = HNSC.selection.output, APOBEC_weight = Sig_2 + Sig_13)
  
  head(HNSC.selection.output)
  
  
  HNSC.selection.output$TCW_TKW <- NA
  HNSC.selection.output$TCN_TKN <- NA
  
  for(j in 1:nrow(HNSC.selection.output)){
    if(is.na(HNSC.selection.output$Nucleotide_trinuc_context[j])){
      #need to make a call of the same nucleotide position and amino acid alternative, find which is NOT NA, and make this the new "j" 
      matches <- which(HNSC.selection.output$Nucleotide_chromosome_position == HNSC.selection.output$Nucleotide_chromosome_position[j] &
                         HNSC.selection.output$Alternative_Nucleotide == HNSC.selection.output$Alternative_Nucleotide[j])
      
      new.j <- matches[which(!is.na(HNSC.selection.output$Nucleotide_trinuc_context[matches]))]
      if(((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="T") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="A")) | 
         ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="T") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="A")) |
         ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="G") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="C")) |
         ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="G") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="C"))){
        HNSC.selection.output$TCW_TKW[j] <- 1
      }else{
        HNSC.selection.output$TCW_TKW[j] <- 0
      }
      
    }else{
      if(((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[j]=="T") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="A")) | 
         ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[j]=="T") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="A")) |
         ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[j]=="G") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="C")) |
         ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[j]=="G") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="C"))){
        HNSC.selection.output$TCW_TKW[j] <- 1
      }else{
        HNSC.selection.output$TCW_TKW[j] <- 0
      }
    }
    
    # Now, mutations that could be TCN --> TKN
    if(is.na(HNSC.selection.output$Nucleotide_trinuc_context[j])){
      #need to make a call of the same nucleotide position and amino acid alternative, find which is NOT NA, and make this the new "j" 
      matches <- which(HNSC.selection.output$Nucleotide_chromosome_position == HNSC.selection.output$Nucleotide_chromosome_position[j] &
                         HNSC.selection.output$Alternative_Nucleotide == HNSC.selection.output$Alternative_Nucleotide[j])
      
      new.j <- matches[which(!is.na(HNSC.selection.output$Nucleotide_trinuc_context[matches]))]
      if(((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="T") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="A")) | 
         ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="T") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="A")) |
         ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="G") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="C")) |
         ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="G") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="C")) |
         
         ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCC" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="T") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="GGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="A")) | 
         ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCG" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="T") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="CGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="A")) |
         ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCC" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="G") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="GGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="C")) |
         ((HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="TCG" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="G") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[new.j]=="CGA" & HNSC.selection.output$Alternative_Nucleotide[new.j]=="C"))
      ){
        HNSC.selection.output$TCN_TKN[j] <- 1
      }else{
        HNSC.selection.output$TCN_TKN[j] <- 0
      }
      
    }else{
      if(((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[j]=="T") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="A")) | 
         ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[j]=="T") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="A")) |
         ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCA" & HNSC.selection.output$Alternative_Nucleotide[j]=="G") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[j]=="TGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="C")) |
         ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCT" & HNSC.selection.output$Alternative_Nucleotide[j]=="G") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[j]=="AGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="C"))|
         
         ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCC" & HNSC.selection.output$Alternative_Nucleotide[j]=="T") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[j]=="GGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="A")) | 
         ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCG" & HNSC.selection.output$Alternative_Nucleotide[j]=="T") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[j]=="CGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="A")) |
         ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCC" & HNSC.selection.output$Alternative_Nucleotide[j]=="G") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[j]=="GGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="C")) |
         ((HNSC.selection.output$Nucleotide_trinuc_context[j]=="TCG" & HNSC.selection.output$Alternative_Nucleotide[j]=="G") | 
          (HNSC.selection.output$Nucleotide_trinuc_context[j]=="CGA" & HNSC.selection.output$Alternative_Nucleotide[j]=="C"))){
        HNSC.selection.output$TCN_TKN[j] <- 1
      }else{
        HNSC.selection.output$TCN_TKN[j] <- 0
      }
    }
    
    
  }
  
  
  HNSC.selection.output$Name <- NA
  for(i in 1:nrow(HNSC.selection.output)){
    HNSC.selection.output$Name[i] <- paste(HNSC.selection.output$Gene[i]," ",ifelse(!is.na(HNSC.selection.output$Amino_acid_reference[i]),paste(HNSC.selection.output$Amino_acid_reference[i],HNSC.selection.output$Amino_acid_position[i],HNSC.selection.output$Amino_acid_alternative[i]," ", HNSC.selection.output$Chromosome[i],"_",HNSC.selection.output$Nucleotide_chromosome_position[i],HNSC.selection.output$Alternative_Nucleotide[i],sep=""),paste(HNSC.selection.output$Reference_Nucleotide[i],HNSC.selection.output$Nucleotide_chromosome_position[i],HNSC.selection.output$Alternative_Nucleotide[i],"NCSNV")),sep="")
  }
  
  HNSC.selection.output
  save(HNSC.selection.output,file = "output_data/HNSC_selection_with_APOBEC_TCW.RData")
  
  HNSC.selection.output.recur <- subset(HNSC.selection.output, Nucleotide_change_tally>1)
  save(HNSC.selection.output.recur, file = "output_data/HNSC_selection_with_APOBEC_TCW_recur.RData")
}else{
  load("output_data/HNSC_selection_with_APOBEC_TCW.RData")
  load("output_data/HNSC_selection_with_APOBEC_TCW_recur.RData")
}
# HNSC.selection.output.recur[which(HNSC.selection.output.recur$Nucleotide_chromosome_position==7577017),]

# View(HNSC.selection.output[which(HNSC.selection.output$Gene=="TP53" & HNSC.selection.output$Amino_acid_position==285),])
# View(HNSC.selection.output.recur[which(HNSC.selection.output.recur$Gene=="TP53" & HNSC.selection.output.recur$Amino_acid_position==285),])

# test <- subset(HNSC.selection.output.recur, TCW_TKW==1)
# HNSC.selection.output[40967:40970,]

# HNSC.selection.output[which(HNSC.selection.output$TCN_TKN==1 & HNSC.selection.output$TCW_TKW==0)[1:5],]
```

# TCW to TKW

## APOBEC mutations that have the highest prevalence and selection intensity in HNSCC

Below is a data tables listing all genomic variants caused by the TCW --> TKW mutations in HNSCC.

```{r listing most prevalent APOBEC muts TCW, echo=F}

just.TCW <- HNSC.selection.output.recur[HNSC.selection.output.recur$TCW_TKW==TRUE,]
TCW.names <- unique(just.TCW$Name)

TCW.prev.df <- as.data.frame(matrix(data = NA, nrow=length(TCW.names),ncol=5))
colnames(TCW.prev.df) <- c("Mutation","Frequency", "Trinuc context", "Alternative Nucleotide","Selection intensity")

TCW.prev.df$Mutation <- TCW.names

for(i in 1:nrow(TCW.prev.df)){
  TCW.prev.df$Frequency[i] <- length(which(just.TCW$Name == TCW.prev.df$Mutation[i]))
  TCW.prev.df$`Trinuc context`[i] <- just.TCW$Nucleotide_trinuc_context[which(just.TCW$Name == TCW.prev.df$Mutation[i])[1]]
  TCW.prev.df$`Alternative Nucleotide`[i] <- just.TCW$Alternative_Nucleotide[which(just.TCW$Name == TCW.prev.df$Mutation[i])[1]]
  TCW.prev.df$`Selection intensity`[i] <- round(just.TCW$Gamma_epistasis[which(just.TCW$Name == TCW.prev.df$Mutation[i])[1]],2)
}

TCW.prev.df <- TCW.prev.df[order(TCW.prev.df$Frequency,decreasing = T),]

# datatable(data = TCW.prev.df)

```


## Mutations with the highest median contribution from APOBEC signatures


```{r Contribution of the APOBEC signature to each mutation in each tumor, include=FALSE}
library(deconstructSigs)

# load("HNSC_all_MAF.RData")
load("HNSC_MAF.RData")

load("trinuc_contexts_HPV_all.RData")
# length(unique(HNSC.HPVall.MAF$Unique_patient_identifier))
# length(trinuc.contexts[[2]])
tumor.trinuc.names <- NULL
for(i in 1:length(trinuc.contexts[[2]])){
  tumor.trinuc.names[i] <- rownames(trinuc.contexts[[2]][[i]]$weights)
}



# signatures.cosmic

# trinuc.contexts[[2]][[1]]$product
# as.matrix(trinuc.contexts[[2]][[1]]$weights) %*% as.matrix(signatures.cosmic) == as.matrix(trinuc.contexts[[2]][[1]]$product)


# head(just.TCW)
first.12 <- function(string_to_12){
  return(paste(unlist(strsplit(string_to_12,split = ""))[1:12],collapse = ""))
}

signature.breakdown.TCW <- as.data.frame(matrix(data = NA, nrow=nrow(just.TCW), ncol= 8))
colnames(signature.breakdown.TCW) <- c("Name","Tumor","Trinuc_context","Alternative_Nuc","deconstructSigs_context","Sig2","Sig13","APOBEC sum")

signature.breakdown.TCW$Name <- just.TCW$Name
signature.breakdown.TCW$Tumor <- unlist(lapply(just.TCW$Tumor_origin,first.12))

for(i in 1:nrow(just.TCW)){
  if(is.na(just.TCW$Nucleotide_trinuc_context[i])){
    matches <- which(just.TCW$Nucleotide_chromosome_position == just.TCW$Nucleotide_chromosome_position[i] &
                       just.TCW$Alternative_Nucleotide == just.TCW$Alternative_Nucleotide[i])
    
    new.i <- matches[which(!is.na(just.TCW$Nucleotide_trinuc_context[matches]))]
    signature.breakdown.TCW$Trinuc_context[i] <- just.TCW$Nucleotide_trinuc_context[new.i]
  }else{
    signature.breakdown.TCW$Trinuc_context[i] <- just.TCW$Nucleotide_trinuc_context[i]
  }
}

signature.breakdown.TCW$Alternative_Nuc <- just.TCW$Alternative_Nucleotide

for(i in 1:nrow(signature.breakdown.TCW)){
  signature.breakdown.TCW$deconstructSigs_context[i] <- paste(
    strsplit(signature.breakdown.TCW$Trinuc_context[i],split = "")[[1]][1],
    "[",
    strsplit(signature.breakdown.TCW$Trinuc_context[i],split = "")[[1]][2],
    ">",
    signature.breakdown.TCW$Alternative_Nuc[i],
    "]",
    strsplit(signature.breakdown.TCW$Trinuc_context[i],split = "")[[1]][3],sep="")
  
}

signature.breakdown.TCW$deconstructSigs_context[which(signature.breakdown.TCW$deconstructSigs_context=="T[G>A]A")] <- "T[C>T]A"
signature.breakdown.TCW$deconstructSigs_context[which(signature.breakdown.TCW$deconstructSigs_context=="A[G>A]A")] <- "T[C>T]T"
signature.breakdown.TCW$deconstructSigs_context[which(signature.breakdown.TCW$deconstructSigs_context=="T[G>C]A")] <- "T[C>G]A"
signature.breakdown.TCW$deconstructSigs_context[which(signature.breakdown.TCW$deconstructSigs_context=="A[G>C]A")] <- "T[C>G]T"


for(i in 1:nrow(signature.breakdown.TCW)){
  if(length(which(tumor.trinuc.names==signature.breakdown.TCW$Tumor[i]))>0){
    this.context <- trinuc.contexts[[2]][[which(tumor.trinuc.names==signature.breakdown.TCW$Tumor[i])]]
    # t(this.context$weights)*signatures.cosmic[signature.breakdown.TCW$deconstructSigs_context[i]]
    # signatures.cosmic[signature.breakdown.TCW$deconstructSigs_context[i]]
    
    # this.context$tumor[,signature.breakdown.TCW$deconstructSigs_context[i]]
    # this.context$product[,signature.breakdown.TCW$deconstructSigs_context[i]]
    
    signature.breakdown.at.this.context <- t(this.context$weights)*signatures.cosmic[signature.breakdown.TCW$deconstructSigs_context[i]]
    
    signature.breakdown.TCW[i,c("Sig2","Sig13")] <- signature.breakdown.at.this.context[c(2,13),]/this.context$product[,signature.breakdown.TCW$deconstructSigs_context[i]]
    
  }
  signature.breakdown.TCW$`APOBEC sum`[i] <- ifelse(all(is.na(c(signature.breakdown.TCW$Sig2[i],signature.breakdown.TCW$Sig13[i]))),NA,sum(signature.breakdown.TCW$Sig2[i],signature.breakdown.TCW$Sig13[i],na.rm = T))
}


```





```{r apobec weight sum plot, echo=FALSE, fig.height=15, fig.width=7}

signature.breakdown.means <- as.data.frame(matrix(data=NA, nrow=length(unique(signature.breakdown.TCW$Name)),ncol=2))
colnames(signature.breakdown.means) <- c("Name","mean")
signature.breakdown.means$Name <- unique(signature.breakdown.TCW$Name)
for(i in 1:nrow(signature.breakdown.means)){
  signature.breakdown.means$mean[i] <- median(signature.breakdown.TCW$`APOBEC sum`[which(signature.breakdown.TCW$Name==signature.breakdown.means$Name[i])],na.rm = T) 
}

signature.breakdown.means <- signature.breakdown.means[order(signature.breakdown.means$mean,decreasing = T),]

signature.breakdown.TCW$Name <- factor(signature.breakdown.TCW$Name, levels = rev(signature.breakdown.means$Name))

TCW.to.TKW.APOBECplot <- ggplot(data = signature.breakdown.TCW) + geom_point(aes(x = `APOBEC sum`, y= Name),alpha=0.5) + theme_bw() + labs(x="Proportion of mutation from APOBEC signatures per tumor", y= "Mutation", title="TCW to TKW mutations and the proportion \nof the trinucleotide context that was APOBEC (2 or 13)", subtitle="Ranked in descending order of median proportion")
# TCW.to.TKW.APOBECplot
ggsave(plot = TCW.to.TKW.APOBECplot,filename = "Figures/TCW.to.TKW.APOBECplot.pdf",width = 7,height = 15)
```


```{r display TCW data table, echo=F}
TCW.prev.df <- cbind(TCW.prev.df,NA,NA)
colnames(TCW.prev.df)[6:7] <- c("Median APOBEC contribution", "Selection from APOBEC")
for(i in 1:nrow(TCW.prev.df)){
  TCW.prev.df$`Median APOBEC contribution`[i] <- round(median(signature.breakdown.TCW$`APOBEC sum`[which(signature.breakdown.TCW$Name==TCW.prev.df$Mutation[i])],na.rm = T),2)
}

TCW.prev.df$`Selection from APOBEC` <- round((TCW.prev.df$Frequency * TCW.prev.df$`Selection intensity` * TCW.prev.df$`Median APOBEC contribution`)/length(unique(HNSC.MAF$Unique_patient_identifier)),2)

TCW.prev.df <- TCW.prev.df[order(TCW.prev.df$`Selection from APOBEC`,decreasing = T),]

datatable(data = TCW.prev.df)

```

```{r display apobec plot tcw, echo=FALSE, fig.height=15, fig.width=7}
TCW.to.TKW.APOBECplot
```



# TCN to TKN 

## APOBEC mutations that have the highest prevalence and selection intensity in HNSCC

Below is a data tables listing all genomic variants caused by the TCN --> TKN mutations in HNSCC.


```{r listing most prevalent APOBEC muts TCN, echo=F}

just.TCN <- HNSC.selection.output.recur[HNSC.selection.output.recur$TCN_TKN==TRUE,]
TCN.names <- unique(just.TCN$Name)

TCN.prev.df <- as.data.frame(matrix(data = NA, nrow=length(TCN.names),ncol=5))
colnames(TCN.prev.df) <- c("Mutation","Frequency", "Trinuc context","Alternative Nucleotide", "Selection intensity")

TCN.prev.df$Mutation <- TCN.names

for(i in 1:nrow(TCN.prev.df)){
  TCN.prev.df$Frequency[i] <- length(which(just.TCN$Name == TCN.prev.df$Mutation[i]))
  TCN.prev.df$`Trinuc context`[i] <- just.TCN$Nucleotide_trinuc_context[which(just.TCN$Name == TCN.prev.df$Mutation[i])[1]]
  TCN.prev.df$`Alternative Nucleotide`[i] <- just.TCN$Alternative_Nucleotide[which(just.TCN$Name == TCN.prev.df$Mutation[i])[1]]
  TCN.prev.df$`Selection intensity`[i] <- round(just.TCN$Gamma_epistasis[which(just.TCN$Name == TCN.prev.df$Mutation[i])[1]],2)
}

TCN.prev.df <- TCN.prev.df[order(TCN.prev.df$Frequency,decreasing = T),]

# datatable(data = TCN.prev.df)

```



## Mutations with the highest median contribution from APOBEC signatures


```{r Contribution of the APOBEC signature to each mutation in each tumor TCN, include=FALSE}
library(deconstructSigs)

# load("HNSC_all_MAF.RData")
load("HNSC_MAF.RData")

load("trinuc_contexts_HPV_all.RData")
# length(unique(HNSC.HPVall.MAF$Unique_patient_identifier))
# length(trinuc.contexts[[2]])
tumor.trinuc.names <- NULL
for(i in 1:length(trinuc.contexts[[2]])){
  tumor.trinuc.names[i] <- rownames(trinuc.contexts[[2]][[i]]$weights)
}



signatures.cosmic

trinuc.contexts[[2]][[1]]$product
# as.matrix(trinuc.contexts[[2]][[1]]$weights) %*% as.matrix(signatures.cosmic) == as.matrix(trinuc.contexts[[2]][[1]]$product)


# head(just.TCW)
first.12 <- function(string_to_12){
  return(paste(unlist(strsplit(string_to_12,split = ""))[1:12],collapse = ""))
}

signature.breakdown.TCN <- as.data.frame(matrix(data = NA, nrow=nrow(just.TCN), ncol= 8))
colnames(signature.breakdown.TCN) <- c("Name","Tumor","Trinuc_context","Alternative_Nuc","deconstructSigs_context","Sig2","Sig13","APOBEC sum")

signature.breakdown.TCN$Name <- just.TCN$Name
signature.breakdown.TCN$Tumor <- unlist(lapply(just.TCN$Tumor_origin,first.12))

for(i in 1:nrow(just.TCN)){
  if(is.na(just.TCN$Nucleotide_trinuc_context[i])){
    matches <- which(just.TCN$Nucleotide_chromosome_position == just.TCN$Nucleotide_chromosome_position[i] &
                       just.TCN$Alternative_Nucleotide == just.TCN$Alternative_Nucleotide[i])
    
    new.i <- matches[which(!is.na(just.TCN$Nucleotide_trinuc_context[matches]))]
    signature.breakdown.TCN$Trinuc_context[i] <- just.TCN$Nucleotide_trinuc_context[new.i]
  }else{
    signature.breakdown.TCN$Trinuc_context[i] <- just.TCN$Nucleotide_trinuc_context[i]
  }
}

signature.breakdown.TCN$Alternative_Nuc <- just.TCN$Alternative_Nucleotide

for(i in 1:nrow(signature.breakdown.TCN)){
  signature.breakdown.TCN$deconstructSigs_context[i] <- paste(
    strsplit(signature.breakdown.TCN$Trinuc_context[i],split = "")[[1]][1],
    "[",
    strsplit(signature.breakdown.TCN$Trinuc_context[i],split = "")[[1]][2],
    ">",
    signature.breakdown.TCN$Alternative_Nuc[i],
    "]",
    strsplit(signature.breakdown.TCN$Trinuc_context[i],split = "")[[1]][3],sep="")
  
}

signature.breakdown.TCN$deconstructSigs_context[which(signature.breakdown.TCN$deconstructSigs_context=="T[G>A]A")] <- "T[C>T]A"
signature.breakdown.TCN$deconstructSigs_context[which(signature.breakdown.TCN$deconstructSigs_context=="A[G>A]A")] <- "T[C>T]T"
signature.breakdown.TCN$deconstructSigs_context[which(signature.breakdown.TCN$deconstructSigs_context=="T[G>C]A")] <- "T[C>G]A"
signature.breakdown.TCN$deconstructSigs_context[which(signature.breakdown.TCN$deconstructSigs_context=="A[G>C]A")] <- "T[C>G]T"

signature.breakdown.TCN$deconstructSigs_context[which(signature.breakdown.TCN$deconstructSigs_context=="C[G>C]A")] <- "T[C>G]G"
signature.breakdown.TCN$deconstructSigs_context[which(signature.breakdown.TCN$deconstructSigs_context=="C[G>A]A")] <- "T[C>T]G"
signature.breakdown.TCN$deconstructSigs_context[which(signature.breakdown.TCN$deconstructSigs_context=="G[G>A]A")] <- "T[C>T]C"
signature.breakdown.TCN$deconstructSigs_context[which(signature.breakdown.TCN$deconstructSigs_context=="G[G>C]A")] <- "T[C>G]C"


for(i in 1:nrow(signature.breakdown.TCN)){
  if(length(which(tumor.trinuc.names==signature.breakdown.TCN$Tumor[i]))>0){
    this.context <- trinuc.contexts[[2]][[which(tumor.trinuc.names==signature.breakdown.TCN$Tumor[i])]]
    # t(this.context$weights)*signatures.cosmic[signature.breakdown.TCN$deconstructSigs_context[i]]
    # signatures.cosmic[signature.breakdown.TCN$deconstructSigs_context[i]]
    
    # this.context$tumor[,signature.breakdown.TCN$deconstructSigs_context[i]]
    # this.context$product[,signature.breakdown.TCN$deconstructSigs_context[i]]
    
    signature.breakdown.at.this.context <- t(this.context$weights)*signatures.cosmic[signature.breakdown.TCN$deconstructSigs_context[i]]
    
    signature.breakdown.TCN[i,c("Sig2","Sig13")] <- signature.breakdown.at.this.context[c(2,13),]/this.context$product[,signature.breakdown.TCN$deconstructSigs_context[i]]
    
  }
  signature.breakdown.TCN$`APOBEC sum`[i] <- ifelse(all(is.na(c(signature.breakdown.TCN$Sig2[i],signature.breakdown.TCN$Sig13[i]))),NA,sum(signature.breakdown.TCN$Sig2[i],signature.breakdown.TCN$Sig13[i],na.rm = T))
}


```





```{r apobec weight sum plot TCN, echo=FALSE, fig.height=18, fig.width=7}

signature.breakdown.means <- as.data.frame(matrix(data=NA, nrow=length(unique(signature.breakdown.TCN$Name)),ncol=2))
colnames(signature.breakdown.means) <- c("Name","mean")
signature.breakdown.means$Name <- unique(signature.breakdown.TCN$Name)
for(i in 1:nrow(signature.breakdown.means)){
  signature.breakdown.means$mean[i] <- median(signature.breakdown.TCN$`APOBEC sum`[which(signature.breakdown.TCN$Name==signature.breakdown.means$Name[i])],na.rm = T) 
}

signature.breakdown.means <- signature.breakdown.means[order(signature.breakdown.means$mean,decreasing = T),]

signature.breakdown.TCN$Name <- factor(signature.breakdown.TCN$Name, levels = rev(signature.breakdown.means$Name))

TCN.to.TKN.APOBECplot <- ggplot(data = signature.breakdown.TCN) + geom_point(aes(x = `APOBEC sum`, y= Name),alpha=0.5) + theme_bw() + labs(x="Proportion of mutation from APOBEC signatures per tumor", y= "Mutation", title="TCN to TKN mutations and the proportion \nof the trinucleotide context that was APOBEC (2 or 13)", subtitle="Ranked in descending order of median proportion")
# TCN.to.TKN.APOBECplot
ggsave(plot = TCN.to.TKN.APOBECplot,filename = "Figures/TCN.to.TKN.APOBECplot.pdf",width = 7,height = 15)
```

```{r display TCN data table, echo=F}
TCN.prev.df <- cbind(TCN.prev.df,NA,NA)
colnames(TCN.prev.df)[6:7] <- c("Median APOBEC contribution", "Selection from APOBEC")
for(i in 1:nrow(TCN.prev.df)){
  TCN.prev.df$`Median APOBEC contribution`[i] <- round(median(signature.breakdown.TCN$`APOBEC sum`[which(signature.breakdown.TCN$Name==TCN.prev.df$Mutation[i])],na.rm = T),2)
}

TCN.prev.df$`Selection from APOBEC` <- round((TCN.prev.df$Frequency * TCN.prev.df$`Selection intensity` * TCN.prev.df$`Median APOBEC contribution`)/length(unique(HNSC.MAF$Unique_patient_identifier)),2)

TCN.prev.df <- TCN.prev.df[order(TCN.prev.df$`Selection from APOBEC`,decreasing = T),]

datatable(data = TCN.prev.df)

```

```{r display apobec plot tcn, echo=FALSE, fig.height=15, fig.width=7}
TCN.to.TKN.APOBECplot
```

# All mutations



```{r all mutations data frame, include=F}

just.ALL <- HNSC.selection.output.recur
ALL.names <- unique(just.ALL$Name)

ALL.prev.df <- as.data.frame(matrix(data = NA, nrow=length(ALL.names),ncol=5))
colnames(ALL.prev.df) <- c("Mutation","Frequency", "Trinuc context", "Alternative Nucleotide","Selection intensity")

ALL.prev.df$Mutation <- ALL.names

for(i in 1:nrow(ALL.prev.df)){
  ALL.prev.df$Frequency[i] <- length(which(just.ALL$Name == ALL.prev.df$Mutation[i]))
  ALL.prev.df$`Trinuc context`[i] <- just.ALL$Nucleotide_trinuc_context[which(just.ALL$Name == ALL.prev.df$Mutation[i])[1]]
  ALL.prev.df$`Alternative Nucleotide`[i] <- just.ALL$Alternative_Nucleotide[which(just.ALL$Name == ALL.prev.df$Mutation[i])[1]]
  ALL.prev.df$`Selection intensity`[i] <- round(just.ALL$Gamma_epistasis[which(just.ALL$Name == ALL.prev.df$Mutation[i])[1]],2)
}

ALL.prev.df <- ALL.prev.df[order(ALL.prev.df$Frequency,decreasing = T),]
```

```{r all mutations determining mutation weight, include = F}
library(deconstructSigs)

# load("HNSC_all_MAF.RData")
load("HNSC_MAF.RData")


load("trinuc_contexts_HPV_all.RData")
# length(unique(HNSC.HPVall.MAF$Unique_patient_identifier))
# length(trinuc.contexts[[2]])
tumor.trinuc.names <- NULL
for(i in 1:length(trinuc.contexts[[2]])){
  tumor.trinuc.names[i] <- rownames(trinuc.contexts[[2]][[i]]$weights)
}



# signatures.cosmic

# trinuc.contexts[[2]][[1]]$product
# as.matrix(trinuc.contexts[[2]][[1]]$weights) %*% as.matrix(signatures.cosmic) == as.matrix(trinuc.contexts[[2]][[1]]$product)


# head(just.TCW)
first.12 <- function(string_to_12){
  return(paste(unlist(strsplit(string_to_12,split = ""))[1:12],collapse = ""))
}

signature.breakdown.ALL <- as.data.frame(matrix(data = NA, nrow=nrow(just.ALL), ncol= 7))
colnames(signature.breakdown.ALL) <- c("Name","Tumor","Trinuc_context","Alternative_Nuc","deconstructSigs_context","APOBEC sum","NonAPOBEC sum")

signature.breakdown.ALL$Name <- just.ALL$Name
signature.breakdown.ALL$Tumor <- unlist(lapply(just.ALL$Tumor_origin,first.12))

for(i in 1:nrow(just.ALL)){
  if(is.na(just.ALL$Nucleotide_trinuc_context[i])){
    matches <- which(just.ALL$Nucleotide_chromosome_position == just.ALL$Nucleotide_chromosome_position[i] &
                       just.ALL$Alternative_Nucleotide == just.ALL$Alternative_Nucleotide[i])
    
    new.i <- matches[which(!is.na(just.ALL$Nucleotide_trinuc_context[matches]))]
    signature.breakdown.ALL$Trinuc_context[i] <- just.ALL$Nucleotide_trinuc_context[new.i]
  }else{
    signature.breakdown.ALL$Trinuc_context[i] <- just.ALL$Nucleotide_trinuc_context[i]
  }
}

signature.breakdown.ALL$Alternative_Nuc <- just.ALL$Alternative_Nucleotide

for(i in 1:nrow(signature.breakdown.ALL)){
  signature.breakdown.ALL$deconstructSigs_context[i] <- paste(
    strsplit(signature.breakdown.ALL$Trinuc_context[i],split = "")[[1]][1],
    "[",
    strsplit(signature.breakdown.ALL$Trinuc_context[i],split = "")[[1]][2],
    ">",
    signature.breakdown.ALL$Alternative_Nuc[i],
    "]",
    strsplit(signature.breakdown.ALL$Trinuc_context[i],split = "")[[1]][3],sep="")
  
}
# 
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="T[G>A]A")] <- "T[C>T]A"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="A[G>A]A")] <- "T[C>T]T"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="T[G>C]A")] <- "T[C>G]A"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="A[G>C]A")] <- "T[C>G]T"
# 
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="C[G>C]A")] <- "T[C>G]G"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="C[G>A]A")] <- "T[C>T]G"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="G[G>A]A")] <- "T[C>T]C"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="G[G>C]A")] <- "T[C>G]C"
# 
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="C[G>A]C")] <- "G[C>T]G"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="C[G>A]T")] <- "A[C>T]G"
# signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context=="T[A>G]G")] <- "C[T>C]A"

flip.function <- function(nucleotide){
  if(nucleotide=="A"){
    return("T")
  }
  if(nucleotide=="T"){
    return("A")
  }
  if(nucleotide=="C"){
    return("G")
  }
  if(nucleotide=="G"){
    return("C")
  }
  if(nucleotide=="N"){
    return("N")
  }
}

unique.sigs <- unique(signature.breakdown.ALL$deconstructSigs_context)
unique.sigs.splt <- strsplit(x = unique.sigs,split = "")
for(i in 1:length(unique.sigs.splt)){
  if(unique.sigs.splt[[i]][3] %in% c("G","A")){
    this.new.str <- unique.sigs.splt[[i]]
    this.new.str[7] <- flip.function(unique.sigs.splt[[i]][1])
    this.new.str[3] <- flip.function(unique.sigs.splt[[i]][3])
    this.new.str[5] <- flip.function(unique.sigs.splt[[i]][5])
    this.new.str[1] <- flip.function(unique.sigs.splt[[i]][7])
    signature.breakdown.ALL$deconstructSigs_context[which(signature.breakdown.ALL$deconstructSigs_context==unique.sigs[i])] <- paste(this.new.str,collapse = "")
  }
}

# unique(signature.breakdown.ALL$deconstructSigs_context)



for(i in 1:nrow(signature.breakdown.ALL)){
  if(length(which(tumor.trinuc.names==signature.breakdown.ALL$Tumor[i]))>0){
    this.context <- trinuc.contexts[[2]][[which(tumor.trinuc.names==signature.breakdown.ALL$Tumor[i])]]
    # t(this.context$weights)*signatures.cosmic[signature.breakdown.ALL$deconstructSigs_context[i]]
    # signatures.cosmic[signature.breakdown.ALL$deconstructSigs_context[i]]
    
    # this.context$tumor[,signature.breakdown.ALL$deconstructSigs_context[i]]
    # this.context$product[,signature.breakdown.ALL$deconstructSigs_context[i]]
    
    signature.breakdown.at.this.context <- t(this.context$weights)*signatures.cosmic[signature.breakdown.ALL$deconstructSigs_context[i]]
    
    signature.breakdown.ALL[i,c("APOBEC sum")] <- ifelse(all(is.na(signature.breakdown.at.this.context[c(2,13),])),NA,sum(signature.breakdown.at.this.context[c(2,13),]/this.context$product[,signature.breakdown.ALL$deconstructSigs_context[i]],na.rm = T))
    signature.breakdown.ALL[i,c("NonAPOBEC sum")] <- ifelse(all(is.na(signature.breakdown.at.this.context[c(1,3:12,14:30),])),NA,sum(signature.breakdown.at.this.context[c(1,3:12,14:30),]/this.context$product[,signature.breakdown.ALL$deconstructSigs_context[i]],na.rm = T))
  }
  # signature.breakdown.ALL$`APOBEC sum`[i] <- sum(signature.breakdown.ALL$Sig2[i],signature.breakdown.ALL$Sig13[i],na.rm = T)
}

#TODO: non apobec sum is just one minus the apobec sum... should we list all signatures for further analysis?

# head(signature.breakdown.ALL)

```


```{r apobec weight sum plot ALL, echo=FALSE, fig.height=40, fig.width=5}

signature.breakdown.means <- as.data.frame(matrix(data=NA, nrow=length(unique(signature.breakdown.ALL$Name)),ncol=3))
colnames(signature.breakdown.means) <- c("Name","median APOBEC","median NonAPOBEC")
signature.breakdown.means$Name <- unique(signature.breakdown.ALL$Name)
for(i in 1:nrow(signature.breakdown.means)){
  signature.breakdown.means$`median APOBEC`[i] <- median(signature.breakdown.ALL$`APOBEC sum`[which(signature.breakdown.ALL$Name==signature.breakdown.means$Name[i])],na.rm = T)
  signature.breakdown.means$`median NonAPOBEC`[i] <- median(signature.breakdown.ALL$`NonAPOBEC sum`[which(signature.breakdown.ALL$Name==signature.breakdown.means$Name[i])],na.rm = T)
}

signature.breakdown.means <- signature.breakdown.means[order(signature.breakdown.means$`median APOBEC`,decreasing = T),]

signature.breakdown.ALL$Name <- factor(signature.breakdown.ALL$Name, levels = rev(signature.breakdown.means$Name))

ALL.APOBECplot <- ggplot(data = signature.breakdown.ALL) + geom_point(aes(x = `APOBEC sum`, y= Name),alpha=0.2)+# geom_point(aes(x = `NonAPOBEC sum`, y= Name),alpha=0.2,color="red") + 
  theme_bw() + labs(x="Proportion of mutation from APOBEC signatures per tumor", y= "Mutation", title="All mutations and the proportion \nof the trinucleotide context that was APOBEC", subtitle="Ranked in descending order of \nmedian proportion APOBEC contribution")
# ALL.to.TKN.APOBECplot
ggsave(plot = ALL.APOBECplot,filename = "Figures/ALL.to.TKN.APOBECplot.pdf",width = 7,height = 40)
```

##  Mutations that have the highest prevalence and selection intensity in HNSCC

```{r display ALL data table, echo=F}
ALL.prev.df <- cbind(ALL.prev.df,NA,NA,NA,NA)
colnames(ALL.prev.df)[6:7] <- c("Median APOBEC contribution", "Selection from APOBEC")
colnames(ALL.prev.df)[8:9] <- c("Median NonAPOBEC contribution", "Selection from NonAPOBEC")
for(i in 1:nrow(ALL.prev.df)){
  ALL.prev.df$`Median APOBEC contribution`[i] <- round(median(signature.breakdown.ALL$`APOBEC sum`[which(signature.breakdown.ALL$Name==ALL.prev.df$Mutation[i])],na.rm = T),2)
  ALL.prev.df$`Median NonAPOBEC contribution`[i] <- round(median(signature.breakdown.ALL$`NonAPOBEC sum`[which(signature.breakdown.ALL$Name==ALL.prev.df$Mutation[i])],na.rm = T),2)
  
}

ALL.prev.df$`Selection from APOBEC` <- round((ALL.prev.df$Frequency * ALL.prev.df$`Selection intensity` * ALL.prev.df$`Median APOBEC contribution`)/length(unique(HNSC.MAF$Unique_patient_identifier)),2)

ALL.prev.df$`Selection from NonAPOBEC` <- round((ALL.prev.df$Frequency * ALL.prev.df$`Selection intensity` * ALL.prev.df$`Median NonAPOBEC contribution`)/length(unique(HNSC.MAF$Unique_patient_identifier)),2)

ALL.prev.df <- ALL.prev.df[order(ALL.prev.df$`Selection from APOBEC`,decreasing = T),]

datatable(data = ALL.prev.df)

# ALL.prev.df[which(ALL.prev.df$Mutation=="PIK3CA E545K 178936091A"),]

```

## Mutations with the highest median contribution from APOBEC signatures

```{r display apobec plot ALL, echo=FALSE, fig.height=40, fig.width=7, warning=FALSE}
ALL.APOBECplot
```


<!-- ## Comparing the distribution of APOBEC signatures in tumors and the selection intensity of variants -->

<!-- ### TCW to TKW -->

```{r plot gamma and APOBEC violin, eval=FALSE, fig.height=70, fig.width=14, message=FALSE, warning=FALSE, include=FALSE}
HNSC.selection.output.recur <- HNSC.selection.output.recur[order(HNSC.selection.output.recur$Gamma_epistasis,decreasing = T),]
# HNSC.selection.output.recur$Gamma_epistasis <- factor(HNSC.selection.output.recur$Gamma_epistasis, levels=unique(HNSC.selection.output.recur$Gamma_epistasis))
HNSC.selection.output.recur$Name <- factor(HNSC.selection.output.recur$Name, levels= rev(unique(HNSC.selection.output.recur$Name)))



library(ggplot2)

full.plot <- ggplot(data = HNSC.selection.output.recur) + geom_point(aes(y=Name, x=APOBEC_weight,color=as.factor(TCW_TKW)),alpha=0.5) + theme_bw() + theme(legend.position = "none")

ggsave(filename = "Figures/full_selection_and_APOBEC.pdf",plot = full.plot,height = 40)

library(grid);library(gridExtra)

# levels(HNSC.selection.output.recur$TCW_TKW)[levels(HNSC.selection.output.recur$TCW_TKW)==1] <- "TRUE"
# levels(HNSC.selection.output.recur$TCW_TKW)[levels(HNSC.selection.output.recur$TCW_TKW)==0] <- "FALSE"
HNSC.selection.output.recur$TCW_TKW[HNSC.selection.output.recur$TCW_TKW==0] <- "FALSE"
HNSC.selection.output.recur$TCW_TKW[HNSC.selection.output.recur$TCW_TKW==1] <- "TRUE"

selection.plot <- ggplot(data= HNSC.selection.output.recur) + geom_point(aes(x=Name, y=Gamma_epistasis,color=TCW_TKW)) + coord_flip() + theme_bw()

ggsave(selection.plot,  filename = "Figures/selection_desc.pdf",height=40)

combined.plot <- grid.arrange(full.plot,selection.plot,ncol=2)

ggsave(combined.plot,filename = "Figures/combined_selection_apobec.pdf", height = 40,width = 14)


selection.plot.log <- ggplot(data= HNSC.selection.output.recur) + geom_point(aes(x=Name, y=log10(Gamma_epistasis),color=TCW_TKW)) + coord_flip() + theme_bw()

combined.plot.log <- grid.arrange(full.plot,selection.plot.log,ncol=2)
ggsave(combined.plot.log,filename = "Figures/combined_selection_apobec_log.pdf", height = 40,width = 14)

```


<!-- ### TCN to TKN -->

```{r plot gamma and APOBEC violin TCN to TKN, eval=FALSE, fig.height=70, fig.width=14, message=FALSE, warning=FALSE, include=FALSE}
# TCN -> TKN

full.plot.TCN.TKN <- ggplot(data = HNSC.selection.output.recur) + geom_point(aes(y=Name, x=APOBEC_weight,color=as.factor(TCN_TKN)),alpha=0.5) + theme_bw() + theme(legend.position = "none")

ggsave(filename = "Figures/full_selection_and_APOBEC_TCN_TKN.pdf",plot = full.plot,height = 40)

library(grid);library(gridExtra)

# levels(HNSC.selection.output.recur$TCW_TKW)[levels(HNSC.selection.output.recur$TCW_TKW)==1] <- "TRUE"
# levels(HNSC.selection.output.recur$TCW_TKW)[levels(HNSC.selection.output.recur$TCW_TKW)==0] <- "FALSE"
HNSC.selection.output.recur$TCN_TKN[HNSC.selection.output.recur$TCN_TKN==0] <- "FALSE"
HNSC.selection.output.recur$TCN_TKN[HNSC.selection.output.recur$TCN_TKN==1] <- "TRUE"

selection.plot.TCN.TKN <- ggplot(data= HNSC.selection.output.recur) + geom_point(aes(x=Name, y=Gamma_epistasis,color=TCN_TKN)) + coord_flip() + theme_bw()

ggsave(selection.plot.TCN.TKN,  filename = "Figures/selection_desc_TCN_TKN.pdf",height=40)

combined.plot.TCN.TKN <- grid.arrange(full.plot.TCN.TKN,selection.plot.TCN.TKN,ncol=2)

ggsave(combined.plot.TCN.TKN,filename = "Figures/combined_selection_apobec_TCN_TKN.pdf", height = 40,width = 14)


selection.plot.log.TCN.TKN <- ggplot(data= HNSC.selection.output.recur) + geom_point(aes(x=Name, y=log10(Gamma_epistasis),color=TCN_TKN)) + coord_flip() + theme_bw()

combined.plot.log.TCN.TKN <- grid.arrange(full.plot.TCN.TKN,selection.plot.log.TCN.TKN,ncol=2)
ggsave(combined.plot.log.TCN.TKN,filename = "Figures/combined_selection_apobec_log_TCN_TKN.pdf", height = 40,width = 14)




```

```{r eval=FALSE, include=FALSE}

```












```{r prevalence prevalence plot, eval=F, include=FALSE}
# load("output_data/HNSC_selection_with_APOBEC_TCW_recur.RData")
# head(HNSC.selection.output.recur)

# load("~/Documents/Selection_analysis/HNSC/selection_output/HNSC_selection_output.RData")
#
# selection.all.muts <- as.tibble(selection.output$all_mutations) %>%
#   subset(freq>1)
#
# selection.all.muts$Name <- NA
# for(i in 1:nrow(selection.all.muts)){
#   selection.all.muts$Name[i] <- paste(selection.all.muts$Gene[i]," ",ifelse(!is.na(selection.all.muts$AA_Ref[i]),paste(selection.all.muts$AA_Ref[i],selection.all.muts$AA_Pos[i],selection.all.muts$AA_Change[i],sep=""),paste(selection.all.muts$Nuc_Ref[i],selection.all.muts$Nucleotide_position[i],selection.all.muts$Nuc_Change[i],"NCSNV")),sep="")
# }
# # colnames(selection.all.muts)
#
HNSC.selection.output.recur$Name_short <- NA
for(i in 1:nrow(HNSC.selection.output.recur)){
  HNSC.selection.output.recur$Name_short[i] <- paste(HNSC.selection.output.recur$Gene[i]," ",ifelse(!is.na(HNSC.selection.output.recur$Amino_acid_reference[i]),paste(HNSC.selection.output.recur$Amino_acid_reference[i],HNSC.selection.output.recur$Amino_acid_position[i],HNSC.selection.output.recur$Amino_acid_alternative[i],sep=""),paste(HNSC.selection.output.recur$Reference_Nucleotide[i],HNSC.selection.output.recur$Nucleotide_chromosome_position[i],HNSC.selection.output.recur$Alternative_Nucleotide[i],"NCSNV")),sep="")
}
head(HNSC.selection.output.recur)

#TODO: should we then delete the mutations that are not recurrent within these smaller datasets?

recur.hpv.pos <- subset(HNSC.selection.output.recur, HPV_call=="positive")
recur.hpv.pos <- subset(recur.hpv.pos, Name_short %in% names(table(recur.hpv.pos$Name_short))[which(table(recur.hpv.pos$Name_short)>1)]) #just the recurrent mutations 
hpv.pos.names <- unique(recur.hpv.pos$Name_short)

recur.hpv.neg <- subset(HNSC.selection.output.recur, HPV_call=="negative")
recur.hpv.neg <- subset(recur.hpv.neg, Name_short %in% names(table(recur.hpv.neg$Name_short))[which(table(recur.hpv.neg$Name_short)>1)]) #just the recurrent mutations 
hpv.neg.names <- unique(recur.hpv.neg$Name_short)

intersect(hpv.pos.names,hpv.neg.names)
both.names <- union(hpv.neg.names,hpv.pos.names)

#make a dataframe with nrows == union of names, and columnes the number in each type of tumor and then the prevalence in each type

prevalence.df <- as.data.frame(matrix(data = NA, nrow = length(both.names), ncol=5))
colnames(prevalence.df) <- c("Name","tally_HPVpos","tally_HPVneg","prev_HPVpos","prev_HPVneg")

prevalence.df$Name <- both.names
prevalence.df$tally_HPVpos <- 0
prevalence.df$tally_HPVneg <- 0
for(i in 1:nrow(prevalence.df)){
  if(length(which(hpv.pos.names == prevalence.df$Name[i]))>0){
    prevalence.df$tally_HPVpos[i] <- length(which(recur.hpv.pos$Name_short == prevalence.df$Name[i]))
  }
  if(length(which(hpv.neg.names == prevalence.df$Name[i]))>0){
    prevalence.df$tally_HPVneg[i] <- length(which(recur.hpv.neg$Name_short == prevalence.df$Name[i]))
  }
}

prevalence.df$prev_HPVpos <- prevalence.df$tally_HPVpos/length(unique(HNSC.selection.output$Tumor_origin[which(HNSC.selection.output$HPV_call=="positive")]))
prevalence.df$prev_HPVneg <- prevalence.df$tally_HPVneg/length(unique(HNSC.selection.output$Tumor_origin[which(HNSC.selection.output$HPV_call=="negative")])) #TODO why only 411? did one tumor have no SNP?

library(ggrepel)
prev_plot <- ggplot(data = prevalence.df) + geom_point(aes(x = prev_HPVneg, y = prev_HPVpos),alpha=0.1,size=5) + geom_text_repel(data = subset(prevalence.df, (prev_HPVneg>0.025 | prev_HPVpos>0.04) | (prev_HPVneg>0.0 & prev_HPVpos>0)),aes(x = prev_HPVneg, y = prev_HPVpos,label=Name),box.padding = 1.12,size=5,color="darkred",fontface="bold") + theme_bw() + labs(x=bquote("Prevalence in "~HPV^{"-"}~ "tumors"), y=bquote("Prevalence in "~HPV^{"+"}~ "tumors")) + theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold")) #+ geom_label(aes(x = prev_HPVneg, y = prev_HPVpos, label=Name))

ggsave(filename = "Figures/prevalence_plot.png",plot = prev_plot,height = 5,width = 5)

# recur.hpv.pos$prevalence <- recur.hpv.pos$Nucleotide_change_tally,

# ggplot(data = HNSC.selection.output.recur) + geom_point(
# HNSC.selection.output[which(HNSC.selection.output$HPV_call=="positive" & HNSC.selection.output$Gene=="TP53"),]

```

```{r gamma gamma plot, eval=F}


load("~/Documents/Selection_analysis/HNSC_HPVpos/selection_output/HNSC_HPVpos_selection_output.RData")
HPV.pos.selectionoutput <- selection.output
HPV.pos.selection_minrecur <- subset(HPV.pos.selectionoutput$all_mutations, freq>1)


load("~/Documents/Selection_analysis/HNSC_HPVneg/selection_output/HNSC_HPVneg_selection_output.RData")
HPV.neg.selectionoutput <- selection.output
HPV.neg.selection_minrecur <- subset(HPV.neg.selectionoutput$all_mutations, freq>1)


HPV.pos.selection_minrecur$Name_short <- NA
for(i in 1:nrow(HPV.pos.selection_minrecur)){
  HPV.pos.selection_minrecur$Name_short[i] <- paste(HPV.pos.selection_minrecur$Gene[i]," ",ifelse(!is.na(HPV.pos.selection_minrecur$AA_Ref[i]),paste(HPV.pos.selection_minrecur$AA_Ref[i],HPV.pos.selection_minrecur$AA_Pos[i],HPV.pos.selection_minrecur$AA_Change[i],sep=""),paste(HPV.pos.selection_minrecur$Nuc_Ref[i],HPV.pos.selection_minrecur$Nucleotide_position[i],HPV.pos.selection_minrecur$Nuc_Change[i],"NCSNV")),sep="")
}


HPV.neg.selection_minrecur$Name_short <- NA
for(i in 1:nrow(HPV.neg.selection_minrecur)){
  HPV.neg.selection_minrecur$Name_short[i] <- paste(HPV.neg.selection_minrecur$Gene[i]," ",ifelse(!is.na(HPV.neg.selection_minrecur$AA_Ref[i]),paste(HPV.neg.selection_minrecur$AA_Ref[i],HPV.neg.selection_minrecur$AA_Pos[i],HPV.neg.selection_minrecur$AA_Change[i],sep=""),paste(HPV.neg.selection_minrecur$Nuc_Ref[i],HPV.neg.selection_minrecur$Nucleotide_position[i],HPV.neg.selection_minrecur$Nuc_Change[i],"NCSNV")),sep="")
}


gamma.df <- cbind(prevalence.df,NA,NA)
colnames(gamma.df)[6:7] <- c("gamma_HPVpos","gamma_HPVneg")

for(i in 1:nrow(gamma.df)){
  if(length(which(HPV.pos.selection_minrecur$Name_short == gamma.df$Name[i]))>0){
    gamma.df$gamma_HPVpos[i] <- HPV.pos.selection_minrecur$gamma_epistasis[which(HPV.pos.selection_minrecur$Name_short == gamma.df$Name[i])][1]
  }
  if(length(which(HPV.neg.selection_minrecur$Name_short == gamma.df$Name[i]))>0){
    gamma.df$gamma_HPVneg[i] <- HPV.neg.selection_minrecur$gamma_epistasis[which(HPV.neg.selection_minrecur$Name_short == gamma.df$Name[i])][1]
  }
}

fancy_scientific <- function(l) {
  # turn in to character string in scientific notation
  l <- format(l, scientific = TRUE)
  l <- gsub("0e\\+00","0",l)
  # quote the part before the exponent to keep all the digits
  l <- gsub("^(.*)e", "'\\1'e", l)
  # turn the 'e+' into plotmath format
  l <- gsub("e", "%*%10^", l)
  # return this as an expression
  parse(text=l)
}



library(ggrepel)
gamma_plot <- ggplot(data = gamma.df) + geom_point(aes(x = log10(gamma_HPVneg), y = log10(gamma_HPVpos)),alpha=0.5,size=5) + geom_text_repel(aes(x = log10(gamma_HPVneg), y = log10(gamma_HPVpos),label=Name),box.padding = 0.8,size=5,color="darkred",fontface="bold") + theme_bw() + labs(x="Log of selection intensity in HPV- tumors", y="Log of selection intensity in HPV+ tumors") + theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold")) #+ geom_label(aes(x = prev_HPVneg, y = prev_HPVpos, label=Name))


gamma_plot <- ggplot(data = gamma.df) + geom_point(aes(x = gamma_HPVneg, y = gamma_HPVpos),alpha=0.5,size=5) + geom_text_repel(aes(x = gamma_HPVneg, y = gamma_HPVpos,label=Name),box.padding = 0.8,size=5,color="darkred",fontface="bold") + theme_bw() + labs(x=bquote("Selection intensity in "~HPV^{"-"}~ "tumors"), y=bquote("Selection intensity in "~HPV^{"+"}~ "tumors")) + theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold"))  + scale_y_log10(labels = fancy_scientific) + scale_x_log10(labels = fancy_scientific)

ggsave(filename = "Figures/gamma_gamma_plot.png",plot = gamma_plot,height = 5,width = 5)

```

```{r gamma and prob apobec plot, fig.height=40, fig.width=7}

signature.breakdown.TCN

TCN.prev.df$Mutation[order(TCN.prev.df$`Selection intensity`,decreasing = T)]

signature.breakdown.ALL
ALL.prev.df


deconstructSigs_contexts_TCW <- unique(signature.breakdown.TCW$deconstructSigs_context)
deconstructSigs_contexts_TCN <- unique(signature.breakdown.TCN$deconstructSigs_context)

signature.breakdown.ALL$`APOBEC trinuc mutation` <- "NonAPOBEC mutation"
signature.breakdown.ALL$`APOBEC trinuc mutation`[which(signature.breakdown.ALL$deconstructSigs_context %in% deconstructSigs_contexts_TCN)] <- "TCN to TKN"
signature.breakdown.ALL$`APOBEC trinuc mutation`[which(signature.breakdown.ALL$deconstructSigs_context %in% deconstructSigs_contexts_TCW)] <- "TCW to TKW"

signature_all_for_plot <- signature.breakdown.ALL

signature_all_for_plot$Name <- factor(signature_all_for_plot$Name, levels=ALL.prev.df$Mutation[rev(order(ALL.prev.df$`Selection intensity`,decreasing = T))])

All.APOBECplot.gamma_ordered <- ggplot(data = signature_all_for_plot) + geom_point(aes(x = `APOBEC sum`, y= Name,color=`APOBEC trinuc mutation`),alpha=0.5) + theme_bw() + labs(x="Proportion of mutation from APOBEC signatures per tumor", y= "Mutation", title="Mutations and the proportion \nof the trinucleotide context that was APOBEC", subtitle="Ranked in descending order of selection intensity")

ggsave(plot = All.APOBECplot.gamma_ordered,filename = "Figures/gamma_vs_APOBEC_weight_proportion.pdf",width = 7,height = 40)

```



```{r are APOBEC TCW type mutations enriched in higher gamma, eval=F}

ALL.prev.df

ALL.prev.df$`APOBEC type TCW` <- "False"
ALL.prev.df$`APOBEC type TCW`[which(ALL.prev.df$Mutation %in% signature.breakdown.TCW$Name)] <- "True"

distribution.of.APOBEC.TCW <- ggplot(data = ALL.prev.df) + geom_violin(aes(y=log10(`Selection intensity`),x=`APOBEC type TCW`)) + geom_jitter(aes(y=log10(`Selection intensity`),x=`APOBEC type TCW`),alpha=0.5,width  = 0.1) + theme_bw() + labs(title="The distribution of selection intensities for APOBEC and nonAPOBEC mutations")

distribution.of.APOBEC.TCW <- ggplot(data = ALL.prev.df) + geom_violin(aes(y=`Selection intensity`,x=`APOBEC type TCW`),width=0.5) + geom_jitter(aes(y=`Selection intensity`,x=`APOBEC type TCW`),alpha=0.5,width  = 0.1) + theme_bw() + labs(y="Selection intensity") + scale_y_log10(labels=fancy_scientific)+ theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold")) + labs(x="TCW to TKW")

ggsave(plot = distribution.of.APOBEC.TCW,filename = "Figures/distribution_of_APOBEC_TCW.png",height = 5,width = 5)


wilcox.test(data=ALL.prev.df, `Selection intensity`~`APOBEC type TCW`)


### trying to get the violin plot

# ALL.prev.df.logged <- ALL.prev.df
# ALL.prev.df.logged$`Selection intensity` <- log10(ALL.prev.df.logged$`Selection intensity`)
# 
# distribution.of.APOBEC.TCW.viol <- ggplot(data = ALL.prev.df.logged) + geom_violin(aes(y=`Selection intensity`,x=as.factor(`APOBEC type TCW`))) + geom_jitter(aes(y=`Selection intensity`,x=as.factor(`APOBEC type TCW`)),alpha=0.2,height = 0.1) + theme_bw() + labs(title="The distribution of selection intensities for \nAPOBEC and nonAPOBEC mutations")



```

```{r are APOBEC TCN type mutations enriched in higher gamma, eval=F}

ALL.prev.df

ALL.prev.df$`APOBEC type TCN` <- "False"
ALL.prev.df$`APOBEC type TCN`[which(ALL.prev.df$Mutation %in% signature.breakdown.TCN$Name)] <- "True"

distribution.of.APOBEC.TCN <- ggplot(data = ALL.prev.df) + geom_violin(aes(y=log10(`Selection intensity`),x=`APOBEC type TCN`)) + geom_jitter(aes(y=log10(`Selection intensity`),x=`APOBEC type TCN`),alpha=0.5,width=0.1) + theme_bw() + labs(title="The distribution of selection intensities for APOBEC and nonAPOBEC mutations")


distribution.of.APOBEC.TCN <- ggplot(data = ALL.prev.df) + geom_violin(aes(y=`Selection intensity`,x=`APOBEC type TCN`),width=0.5) + geom_jitter(aes(y=`Selection intensity`,x=`APOBEC type TCN`),alpha=0.5,width=0.1) + theme_bw() + labs(y="Selection intensity") + scale_y_log10(labels=fancy_scientific)+ theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold")) + labs(x="TCN to TKN")


ggsave(plot = distribution.of.APOBEC.TCN,filename = "Figures/distribution_of_APOBEC_TCN.png",height = 5,width = 5)


wilcox.test(data=ALL.prev.df, `Selection intensity`~`APOBEC type TCN`)
```


```{r Bernoulli-weighted logistic regression of probability mutation was induced by APOBEC against selection intensity (try gamma or log gamma), eval=F}

signature.breakdown.ALL$`Selection intensity` <- NA

for(i in 1:nrow(signature.breakdown.ALL)){
 signature.breakdown.ALL$`Selection intensity`[i] <- ALL.prev.df$`Selection intensity`[which(ALL.prev.df$Mutation==signature.breakdown.ALL$Name[i])] 
}



APOBEC.vs.gamma <- ggplot(data = signature.breakdown.ALL,aes(x=`Selection intensity`, y=`APOBEC sum`)) + geom_point(alpha=0.2) + theme_bw() + labs(y="Proportion of mutation contributed by APOBEC \ntrinucleotide context")+ stat_smooth(method="glm",  se=FALSE)

APOBEC.vs.gamma.log <- ggplot(data = signature.breakdown.ALL,aes(x=log10(`Selection intensity`), y=`APOBEC sum`)) + geom_point(alpha=0.2) + theme_bw() + labs(y="Proportion of mutation contributed by APOBEC \ntrinucleotide context") + stat_smooth(method="glm",  se=FALSE)

# signature.breakdown.ALL[order(signature.breakdown.ALL$`Selection intensity`,decreasing = T),]

ggsave(filename = "Figures/APOBECvsGamma.png",plot = APOBEC.vs.gamma)
ggsave(filename = "Figures/APOBECvsGamma_log.png",plot = APOBEC.vs.gamma.log)

summary(lm(`APOBEC sum`~`Selection intensity`,data=signature.breakdown.ALL))

summary(lm(`APOBEC sum`~log10(`Selection intensity`),data=signature.breakdown.ALL))

```



```{r Mutation type vs gamma from APOBEC or not, eval=F}

head(signature.breakdown.ALL)
head(ALL.prev.df)

ALL.prev.df.split <- as.data.frame(matrix(data = NA,nrow = 2*nrow(ALL.prev.df),ncol=5))
colnames(ALL.prev.df.split) <- c("Name","Selection from mutational context","APOBEC context", "TCW to TKW", "TCN to TKN")
ALL.prev.df.split$Name <- c(ALL.prev.df$Mutation,ALL.prev.df$Mutation)
ALL.prev.df.split$`APOBEC context` <-  c(rep("APOBEC signatures",nrow(ALL.prev.df.split)/2),rep("NonAPOBEC signatures",nrow(ALL.prev.df.split)/2))


for(i in 1:length(unique(ALL.prev.df.split$Name))){
 these.pos <- which(ALL.prev.df.split$Name ==  unique(ALL.prev.df.split$Name)[i])
  ALL.prev.df.split$`TCW to TKW`[these.pos] <- ALL.prev.df$`APOBEC type TCW`[i]
  ALL.prev.df.split$`TCN to TKN`[these.pos] <- ALL.prev.df$`APOBEC type TCN`[i]
  ALL.prev.df.split$`Selection from mutational context`[these.pos] <- c(ALL.prev.df$`Selection from APOBEC`[i],ALL.prev.df$`Selection from NonAPOBEC`[i])
}

head(ALL.prev.df.split)

ALL.prev.df.split$short_name <- NA
for(i in 1:nrow(ALL.prev.df.split)){
  ALL.prev.df.split$short_name[i] <- paste(strsplit(x = ALL.prev.df.split$Name[i], split=" ")[[1]][1:2],collapse = " ")
}


# ggplot(data = ALL.prev.df.split) +geom_violin(aes(x=`APOBEC context`, y = `Selection from mutational context`))  + geom_jitter(aes(x=`APOBEC context`, y = `Selection from mutational context`))

selection.from.contexts <- ggplot(data = ALL.prev.df.split) +geom_violin(aes(x=`APOBEC context`, y = log10(`Selection from mutational context`)))  + geom_jitter(aes(x=`APOBEC context`, y = log10(`Selection from mutational context`))) + labs(x="Mutational signature attributed to mutation")


selection.from.contexts <- ggplot(data = ALL.prev.df.split) +geom_violin(aes(x=`APOBEC context`, y = `Selection from mutational context`))  + geom_jitter(aes(x=`APOBEC context`, y = `Selection from mutational context`)) + scale_y_log10(labels=fancy_scientific) + labs(x="Mutational signatures attributed to mutation", y="Proportion of selection intensity attributed to signature and \nweighted by mutation prevalence among tumors") + theme_bw()

ALL.prev.df.split$`APOBEC context` <- factor(ALL.prev.df.split$`APOBEC context`, levels = c("NonAPOBEC signatures", "APOBEC signatures"))

selection.from.contexts.labels <- ggplot(data = ALL.prev.df.split) +#geom_violin(aes(x=`APOBEC context`, y = `Selection from mutational context`))  +
  geom_point(aes(x=`APOBEC context`, y = `Selection from mutational context`),alpha=0.2) + 
  geom_text_repel(data = subset(ALL.prev.df.split,`APOBEC context`=="APOBEC signatures" & `Selection from mutational context` > 100),aes(x=`APOBEC context`, y = `Selection from mutational context`,label=short_name),color="darkred",fontface="bold") + #scale_y_log10(labels=fancy_scientific) + 
  geom_text_repel(data = subset(ALL.prev.df.split,`APOBEC context`=="NonAPOBEC signatures" & `Selection from mutational context` > 650),aes(x=`APOBEC context`, y = `Selection from mutational context`,label=short_name),box.padding = .31,color="darkred",fontface="bold") + 
  labs(x="Mutational signatures attributed to mutation", y="Selection intensity attributed to signature, weighted \nby mutation prevalence among tumors") + theme_bw() + theme(axis.text=element_text(size=12), axis.title=element_text(size=14,face="bold"))

ggsave(filename = "Figures/selection_from_process.png",plot = selection.from.contexts)

ggsave(filename = "Figures/selection_from_process_labels.png",plot = selection.from.contexts.labels)

wilcox.test(`Selection from mutational context`~`APOBEC context`,data = ALL.prev.df.split)

```





