{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Yale data processing\n",
    "\n",
    "- Data from Hedberg et al, 2016 (doi 10.1172/JCI82066)\n",
    "\n",
    "- Annotation requires https://github.com/mskcc/vcf2maf.\n",
    "    - Version used: v1.6.13."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# IMPORTS\n",
    "from io import StringIO\n",
    "import gzip\n",
    "\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Somatic mutations from supplementary data\n",
    "\n",
    "- DNPs/DNVs merged in parse_hnc_contexts.ipynb in hnc_apobec/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_excel('/Users/sgg/Dropbox/Townsend/head_neck_data/JCI82066.sdt1-10.xlsx',\n",
    "                   sheet_name=\"Table S10\", header=[0,1])\n",
    "\n",
    "df.columns = df.columns.droplevel(0)\n",
    "\n",
    "df.index.name = 'patient_id'\n",
    "df.columns.name = None\n",
    "df.reset_index(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>Group</th>\n",
       "      <th>Variant Status</th>\n",
       "      <th>Status</th>\n",
       "      <th>Gene</th>\n",
       "      <th>AA change</th>\n",
       "      <th>Fisher</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Position</th>\n",
       "      <th>Base Change</th>\n",
       "      <th>...</th>\n",
       "      <th>Adj Nonref Read Ratio</th>\n",
       "      <th>LOH in Primary</th>\n",
       "      <th>Ref Cov</th>\n",
       "      <th>Nonref Cov</th>\n",
       "      <th>Nonref Read Ratio</th>\n",
       "      <th>Adj Nonref Read Ratio</th>\n",
       "      <th>LOH in SLNM/MR</th>\n",
       "      <th>Ref cov</th>\n",
       "      <th>Nonref cov</th>\n",
       "      <th>Nonref allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PY-1</td>\n",
       "      <td>Metachronous Recurrence</td>\n",
       "      <td>Transmitted</td>\n",
       "      <td>Coding-missense</td>\n",
       "      <td>NPC1</td>\n",
       "      <td>R711I</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>chr18</td>\n",
       "      <td>21123532</td>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>...</td>\n",
       "      <td>0.204082</td>\n",
       "      <td>LOH</td>\n",
       "      <td>33</td>\n",
       "      <td>13</td>\n",
       "      <td>0.282609</td>\n",
       "      <td>1.34576</td>\n",
       "      <td>na</td>\n",
       "      <td>72</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PY-1</td>\n",
       "      <td>Metachronous Recurrence</td>\n",
       "      <td>Transmitted</td>\n",
       "      <td>Coding-silent</td>\n",
       "      <td>SSH1</td>\n",
       "      <td>P199P</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>chr12</td>\n",
       "      <td>109201543</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>...</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>LOH</td>\n",
       "      <td>31</td>\n",
       "      <td>14</td>\n",
       "      <td>0.311111</td>\n",
       "      <td>1.48148</td>\n",
       "      <td>LOH</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PY-1</td>\n",
       "      <td>Metachronous Recurrence</td>\n",
       "      <td>Primary Only</td>\n",
       "      <td>Coding-missense</td>\n",
       "      <td>C11orf16</td>\n",
       "      <td>H437Y</td>\n",
       "      <td>0.000456</td>\n",
       "      <td>chr11</td>\n",
       "      <td>8942958</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>...</td>\n",
       "      <td>0.201207</td>\n",
       "      <td>LOH</td>\n",
       "      <td>63</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>LOH</td>\n",
       "      <td>109</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PY-1</td>\n",
       "      <td>Metachronous Recurrence</td>\n",
       "      <td>Transmitted</td>\n",
       "      <td>Coding-missense</td>\n",
       "      <td>SLCO2B1</td>\n",
       "      <td>I211S</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>chr11</td>\n",
       "      <td>74880401</td>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>...</td>\n",
       "      <td>0.248016</td>\n",
       "      <td>LOH</td>\n",
       "      <td>79</td>\n",
       "      <td>5</td>\n",
       "      <td>0.059524</td>\n",
       "      <td>0.283447</td>\n",
       "      <td>LOH</td>\n",
       "      <td>80</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PY-1</td>\n",
       "      <td>Metachronous Recurrence</td>\n",
       "      <td>Transmitted</td>\n",
       "      <td>Coding-missense</td>\n",
       "      <td>ATP8B4</td>\n",
       "      <td>A618V</td>\n",
       "      <td>0.000077</td>\n",
       "      <td>chr15</td>\n",
       "      <td>50212513</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>...</td>\n",
       "      <td>0.200437</td>\n",
       "      <td>LOH</td>\n",
       "      <td>125</td>\n",
       "      <td>11</td>\n",
       "      <td>0.080882</td>\n",
       "      <td>0.385154</td>\n",
       "      <td>LOH</td>\n",
       "      <td>127</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  patient_id                    Group Variant Status           Status  \\\n",
       "0       PY-1  Metachronous Recurrence    Transmitted  Coding-missense   \n",
       "1       PY-1  Metachronous Recurrence    Transmitted    Coding-silent   \n",
       "2       PY-1  Metachronous Recurrence   Primary Only  Coding-missense   \n",
       "3       PY-1  Metachronous Recurrence    Transmitted  Coding-missense   \n",
       "4       PY-1  Metachronous Recurrence    Transmitted  Coding-missense   \n",
       "\n",
       "       Gene AA change    Fisher    Chr   Position Base Change      ...       \\\n",
       "0      NPC1     R711I  0.000001  chr18   21123532         C>A      ...        \n",
       "1      SSH1     P199P  0.000037  chr12  109201543         G>A      ...        \n",
       "2  C11orf16     H437Y  0.000456  chr11    8942958         G>A      ...        \n",
       "3   SLCO2B1     I211S  0.000400  chr11   74880401         T>G      ...        \n",
       "4    ATP8B4     A618V  0.000077  chr15   50212513         G>A      ...        \n",
       "\n",
       "   Adj Nonref Read Ratio  LOH in Primary  Ref Cov Nonref Cov  \\\n",
       "0               0.204082             LOH       33         13   \n",
       "1               0.285714             LOH       31         14   \n",
       "2               0.201207             LOH       63          0   \n",
       "3               0.248016             LOH       79          5   \n",
       "4               0.200437             LOH      125         11   \n",
       "\n",
       "  Nonref Read Ratio  Adj Nonref Read Ratio  LOH in SLNM/MR  Ref cov  \\\n",
       "0          0.282609                1.34576              na       72   \n",
       "1          0.311111                1.48148             LOH       45   \n",
       "2          0.000000                      0             LOH      109   \n",
       "3          0.059524               0.283447             LOH       80   \n",
       "4          0.080882               0.385154             LOH      127   \n",
       "\n",
       "  Nonref cov Nonref allele  \n",
       "0          0             -  \n",
       "1          0             -  \n",
       "2          0             -  \n",
       "3          0             -  \n",
       "4          0             -  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Transmitted                          1593\n",
       "Primary Only                          368\n",
       "Metachronous Recurrence Only          235\n",
       "Synchronous Nodal Metastasis Only     208\n",
       "Name: Variant Status, dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Variant Status'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extend annotation df to have row for each sample (not just each mutation)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- for every patient:\n",
    "    - for each row (mutation), duplicate for samples specified by 'Variant Status'\n",
    "    - combine new mutations into single dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "status_dict = {\n",
    "    'Transmitted': ['T', 'R'],\n",
    "    'Primary Only': ['T'],\n",
    "    'Metachronous Recurrence Only': ['R'],\n",
    "    'Synchronous Nodal Metastasis Only': ['R'],\n",
    "}\n",
    "\n",
    "def mut_series_from_status(s):\n",
    "    \"\"\"Return list of pd.Series annotations: one for each sample.\"\"\"\n",
    "    s_list = []\n",
    "    sample_suffixes = status_dict[s['Variant Status']]\n",
    "    for suffix in sample_suffixes:\n",
    "        temp_s = s.copy()\n",
    "        temp_s['sample_id'] = ''.join([s.patient_id, suffix])\n",
    "        s_list.append(temp_s)\n",
    "    return s_list\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "s_list = []\n",
    "for ind, g in df.groupby('patient_id'):\n",
    "    for ind, i in g.iterrows():\n",
    "        s_list.extend(mut_series_from_status(i))\n",
    "\n",
    "\n",
    "cols = ['sample_id'] + list(df.columns)\n",
    "\n",
    "df = pd.concat(s_list, axis=1, ignore_index=True).transpose()[cols]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PY-9T has no mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>R</th>\n",
       "      <th>T</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PY-1</th>\n",
       "      <td>86</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-10</th>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-11</th>\n",
       "      <td>31</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-12</th>\n",
       "      <td>85</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-13</th>\n",
       "      <td>6</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-14</th>\n",
       "      <td>68</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-15</th>\n",
       "      <td>436</td>\n",
       "      <td>413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-16</th>\n",
       "      <td>199</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-17</th>\n",
       "      <td>205</td>\n",
       "      <td>256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-19</th>\n",
       "      <td>7</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-20</th>\n",
       "      <td>71</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-21</th>\n",
       "      <td>151</td>\n",
       "      <td>152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-22</th>\n",
       "      <td>100</td>\n",
       "      <td>104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-23</th>\n",
       "      <td>59</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-24</th>\n",
       "      <td>45</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-25</th>\n",
       "      <td>80</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-3</th>\n",
       "      <td>29</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-4</th>\n",
       "      <td>83</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-5</th>\n",
       "      <td>91</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-6</th>\n",
       "      <td>20</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-7</th>\n",
       "      <td>15</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-8</th>\n",
       "      <td>113</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PY-9</th>\n",
       "      <td>52</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              R    T\n",
       "patient_id          \n",
       "PY-1         86   87\n",
       "PY-10         4    3\n",
       "PY-11        31   33\n",
       "PY-12        85   84\n",
       "PY-13         6   47\n",
       "PY-14        68   70\n",
       "PY-15       436  413\n",
       "PY-16       199  150\n",
       "PY-17       205  256\n",
       "PY-19         7   86\n",
       "PY-20        71   79\n",
       "PY-21       151  152\n",
       "PY-22       100  104\n",
       "PY-23        59   60\n",
       "PY-24        45   48\n",
       "PY-25        80   88\n",
       "PY-3         29   11\n",
       "PY-4         83   50\n",
       "PY-5         91   84\n",
       "PY-6         20    7\n",
       "PY-7         15   20\n",
       "PY-8        113   29\n",
       "PY-9         52    0"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.groupby('patient_id')['sample_id'].apply(lambda v: v.str[-1].value_counts())\\\n",
    "    .unstack().fillna(0).astype(int)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge DNVs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# print(list(df.columns))\n",
    "\n",
    "df = df[['sample_id', 'patient_id', 'Chr', 'Position', 'Base Change', 'Gene']].copy().reset_index()\n",
    "\n",
    "df.rename(columns={'Chr': 'chrom', 'Position': 'pos',\n",
    "                   'Base Change': 'change', 'Gene': 'symbol'}, inplace=True)\n",
    "\n",
    "\n",
    "df['ref'], df['alt'] = zip(*df.change.apply(lambda v: v.split('>')))\n",
    "\n",
    "# df.chrom = df.chrom.str.lstrip('chr')\n",
    "\n",
    "df.drop('change', axis=1, inplace=True)\n",
    "\n",
    "df.chrom = df.chrom.apply(lambda x: 'chrX' if x == 'chr24' else x)\n",
    "\n",
    "# REMOVE DUPLICATE MUTATIONS\n",
    "# df.groupby(['sample_id', 'chrom', 'pos']).filter(lambda d: len(d)>1)\n",
    "cols = df.columns\n",
    "df = df.groupby(['sample_id', 'chrom', 'pos']).first().reset_index()[cols]\n",
    "\n",
    "df.drop('index', axis=1, inplace=True)\n",
    "\n",
    "# IDENTIFY DUPLICATE MUTATIONS\n",
    "\n",
    "# Remove second mutation in DNVs\n",
    "df['pos_next'] = df.pos + 1\n",
    "\n",
    "df['categ'] = 'SNV'\n",
    "dnv_drop = df.merge(df.reset_index(), left_on=['sample_id', 'chrom', 'pos_next'], \n",
    "                    right_on=['sample_id', 'chrom', 'pos'], \n",
    "                    suffixes=['', '_right']).set_index('index')\n",
    "drop_inds = dnv_drop.index.values\n",
    "dnv = df.reset_index().merge(df.reset_index(), left_on=['sample_id', 'chrom', 'pos_next'], \n",
    "               right_on=['sample_id', 'chrom', 'pos'], \n",
    "               suffixes=['', '_right']).set_index('index')\n",
    "dnv_inds = dnv.index.values\n",
    "\n",
    "df.drop(drop_inds, inplace=True)\n",
    "df.loc[dnv_inds, 'categ'] = 'DNV'\n",
    "\n",
    "def drop_lower(v):\n",
    "    if v[0].isupper() and v[1].islower():\n",
    "        return v[0]\n",
    "    elif v[1].isupper() and v[0].islower():\n",
    "        return v[1]\n",
    "    else:\n",
    "        raise Exception('Bad value: {}'.format(v))\n",
    "\n",
    "for i in df[(df.categ == 'SNV') & (dnv.ref.apply(len) > 1)].itertuples():\n",
    "    ind = i.Index\n",
    "    ref = drop_lower(i.ref)\n",
    "    alt = drop_lower(i.alt)\n",
    "    df.loc[ind, 'ref'] = ref\n",
    "    df.loc[ind, 'alt'] = alt\n",
    "\n",
    "df.ref = df.ref.str.upper()\n",
    "df.alt = df.alt.str.upper()\n",
    "\n",
    "df.drop('pos_next', axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SNV    3949\n",
       "DNV      23\n",
       "Name: categ, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.categ.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Modify mutations table for use with maf2maf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.symbol = df.symbol.astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "rename_dict = {'chrom': 'Chromosome',\n",
    "'pos': 'Start_Position',\n",
    "'ref': 'Reference_Allele',\n",
    "'alt': 'Tumor_Seq_Allele2',\n",
    "'sample_id': 'Tumor_Sample_Barcode'}\n",
    "df.rename(columns=rename_dict, inplace=True)\n",
    "\n",
    "df.Chromosome = df.Chromosome.str[3:]\n",
    "chroms = [str(i) for i in pd.np.arange(22) + 1] + ['X', 'Y']\n",
    "df.Chromosome = df.Chromosome.astype(pd.Categorical(chroms, categories=chroms, ordered=True))\n",
    "\n",
    "df = df.sort_values(['Chromosome', 'Start_Position', 'Tumor_Sample_Barcode']).reset_index()\n",
    "df.drop('index', axis=1, inplace=True)\n",
    "df.rename(columns={'Transcript_ID': 'Refseq_NM'}, inplace=True)\n",
    "\n",
    "# All categories are SNV or DNV, so set end position accordingly\n",
    "df['End_Position'] = df.Start_Position.where(df.categ == 'SNV', df.Start_Position + 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>patient_id</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>symbol</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>categ</th>\n",
       "      <th>End_Position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PY-1R</td>\n",
       "      <td>PY-1</td>\n",
       "      <td>1</td>\n",
       "      <td>1325617</td>\n",
       "      <td>CCNL2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>1325617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PY-1T</td>\n",
       "      <td>PY-1</td>\n",
       "      <td>1</td>\n",
       "      <td>1325617</td>\n",
       "      <td>CCNL2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>1325617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PY-8R</td>\n",
       "      <td>PY-8</td>\n",
       "      <td>1</td>\n",
       "      <td>1887253</td>\n",
       "      <td>KIAA1751</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>1887253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PY-6R</td>\n",
       "      <td>PY-6</td>\n",
       "      <td>1</td>\n",
       "      <td>2103775</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>2103775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PY-17T</td>\n",
       "      <td>PY-17</td>\n",
       "      <td>1</td>\n",
       "      <td>6100684</td>\n",
       "      <td>KCNAB2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>6100684</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Tumor_Sample_Barcode patient_id Chromosome  Start_Position    symbol  \\\n",
       "0                PY-1R       PY-1          1         1325617     CCNL2   \n",
       "1                PY-1T       PY-1          1         1325617     CCNL2   \n",
       "2                PY-8R       PY-8          1         1887253  KIAA1751   \n",
       "3                PY-6R       PY-6          1         2103775     PRKCZ   \n",
       "4               PY-17T      PY-17          1         6100684    KCNAB2   \n",
       "\n",
       "  Reference_Allele Tumor_Seq_Allele2 categ  End_Position  \n",
       "0                G                 A   SNV       1325617  \n",
       "1                G                 A   SNV       1325617  \n",
       "2                C                 T   SNV       1887253  \n",
       "3                C                 T   SNV       2103775  \n",
       "4                G                 A   SNV       6100684  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Export and reannotate\n",
    "```python\n",
    "df.to_csv('yale_for_anno.maf', sep='\\t', index=False)\n",
    "```\n",
    "\n",
    "```bash\n",
    "# RUN maf2maf.pl and gzip\n",
    "maf2maf.pl --input-maf yale_for_anno.maf --output-maf yale_annotated.maf\n",
    "gzip yale_annotated.maf\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import annotated and filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_table(gzip.open('output_data/yale_annotated.maf.gz'), comment='#')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove recurrent tumors\n",
    "df = df[df.Tumor_Sample_Barcode.str.endswith('T')].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       ".                 1891\n",
       "common_variant      56\n",
       "Name: FILTER, dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.FILTER.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Strip common variants\n",
    "df = df[df.FILTER != 'common_variant'].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Missense_Mutation         1313\n",
       "Silent                     428\n",
       "Nonsense_Mutation           72\n",
       "Splice_Site                 29\n",
       "Splice_Region               15\n",
       "Intron                      14\n",
       "RNA                          8\n",
       "3'Flank                      4\n",
       "Translation_Start_Site       2\n",
       "5'Flank                      2\n",
       "Nonstop_Mutation             2\n",
       "5'UTR                        1\n",
       "3'UTR                        1\n",
       "Name: Variant_Classification, dtype: int64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.Variant_Classification.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filter intronic and flanking mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[~df.Variant_Classification.isin(['Intron', \"3'Flank\", \"5'Flank\"])].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Missense_Mutation         1313\n",
       "Silent                     428\n",
       "Nonsense_Mutation           72\n",
       "Splice_Site                 29\n",
       "Splice_Region               15\n",
       "RNA                          8\n",
       "Translation_Start_Site       2\n",
       "Nonstop_Mutation             2\n",
       "5'UTR                        1\n",
       "3'UTR                        1\n",
       "Name: Variant_Classification, dtype: int64"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.Variant_Classification.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>...</th>\n",
       "      <th>ExAC_AC_AN_Adj</th>\n",
       "      <th>ExAC_AC_AN</th>\n",
       "      <th>ExAC_AC_AN_AFR</th>\n",
       "      <th>ExAC_AC_AN_AMR</th>\n",
       "      <th>ExAC_AC_AN_EAS</th>\n",
       "      <th>ExAC_AC_AN_FIN</th>\n",
       "      <th>ExAC_AC_AN_NFE</th>\n",
       "      <th>ExAC_AC_AN_OTH</th>\n",
       "      <th>ExAC_AC_AN_SAS</th>\n",
       "      <th>ExAC_FILTER</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOCK7</td>\n",
       "      <td>85440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>63119695</td>\n",
       "      <td>63119695</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BCL10</td>\n",
       "      <td>8915</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>85736573</td>\n",
       "      <td>85736573</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NR5A2</td>\n",
       "      <td>2494</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>200012919</td>\n",
       "      <td>200012919</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EFCAB14</td>\n",
       "      <td>9813</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>47182102</td>\n",
       "      <td>47182102</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ITGB3BP</td>\n",
       "      <td>23421</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>63912521</td>\n",
       "      <td>63912521</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>2/106020</td>\n",
       "      <td>2/106200</td>\n",
       "      <td>0/9054</td>\n",
       "      <td>0/11192</td>\n",
       "      <td>0/7850</td>\n",
       "      <td>1/6596</td>\n",
       "      <td>1/54242</td>\n",
       "      <td>0/692</td>\n",
       "      <td>0/16394</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 123 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo_Symbol  Entrez_Gene_Id  Center NCBI_Build Chromosome  Start_Position  \\\n",
       "0       DOCK7           85440     NaN     GRCh37          1        63119695   \n",
       "1       BCL10            8915     NaN     GRCh37          1        85736573   \n",
       "2       NR5A2            2494     NaN     GRCh37          1       200012919   \n",
       "3     EFCAB14            9813     NaN     GRCh37          1        47182102   \n",
       "4     ITGB3BP           23421     NaN     GRCh37          1        63912521   \n",
       "\n",
       "   End_Position Strand Variant_Classification Variant_Type     ...       \\\n",
       "0      63119695      +      Missense_Mutation          SNP     ...        \n",
       "1      85736573      +      Missense_Mutation          SNP     ...        \n",
       "2     200012919      +      Missense_Mutation          SNP     ...        \n",
       "3      47182102      +      Missense_Mutation          SNP     ...        \n",
       "4      63912521      +      Missense_Mutation          SNP     ...        \n",
       "\n",
       "  ExAC_AC_AN_Adj ExAC_AC_AN ExAC_AC_AN_AFR ExAC_AC_AN_AMR  ExAC_AC_AN_EAS  \\\n",
       "0            NaN        NaN            NaN            NaN             NaN   \n",
       "1            NaN        NaN            NaN            NaN             NaN   \n",
       "2            NaN        NaN            NaN            NaN             NaN   \n",
       "3            NaN        NaN            NaN            NaN             NaN   \n",
       "4       2/106020   2/106200         0/9054        0/11192          0/7850   \n",
       "\n",
       "  ExAC_AC_AN_FIN ExAC_AC_AN_NFE ExAC_AC_AN_OTH ExAC_AC_AN_SAS  ExAC_FILTER  \n",
       "0            NaN            NaN            NaN            NaN          NaN  \n",
       "1            NaN            NaN            NaN            NaN          NaN  \n",
       "2            NaN            NaN            NaN            NaN          NaN  \n",
       "3            NaN            NaN            NaN            NaN         PASS  \n",
       "4         1/6596        1/54242          0/692        0/16394         PASS  \n",
       "\n",
       "[5 rows x 123 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('output_data/yale_filtered.maf', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Reference aside: clinical data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "OC, oral cavity\n",
    "P, pharynx\n",
    "L, larynx, LN, lymph node (cervical)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient ID</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Age at Diagnosis (yr)</th>\n",
       "      <th>Location of Primary Site</th>\n",
       "      <th>Path. stage</th>\n",
       "      <th>AJCC stage</th>\n",
       "      <th>Adjuvant treatment</th>\n",
       "      <th>Time to recurrence (mo)</th>\n",
       "      <th>Metastasis site</th>\n",
       "      <th>patient_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PY-1</td>\n",
       "      <td>M</td>\n",
       "      <td>78</td>\n",
       "      <td>oral cavity</td>\n",
       "      <td>T3 N1 M0</td>\n",
       "      <td>III</td>\n",
       "      <td>None</td>\n",
       "      <td>2.3</td>\n",
       "      <td>pharynx</td>\n",
       "      <td>metachronous_recurrent_tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PY-3</td>\n",
       "      <td>F</td>\n",
       "      <td>67</td>\n",
       "      <td>oral cavity</td>\n",
       "      <td>T2 N2B M0</td>\n",
       "      <td>IVA</td>\n",
       "      <td>RT</td>\n",
       "      <td>6.3</td>\n",
       "      <td>oral cavity</td>\n",
       "      <td>metachronous_recurrent_tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PY-4</td>\n",
       "      <td>M</td>\n",
       "      <td>44</td>\n",
       "      <td>?</td>\n",
       "      <td>T4A N2B M0</td>\n",
       "      <td>IVA</td>\n",
       "      <td>RT, Cisplatin, Vectibix</td>\n",
       "      <td>6.8</td>\n",
       "      <td>pharynx</td>\n",
       "      <td>metachronous_recurrent_tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PY-5</td>\n",
       "      <td>M</td>\n",
       "      <td>52</td>\n",
       "      <td>oral cavity</td>\n",
       "      <td>T3 N0 M0</td>\n",
       "      <td>III</td>\n",
       "      <td>None</td>\n",
       "      <td>4.3</td>\n",
       "      <td>larynx</td>\n",
       "      <td>metachronous_recurrent_tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PY-6</td>\n",
       "      <td>M</td>\n",
       "      <td>52</td>\n",
       "      <td>pharynx</td>\n",
       "      <td>TX NX MX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IMRT, cetuximab, pemetrexed</td>\n",
       "      <td>10.5</td>\n",
       "      <td>lymph node (cervical)</td>\n",
       "      <td>metachronous_recurrent_tumor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Patient ID Sex  Age at Diagnosis (yr) Location of Primary Site Path. stage  \\\n",
       "0       PY-1   M                     78              oral cavity    T3 N1 M0   \n",
       "1       PY-3   F                     67              oral cavity   T2 N2B M0   \n",
       "2       PY-4   M                     44                        ?  T4A N2B M0   \n",
       "3       PY-5   M                     52              oral cavity    T3 N0 M0   \n",
       "4       PY-6   M                     52                  pharynx    TX NX MX   \n",
       "\n",
       "  AJCC stage           Adjuvant treatment  Time to recurrence (mo)  \\\n",
       "0        III                         None                      2.3   \n",
       "1        IVA                           RT                      6.3   \n",
       "2        IVA      RT, Cisplatin, Vectibix                      6.8   \n",
       "3        III                         None                      4.3   \n",
       "4        NaN  IMRT, cetuximab, pemetrexed                     10.5   \n",
       "\n",
       "         Metastasis site                  patient_type  \n",
       "0                pharynx  metachronous_recurrent_tumor  \n",
       "1            oral cavity  metachronous_recurrent_tumor  \n",
       "2                pharynx  metachronous_recurrent_tumor  \n",
       "3                 larynx  metachronous_recurrent_tumor  \n",
       "4  lymph node (cervical)  metachronous_recurrent_tumor  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Clinical data. Text extracted from pdf.\n",
    "\n",
    "metachronous = \"\"\"Patient ID\tSex\tAge at Diagnosis (yr)\tLocation of Primary Site\tPath. stage\tAJCC stage\tAdjuvant treatment\tTime to recurrence (mo)\tMetastasis site\tpatient_type\n",
    "PY-1\tM\t78\tOC\tT3 N1 M0\tIII\tNone\t2.3\tP\tmetachronous_recurrent_tumor\n",
    "PY-3\tF\t67\tOC\tT2 N2B M0\tIVA\tRT\t6.3\tOC\tmetachronous_recurrent_tumor\n",
    "PY-4\tM\t44\t?\tT4A N2B M0\tIVA\tRT, Cisplatin, Vectibix\t6.8\tP\tmetachronous_recurrent_tumor\n",
    "PY-5\tM\t52\tOC\tT3 N0 M0\tIII\tNone\t4.3\tL\tmetachronous_recurrent_tumor\n",
    "PY-6\tM\t52\tP\tTX NX MX\tNA\tIMRT, cetuximab, pemetrexed\t10.5\tLN\tmetachronous_recurrent_tumor\n",
    "PY-7\tM\t49\tL\tT4A N1 M0\tIVA\tNone\t2.9\tLN\tmetachronous_recurrent_tumor\n",
    "PY-8\tM\t76\tOC\tTX NX MX\tNA\tRT\t33.9\tP\tmetachronous_recurrent_tumor\n",
    "PY-9\tM\t60\tOC\tT2 N0 MX\tII\tNone\t4.7\tLN\tmetachronous_recurrent_tumor\n",
    "PY-10\tM\t58\tOC\tTX NX MX\tNA\tNone\t2.2\tOC\tmetachronous_recurrent_tumor\n",
    "PY-11\tM\t74\tOC\tT3 N2C M0\tIVA\tRT, Cisplatin\t9.7\tOC\tmetachronous_recurrent_tumor\"\"\"\n",
    "\n",
    "synchronous = \"\"\"Patient ID\tSex\tAge at Diagnosis (yr)\tLocation of Primary Site\tPath. stage\tAJCC stage\tpatient_type\n",
    "PY-12\tF\t46\tOC\tT3 N2C M0\tIVA\tsynchronous_nodal_metastasis\n",
    "PY-13\tM\t53\tOC\tT3 N2B M0\tIVA\tsynchronous_nodal_metastasis\n",
    "PY-14\tM\t71\tL\tT3 N1 MX\tIII\tsynchronous_nodal_metastasis\n",
    "PY-15\tM\t63\tL\tT3 N2C M0\tIVA\tsynchronous_nodal_metastasis\n",
    "PY-16\tM\t70\tP\tT1 N2B MX\tIVA\tsynchronous_nodal_metastasis\n",
    "PY-17\tF\t68\tL\tT4A N2A MX\tIVA\tsynchronous_nodal_metastasis\n",
    "PY-19\tM\t73\tP\tT2 N1 MX\tIII\tsynchronous_nodal_metastasis\n",
    "PY-20\tF\t79\tOC\tT1 N1 MX\tIII\tsynchronous_nodal_metastasis\n",
    "PY-21\tM\t49\tL\tT4A N2C MO\tIVA\tsynchronous_nodal_metastasis\n",
    "PY-22\tM\t49\tOC\tT4A N2C M0\tIVA\tsynchronous_nodal_metastasis\n",
    "PY-23\tM\t56\tP\tT4A N2B M0\tIVA\tsynchronous_nodal_metastasis\n",
    "PY-24\tM\t73\tOC\tT4A N2C M0\tIVA\tsynchronous_nodal_metastasis\n",
    "PY-25\tM\t64\tOC\tT1 N2 M0\tIVA\tsynchronous_nodal_metastasis\"\"\"\n",
    "\n",
    "s_met = pd.read_table(StringIO(metachronous))\n",
    "s_syn = pd.read_table(StringIO(synchronous))\n",
    "clinical = pd.concat([s_met, s_syn], ignore_index=True)\n",
    "\n",
    "site_dict = {'OC': 'oral cavity',\n",
    "'P': 'pharynx',\n",
    "'L': 'larynx',\n",
    "'LN': 'lymph node (cervical)'}\n",
    "\n",
    "clinical['Metastasis site'] = clinical['Metastasis site'].apply(lambda v: site_dict[v] if type(v) == str else v)\n",
    "clinical['Location of Primary Site'] = clinical['Location of Primary Site'].apply(lambda v: site_dict[v] if v != '?' else v)\n",
    "\n",
    "clinical[s_met.columns].head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:anaconda3-4.1.0]",
   "language": "python",
   "name": "conda-env-anaconda3-4.1.0-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.4"
  },
  "toc": {
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": "block",
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
